<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor • recountWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor">
<meta property="og:description" content="recountWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recountWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.25.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/recount-workflow.html">recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/recountWorkflow/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>recount workflow: accessing over 70,000 human
RNA-seq samples with Bioconductor</h1>
                        <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus, Baltimore, MD, 21205, USACenter for Computational Biology, Johns
Hopkins University, Baltimore, MD, 21205,
USA<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Abhinav
Nellore</h4>
            <address class="author_afil">
      Department of Biomedical Engineering, Oregon Health and Science
University, Portland, OR, 97239, USADepartment of Surgery, Oregon Health
and Science University, Portland, OR, 97239, USAComputational Biology
Program, Oregon Health and Science University, Portland, OR, 97239,
USA<br><h4 data-toc-skip class="author">Andrew
E. Jaffe</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus, Baltimore, MD, 21205, USACenter for Computational Biology, Johns
Hopkins University, Baltimore, MD, 21205, USADepartment of
Biostatistics, Johns Hopkins Bloomberg School of Public Health,
Baltimore, MD, 21205, USADepartment of Mental Health, Johns Hopkins
Bloomberg School of Public Health, Baltimore, MD, 21205,
USA<br><small class="dont-index">Source: <a href="https://github.com/LieberInstitute/recountWorkflow/blob/HEAD/vignettes/recount-workflow.Rmd" class="external-link"><code>vignettes/recount-workflow.Rmd</code></a></small>
      <div class="hidden name"><code>recount-workflow.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      The recount2 resource is composed of over 70,000 uniformly
      processed human RNA-seq samples spanning TCGA and SRA, including
      GTEx. The processed data can be accessed via the recount2 website
      and the <code>recount</code> Bioconductor package. This workflow
      explains in detail how to use the <code>recount</code> package and
      how to integrate it with other Bioconductor packages for several
      analyses that can be carried out with the recount2 resource. In
      particular, we describe how the coverage count matrices were
      computed in recount2 as well as different ways of obtaining public
      metadata, which can facilitate downstream analyses. Step-by-step
      directions show how to do a gene-level differential expression
      analysis, visualize base-level genome coverage data, and perform
      an analyses at multiple feature levels. This workflow thus
      provides further information to understand the data in recount2
      and a compendium of R code to use the data.
    </div>
    
<p><strong>R version</strong>: R version 4.3.0 (2023-04-21 ucrt)</p>
<p><strong>Bioconductor version</strong>: 3.17</p>
<p><strong>Package</strong>: 1.24.0</p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>RNA sequencing (RNA-seq) is now the most widely used high-throughput
assay for measuring gene expression. In a typical RNA-seq experiment,
several million reads are sequenced per sample. The reads are often
aligned to the reference genome using a splice-aware aligner to identify
where reads originated. Resulting alignment files are then used to
compute count matrices for several analyses such as identifying
differentially expressed genes. The Bioconductor project <span class="citation">(1)</span> has many contributed packages that
specialize in analyzing this type of data and previous workflows have
explained how to use them <span class="citation">(2–4)</span>. Initial
steps are typically focused on generating the count matrices. Some
pre-computed matrices have been made available via the ReCount project
<span class="citation">(5)</span> or Bioconductor Experiment data
packages such as the <code>airway</code> dataset <span class="citation">(6)</span>. The pre-computed count matrices in ReCount
have been useful to RNA-seq methods developers and to researchers
seeking to avoid the computationally intensive process of creating these
matrices. In the years since ReCount was published, hundreds of new
RNA-seq projects have been carried out, and researchers have shared the
data publicly.</p>
<p>We recently uniformly processed over 70,000 publicly available human
RNA-seq samples, and made the data available via the recount2 resource
<span class="citation">(7)</span> at <a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">jhubiostatistics.shinyapps.io/recount/</a>.
Samples in recount2 are grouped by project (over 2,000) originating from
the Sequence Read Archive, the Genotype-Tissue Expression study (GTEx)
and the Cancer Genome Atlas (TCGA). The processed data can be accessed
via the <code>recount</code> Bioconductor package available at <a href="http://bioconductor.org/packages/recount" class="external-link">bioconductor.org/packages/recount</a>.
Together, recount2 and the <code>recount</code> Bioconductor package
should be considered a successor to ReCount.</p>
<p>Due to space constraints, the recount2 publication <span class="citation">(7)</span> did not cover how to use the
<code>recount</code> package and other useful information for carrying
out analyses with recount2 data. We describe how the count matrices in
recount2 were generated. We also review the R code necessary for using
the recount2 data, whose details are important because some of this code
involves multiple Bioconductor packages and changing default options. We
further show: (a) how to augment metadata that comes with datasets with
metadata learned from natural language processing of associated papers
as well as expression data (b) how to perform differential expression
analyses, and (c) how to visualize the base-pair data available from
recount2.</p>
</div>
<div class="section level2">
<h2 id="analysis-of-rna-seq-data-available-at-recount2">Analysis of RNA-seq data available at recount2<a class="anchor" aria-label="anchor" href="#analysis-of-rna-seq-data-available-at-recount2"></a>
</h2>
<div class="section level3">
<h3 id="recount2-overview">recount2 overview<a class="anchor" aria-label="anchor" href="#recount2-overview"></a>
</h3>
<p>The recount2 resource provides expression data summarized at
different feature levels to enable novel cross-study analyses. Generally
when investigators use the term <em>expression</em>, they think about
gene expression. But more information can be extracted from RNA-seq
data. Once RNA-seq reads have been aligned to the reference genome it is
possible to determine the number of aligned reads overlapping each
base-pair resulting in the genome base-pair coverage curve as shown in
Figure @ref(fig:Figure1). In the example shown in Figure
@ref(fig:Figure1), most of the reads overlap known exons from a gene.
Those reads can be used to compute a count matrix at the exon or gene
feature levels. Some reads span exon-exon junctions (jx) and while most
match the annotation, some do not (jx 3 and 4). An exon-exon junction
count matrix can be used to identify differentially expressed junctions,
which can show which isoforms are differentially expressed given
sufficient coverage. For example, junctions 2 and 5 are unique to
isoform 2, while junction 6 is unique to isoform 1. The genome base-pair
coverage data can be used with <code>derfinder</code> <span class="citation">(8)</span> to identify expressed regions; some of them
could be unannotated exons, which together with the exon-exon junction
data could help establish new isoforms.</p>
<div class="figure" style="text-align: center">
<img src="Figure1.png" alt="Overview of the data available in recount2. Reads (pink boxes) aligned to the reference genome can be used to compute a base-pair coverage curve and identify exon-exon junctions (split reads). Gene and exon count matrices are generated using annotation information providing the gene (green boxes) and exon (blue boxes) coordinates together with the base-level coverage curve. The reads spanning exon-exon junctions (jx) are used to compute a third count matrix that might include unannotated junctions (jx 3 and 4). Without using annotation information, expressed regions (orange box) can be determined from the base-level coverage curve to then construct data-driven count matrices." width="100%"><p class="caption">
Overview of the data available in recount2. Reads (pink boxes) aligned
to the reference genome can be used to compute a base-pair coverage
curve and identify exon-exon junctions (split reads). Gene and exon
count matrices are generated using annotation information providing the
gene (green boxes) and exon (blue boxes) coordinates together with the
base-level coverage curve. The reads spanning exon-exon junctions (jx)
are used to compute a third count matrix that might include unannotated
junctions (jx 3 and 4). Without using annotation information, expressed
regions (orange box) can be determined from the base-level coverage
curve to then construct data-driven count matrices.
</p>
</div>
<p>recount2 provides gene, exon, and exon-exon junction count matrices
both in text format and <em>RangedSummarizedExperiment</em> objects
(rse) <span class="citation">(9)</span> as shown in Figure
@ref(fig:Figure2). These rse objects provide information about the
expression features (for example gene IDs) and the samples. In this
workflow we will explain how to add metadata to the rse objects in
recount2 in order to ask biological questions. recount2 also provides
coverage data in the form of bigWig files. All four features can be
accessed with the <code>recount</code> Bioconductor package <span class="citation">(7)</span>. <code>recount</code> also allows sending
queries to <code>snaptron</code> <span class="citation">(10)</span> to
search for specific exon-exon junctions.</p>
<div class="figure" style="text-align: center">
<img src="Figure2.png" alt="recount2 provides coverage count matrices in RangedSummarizedExperiment (rse) objects. Once the rse object has been downloaded and loaded into R, the feature information is accessed with rowRanges(rse) (blue box), the counts with assays(rse)\$counts (pink box) and the sample metadata with colData(rse) (green box). The sample metadata can be expanded using add\_predictions(rse) (orange box) or with custom code (brown box) matching by a unique sample identifier such as the SRA Run ID. The rse object is inside the purple box and matching data is highlighted in each box." width="100%"><p class="caption">
recount2 provides coverage count matrices in RangedSummarizedExperiment
(rse) objects. Once the rse object has been downloaded and loaded into
R, the feature information is accessed with rowRanges(rse) (blue box),
the counts with assays(rse)$counts (pink box) and the sample metadata
with colData(rse) (green box). The sample metadata can be expanded using
add_predictions(rse) (orange box) or with custom code (brown box)
matching by a unique sample identifier such as the SRA Run ID. The rse
object is inside the purple box and matching data is highlighted in each
box.
</p>
</div>
</div>
<div class="section level3">
<h3 id="packages-used-in-the-workflow">Packages used in the workflow<a class="anchor" aria-label="anchor" href="#packages-used-in-the-workflow"></a>
</h3>
<p>In this workflow we will use several Bioconductor packages. To
reproduce the entirety of this workflow, install the packages using the
following code after installing R 3.4.x from CRAN in order to use
Bioconductor version 3.5 or newer.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Install packages from Bioconductor</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"recount"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"limma"</span>, <span class="st">"edgeR"</span>, <span class="st">"DESeq2"</span>,</span>
<span>    <span class="st">"regionReport"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"gplots"</span>,</span>
<span>    <span class="st">"derfinder"</span>, <span class="st">"GenomicState"</span>, <span class="st">"bumphunter"</span>, <span class="st">"derfinderPlot"</span>, <span class="st">"sessioninfo"</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Once they are installed, load all the packages with the following
code.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/regionReport" class="external-link">"regionReport"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">"clusterProfiler"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/talgalili/gplots" class="external-link">"gplots"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lcolladotor/derfinder" class="external-link">"derfinder"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/GenomicState" class="external-link">"GenomicState"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rafalab/bumphunter" class="external-link">"bumphunter"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/derfinderPlot" class="external-link">"derfinderPlot"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">"sessioninfo"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="coverage-counts-provided-by-recount2">Coverage counts provided by recount2<a class="anchor" aria-label="anchor" href="#coverage-counts-provided-by-recount2"></a>
</h3>
<p>The most accessible features are the gene, exon and exon-exon
junction count matrices. This section explains them in greater detail.
Figure @ref(fig:Figure3) shows 16 RNA-seq reads, each 3 base-pairs long,
and a reference genome.</p>
<div class="figure" style="text-align: center">
<img src="Figure3.png" alt="RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs long are shown (pink boxes) alongside a reference genome that is 16 base-pairs long (white box)." width="70%"><p class="caption">
RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs
long are shown (pink boxes) alongside a reference genome that is 16
base-pairs long (white box).
</p>
</div>
<p>Reads in the recount2 resource were aligned with the splice-aware
Rail-RNA aligner <span class="citation">(11)</span>. Figure
@ref(fig:Figure4) shows the reads aligned to the reference genome. Some
of the reads are split as they span an exon-exon junction. Two of the
reads were soft clipped meaning that just a portion of the reads aligned
(top left in purple).</p>
<div class="figure" style="text-align: center">
<img src="Figure4.png" alt="Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are able to find the coordinates to which the reads map, even if they span exon-exon junctions (connected boxes). Rail-RNA soft clips some reads (purple boxes with rough edges) such that a portion of these reads align to the reference genome." width="70%"><p class="caption">
Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are
able to find the coordinates to which the reads map, even if they span
exon-exon junctions (connected boxes). Rail-RNA soft clips some reads
(purple boxes with rough edges) such that a portion of these reads align
to the reference genome.
</p>
</div>
<p>In order to compute the gene and exon count matrices we first have to
process the annotation, which for recount2 is Gencode v25 (CHR regions)
with hg38 coordinates. Although <code>recount</code> can generate count
matrices for other annotations using hg38 coordinates. Figure
@ref(fig:Figure5) shows two isoforms for a gene composed of 3 different
exons.</p>
<div class="figure" style="text-align: center">
<img src="Figure5.png" alt="Gene annotation. A single gene with two isoforms composed by three distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the first five base-pairs while exon 2 is common to both isoforms." width="70%"><p class="caption">
Gene annotation. A single gene with two isoforms composed by three
distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the
first five base-pairs while exon 2 is common to both isoforms.
</p>
</div>
<p>The coverage curve is at base-pair resolution, so if we are
interested in gene counts we have to be careful not to double count
base-pairs 1 through 5 that are shared by exons 1 and 3 (Figure
@ref(fig:Figure5)). Using the function <code><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">disjoin()</a></code> from
<code>GenomicRanges</code> <span class="citation">(12)</span> we
identified the distinct exonic sequences (disjoint exons). The following
code defines the exon coordinates that match Figure @ref(fig:Figure5)
and the resulting disjoint exons for our example gene. The resulting
disjoint exons are shown in Figure @ref(fig:Figure6).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"seq"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">13</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">8</span>, <span class="fl">15</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">exons</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]      seq       1-5      *</span></span>
<span><span class="co">##   [2]      seq       1-8      *</span></span>
<span><span class="co">##   [3]      seq     13-15      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">disjoin</a></span><span class="op">(</span><span class="va">exons</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]      seq       1-5      *</span></span>
<span><span class="co">##   [2]      seq       6-8      *</span></span>
<span><span class="co">##   [3]      seq     13-15      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="figure" style="text-align: center">
<img src="Figure6.png" alt="Disjoint exons. Windows of distinct exonic sequence for the example gene. Disjoint exons 1 and 2 form exon 1." width="70%"><p class="caption">
Disjoint exons. Windows of distinct exonic sequence for the example
gene. Disjoint exons 1 and 2 form exon 1.
</p>
</div>
<p>Now that we have disjoint exons, we can compute the base-pair
coverage for each of them as shown in Figure @ref(fig:Figure7). That is,
for each base-pair that corresponds to exonic sequence, we compute the
number of reads overlapping that given base-pair. For example, the first
base-pair is covered by 3 different reads and it does not matter whether
the reads themselves were soft clipped. Not all reads or bases of a read
contribute information to this step, as some do not overlap known exonic
sequence (light pink in Figure @ref(fig:Figure7)).</p>
<div class="figure" style="text-align: center">
<img src="Figure7.png" alt="Base-pair coverage counting for exonic base-pairs. At each exonic base-pair we compute the number of reads overlapping that given base-pair. The first base (orange arrow) has 3 reads overlapping that base-pair. Base-pair 11 has a coverage of 3 but does not overlap known exonic sequence, so that information is not used for the gene and exon count matrices (grey arrow). If a read partially overlaps exonic sequence, only the portion that overlaps is used in the computation (see right most read)." width="70%"><p class="caption">
Base-pair coverage counting for exonic base-pairs. At each exonic
base-pair we compute the number of reads overlapping that given
base-pair. The first base (orange arrow) has 3 reads overlapping that
base-pair. Base-pair 11 has a coverage of 3 but does not overlap known
exonic sequence, so that information is not used for the gene and exon
count matrices (grey arrow). If a read partially overlaps exonic
sequence, only the portion that overlaps is used in the computation (see
right most read).
</p>
</div>
<p>With base-pair coverage for the exonic sequences computed, the
coverage count for each distinct exon is simply the sum of the base-pair
coverage for each base in a given distinct exon. For example, the
coverage count for disjoint exon 2 is <span class="math inline">\(2 + 2
+ 3 = 7\)</span> as shown in Figure @ref(fig:Figure8). The gene coverage
count is then <span class="math inline">\(\sum_i^n
\texttt{coverage}_i\)</span> where <span class="math inline">\(n\)</span> is the number of exonic base-pairs for
the gene and is equal to the sum of the coverage counts for its disjoint
exons as shown in Figure @ref(fig:Figure8).</p>
<div class="figure" style="text-align: center">
<img src="Figure8.png" alt="Exon and gene coverage counts. The coverage counts for each disjoint exon are the sum of the base-pair coverage. The gene coverage count is the sum of the disjoint exons coverage counts." width="70%"><p class="caption">
Exon and gene coverage counts. The coverage counts for each disjoint
exon are the sum of the base-pair coverage. The gene coverage count is
the sum of the disjoint exons coverage counts.
</p>
</div>
<p>For the exons, recount2 provides the disjoint exons coverage count
matrix. It is possible to reconstruct the exon coverage count matrix by
summing the coverage count for the disjoint exons that compose each
exon. For example, the coverage count for exon 1 would be the sum of the
coverage counts for disjoint exons 1 and 2, that is <span class="math inline">\(19 + 7 = 26\)</span>. Some methods might assume
that double counting of the shared base-pairs was performed while others
assume or recommend the opposite.</p>
<div class="section level4">
<h4 id="scaling-coverage-counts">Scaling coverage counts<a class="anchor" aria-label="anchor" href="#scaling-coverage-counts"></a>
</h4>
<p>The coverage counts described previously are the ones actually
included in the rse objects in recount2 instead of typical read count
matrices. This is an important difference to keep in mind as most
methods were developed for read count matrices. Part of the sample
metadata available from recount2 includes the read length and number of
mapped reads. Given a target library size (40 million reads by default),
the coverage counts in recount2 can be scaled to read counts for a given
library size as shown in Equation @ref(eq:scale). Note that the
resulting scaled read counts are not necessarily integers so it might be
necessary to round them if a differential expression (DE) method assumes
integer data.</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{Read Length}} *
\frac{\text{target}}{\text{mapped}} = \text{scaled read counts}
  (\#eq:scale)
\end{equation}\]</span></p>
<p>From Figure @ref(fig:Figure4) we know that Rail-RNA soft clipped some
reads, so a more precise measure than the denominator of Equation
@ref(eq:scale) is the area under coverage (AUC) which is the sum of the
coverage for all base-pairs of the genome, regardless of the annotation
as shown in Figure @ref(fig:Figure9). Without soft clipping reads, the
AUC would be equal to the number of reads mapped multiplied by the read
length. So for our example gene, the scaled counts for a library size of
20 reads would be <span class="math inline">\(\frac{36}{45} * 20 =
16\)</span> and in general calculated with Equation @ref(eq:scale2). The
following code shows how to compute the AUC given a set of aligned reads
and reproduce a portion of Figure @ref(fig:Figure9).</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{AUC}} * \text{target} =
\text{scaled read counts}
  (\#eq:scale2)
\end{equation}\]</span></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Take the example and translate it to R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="va">reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"seq"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span></span>
<span>    start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">13</span>, <span class="fl">14</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="op">)</span>, width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Get the base-level genome coverage curve</span></span>
<span><span class="va">cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/coverage-methods.html" class="external-link">coverage</a></span><span class="op">(</span><span class="va">reads</span><span class="op">)</span><span class="op">$</span><span class="va">seq</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## AUC</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 45</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Code for reproducing the bottom portion of Figure 8.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"base_pair_coverage.pdf"</span>, width <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cov</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"o"</span>, col <span class="op">=</span> <span class="st">"violetred1"</span>, lwd <span class="op">=</span> <span class="fl">10</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"Genome"</span>, ylab <span class="op">=</span> <span class="st">"Coverage"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, cex.lab <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    bty <span class="op">=</span> <span class="st">"n"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/polygon.html" class="external-link">polygon</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">cov</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>    border <span class="op">=</span> <span class="cn">NA</span>, density <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, col <span class="op">=</span> <span class="st">"light blue"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span><span class="op">)</span>, <span class="va">cov</span>,</span>
<span>    col <span class="op">=</span> <span class="st">"violetred1"</span>, type <span class="op">=</span> <span class="st">"o"</span>,</span>
<span>    lwd <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="Figure9.png" alt="Area under coverage (AUC). The area under coverage is the sum of the base-pair coverage for all positions in the genome regardless of the annotation. It is the area under the base-level coverage curve shown as the light blue area under the pink curve." width="70%"><p class="caption">
Area under coverage (AUC). The area under coverage is the sum of the
base-pair coverage for all positions in the genome regardless of the
annotation. It is the area under the base-level coverage curve shown as
the light blue area under the pink curve.
</p>
</div>
<p>The <code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts()</a></code>
computes the scaled read counts for a target library size of 40 million
reads and we highly recommend using it before doing other analyses. The
following code shows how to use <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts()</a></code> and that the
resulting read counts per sample can be lower than the target size (40
million). This happens when not all mapped reads overlap known exonic
base-pairs of the genome. In our example, the gene has a scaled count of
16 reads for a library size of 20 reads, meaning that 4 reads did not
overlap exonic sequences.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Check that the number of reads is less than or equal to 40 million</span></span>
<span><span class="co">## after scaling.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span><span class="va">rse_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span>, round <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1e6</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##   22.62   29.97   34.00   31.96   34.86   36.78</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="enriching-the-annotation">Enriching the annotation<a class="anchor" aria-label="anchor" href="#enriching-the-annotation"></a>
</h4>
<p>Data in recount2 can be used for annotation-agnostic analyses and
enriching the known annotation. Just like exon and gene coverage count
matrices, recount2 provides exon-exon junction count matrices. These
matrices can be used to identify new isoforms (Figure
@ref(fig:Figure10)) or identify differentially expressed isoforms. For
example, exon-exon junctions 2, 5 and 6 in Figure @ref(fig:Figure1) are
only present in one annotated isoform. <code>Snaptron</code> <span class="citation">(10)</span> allows programatic and high-level queries
of the exon-exon junction information and its graphical user interface
is specially useful for visualizing this data. Inside R, the
<code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/snaptron_query.html" class="external-link">snaptron_query()</a></code> can be used
for searching specific exon-exon junctions in recount2.</p>
<div class="figure" style="text-align: center">
<img src="Figure10.png" alt="Exon-exon junctions go beyond the annotation. Reads spanning exon-exon junctions are highlighted and compared against the annotation. Three of them match the annotated junctions, but one (blue and orange read) spans an unannotated exon-exon junction with the left end matching the annotation and the right end hinting at a possible new isoform for this gene (blue and orange isoform)." width="70%"><p class="caption">
Exon-exon junctions go beyond the annotation. Reads spanning exon-exon
junctions are highlighted and compared against the annotation. Three of
them match the annotated junctions, but one (blue and orange read) spans
an unannotated exon-exon junction with the left end matching the
annotation and the right end hinting at a possible new isoform for this
gene (blue and orange isoform).
</p>
</div>
<p>The base-pair coverage data from recount2 can be used together with
<code>derfinder</code> <span class="citation">(8)</span> to identify
expressed regions of the genome, providing another annotation-agnostic
analysis of the expression data. Using the function
<code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html" class="external-link">expressed_regions()</a></code> we can identify regions of expression
based on a given data set in recount2. These regions might overlap known
exons but can also provide information about intron retention events
(Figure @ref(fig:Figure11)), improve detection of exon boundaries
(Figure @ref(fig:Figure12)), and help identify new exons (Fig
@ref(fig:Figure1)) or expressed sequences in intergenic regions. Using
<code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html" class="external-link">coverage_matrix()</a></code> we can compute a coverage matrix based on
the expressed regions or another set of genomic intervals. The resulting
matrix can then be used for a DE analysis, just like the exon, gene and
exon-exon junction matrices.</p>
<div class="figure" style="text-align: center">
<img src="Figure11.png" alt="Intron retention events. Some reads might align with known intronic segments of the genome and provide information for exploring intron retention events (pink read). Some might support an intron retention event or a new isoform when coupled with exon-exon junction data (orange read)." width="70%"><p class="caption">
Intron retention events. Some reads might align with known intronic
segments of the genome and provide information for exploring intron
retention events (pink read). Some might support an intron retention
event or a new isoform when coupled with exon-exon junction data (orange
read).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="Figure12.png" alt="Exon boundaries. Reads that go beyond the known exon boundaries can inform us of whether the annotated boundaries are correct or if there was a run-off transcription event." width="70%"><p class="caption">
Exon boundaries. Reads that go beyond the known exon boundaries can
inform us of whether the annotated boundaries are correct or if there
was a run-off transcription event.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="gene-level-analysis">Gene-level analysis<a class="anchor" aria-label="anchor" href="#gene-level-analysis"></a>
</h3>
<p>Having reviewed how the coverage counts in recount2 were produced, we
can now do a DE analysis. We will use data from 72 individuals spanning
the human lifespan, split into 6 age groups with SRA accession SRP045638
<span class="citation">(13)</span>. The function
<code><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study()</a></code> requires a SRA accession which can be
found using <code><a href="https://rdrr.io/pkg/recount/man/abstract_search.html" class="external-link">abstract_search()</a></code>.
<code><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study()</a></code> can then be used to download the gene
coverage count data as well as other expression features. The files are
saved in a directory named after the SRA accession, in this case
SRP045638.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount" class="external-link">"recount"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find the project ID by searching abstracts of studies</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/abstract_search.html" class="external-link">abstract_search</a></span><span class="op">(</span><span class="st">"human brain development by age"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      number_samples species</span></span>
<span><span class="co">## 1296             72   human</span></span>
<span><span class="co">##                                                                             abstract</span></span>
<span><span class="co">## 1296 RNAseq data of 36 samples across human brain development by age group from LIBD</span></span>
<span><span class="co">##        project</span></span>
<span><span class="co">## 1296 SRP045638</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the data if it is not there</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, type <span class="op">=</span> <span class="st">"rse-gene"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 19:38:00.521397 downloading file rse_gene.Rdata to SRP045638</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Check that the file was downloaded</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The coverage count matrices are provided as
<em>RangedSummarizedExperiment</em> objects (rse) <span class="citation">(9)</span>. These objects store information at the
feature level, the samples and the actual count matrix as shown in
Figure 1 of Love et al., 2016 <span class="citation">(3)</span>. Figure
@ref(fig:Figure2) shows the actual rse objects provided by recount2 and
how to access the different portions of the data. Using a unique sample
ID such as the SRA Run ID it is possible to expand the sample metadata.
This can be done using the predicted phenotype provided by
<code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html" class="external-link">add_predictions()</a></code> <span class="citation">(14)</span>,
pulling information from GEO via <code><a href="https://rdrr.io/pkg/recount/man/find_geo.html" class="external-link">find_geo()</a></code> and
<code><a href="https://rdrr.io/pkg/recount/man/geo_characteristics.html" class="external-link">geo_characteristics()</a></code>, or with custom code.</p>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>Using the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData()</a></code> function we can access sample
metadata. More information on these metadata is provided in the
supplementary material of the recount2 paper <span class="citation">(7)</span>, and we provide a brief review here. The rse
objects for SRA data sets include 21 columns with mostly technical
information. The GTEx and TCGA rse objects include additional metadata
as available from the raw sources. In particular, we compiled metadata
for GTEx using the v6 phenotype information available at <a href="gtexportal.org">gtexportal.org</a>, and we put together a large
table of TCGA case and sample information by combining information
accumulated across <a href="http://www.cancergenomicscloud.org/" class="external-link">Seven
Bridges’ Cancer Genomics Cloud</a> and <code>TCGAbiolinks</code> <span class="citation">(15)</span>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## One row per sample, one column per phenotype variable</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72 21</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Mostly technical variables are included</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "project"                                       </span></span>
<span><span class="co">##  [2] "sample"                                        </span></span>
<span><span class="co">##  [3] "experiment"                                    </span></span>
<span><span class="co">##  [4] "run"                                           </span></span>
<span><span class="co">##  [5] "read_count_as_reported_by_sra"                 </span></span>
<span><span class="co">##  [6] "reads_downloaded"                              </span></span>
<span><span class="co">##  [7] "proportion_of_reads_reported_by_sra_downloaded"</span></span>
<span><span class="co">##  [8] "paired_end"                                    </span></span>
<span><span class="co">##  [9] "sra_misreported_paired_end"                    </span></span>
<span><span class="co">## [10] "mapped_read_count"                             </span></span>
<span><span class="co">## [11] "auc"                                           </span></span>
<span><span class="co">## [12] "sharq_beta_tissue"                             </span></span>
<span><span class="co">## [13] "sharq_beta_cell_type"                          </span></span>
<span><span class="co">## [14] "biosample_submission_date"                     </span></span>
<span><span class="co">## [15] "biosample_publication_date"                    </span></span>
<span><span class="co">## [16] "biosample_update_date"                         </span></span>
<span><span class="co">## [17] "avg_read_length"                               </span></span>
<span><span class="co">## [18] "geo_accession"                                 </span></span>
<span><span class="co">## [19] "bigwig_file"                                   </span></span>
<span><span class="co">## [20] "title"                                         </span></span>
<span><span class="co">## [21] "characteristics"</span></span></code></pre>
<div class="section level5">
<h5 id="technical-variables">Technical variables<a class="anchor" aria-label="anchor" href="#technical-variables"></a>
</h5>
<p>Several of these technical variables include the number of reads as
reported by SRA, the actual number of reads Rail-RNA was able to
download (which might be lower in some cases), the number of reads
mapped by Rail-RNA, whether the sample is paired-end or not, the
coverage AUC and the average read length (times 2 for paired-end
samples). Note that the sample with SRA Run ID SRR2071341 has about
240.8 million reads as reported by SRA, while it has 120.4 million spots
reported in <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341" class="external-link uri">https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341</a>;
that is because it is a paired-end sample (2 reads per spot). These
details are important for those interested in writing alternative
scaling functions to <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts()</a></code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Input reads: number reported by SRA might be larger than number</span></span>
<span><span class="co">## of reads Rail-RNA downloaded</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span></span>
<span>    ,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"read_count_as_reported_by_sra"</span>, <span class="st">"reads_downloaded"</span><span class="op">)</span></span>
<span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 72 rows and 2 columns</span></span>
<span><span class="co">##            read_count_as_reported_by_sra reads_downloaded</span></span>
<span><span class="co">##                                &lt;integer&gt;        &lt;integer&gt;</span></span>
<span><span class="co">## SRR2071341                     240797206        240797206</span></span>
<span><span class="co">## SRR2071345                      82266652         82266652</span></span>
<span><span class="co">## SRR2071346                     132911310        132911310</span></span>
<span><span class="co">## SRR2071347                      74051302         74051302</span></span>
<span><span class="co">## SRR2071348                     250259914        250259914</span></span>
<span><span class="co">## ...                                  ...              ...</span></span>
<span><span class="co">## SRR1554541                     186250218        162403466</span></span>
<span><span class="co">## SRR1554554                     140038024        121793680</span></span>
<span><span class="co">## SRR1554535                     106244496         91185969</span></span>
<span><span class="co">## SRR1554558                     200687480        170754145</span></span>
<span><span class="co">## SRR1554553                      90579486         51803404</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">proportion_of_reads_reported_by_sra_downloaded</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##  0.5719  0.9165  0.9788  0.9532  1.0000  1.0000</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## AUC information used by scale_counts() by default</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">auc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 22950214241  7553726235 12018044330  7041243857 24062460144 45169026301</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Alternatively, scale_scounts() can use the number of mapped reads</span></span>
<span><span class="co">## and other information</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"mapped_read_count"</span>, <span class="st">"paired_end"</span>,</span>
<span>    <span class="st">"avg_read_length"</span></span>
<span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 72 rows and 3 columns</span></span>
<span><span class="co">##            mapped_read_count paired_end avg_read_length</span></span>
<span><span class="co">##                    &lt;integer&gt;  &lt;logical&gt;       &lt;integer&gt;</span></span>
<span><span class="co">## SRR2071341         232970536       TRUE             200</span></span>
<span><span class="co">## SRR2071345          78431778       TRUE             200</span></span>
<span><span class="co">## SRR2071346         124493632       TRUE             200</span></span>
<span><span class="co">## SRR2071347          71742875       TRUE             200</span></span>
<span><span class="co">## SRR2071348         242992735       TRUE             200</span></span>
<span><span class="co">## ...                      ...        ...             ...</span></span>
<span><span class="co">## SRR1554541         162329325       TRUE             174</span></span>
<span><span class="co">## SRR1554554         121738246       TRUE             173</span></span>
<span><span class="co">## SRR1554535          91120421       TRUE             171</span></span>
<span><span class="co">## SRR1554558         170648458       TRUE             170</span></span>
<span><span class="co">## SRR1554553          51684462       TRUE             114</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="biological-information">Biological information<a class="anchor" aria-label="anchor" href="#biological-information"></a>
</h5>
<p>Other metadata variables included provide more biological
information, such as the <a href="http://www.cs.cmu.edu/~ckingsf/sharq/" class="external-link">SHARQ beta</a> tissue and
cell type predictions, which are based on processing the abstract of
papers. This information is available for some of the SRA projects.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## SHARQ tissue predictions: not present for all studies</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">sharq_beta_tissue</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] NA NA NA NA NA NA</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">sharq_beta_tissue</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "blood" "blood" "blood" "blood" "blood" "blood"</span></span></code></pre>
<p>For some data sets we were able to find the GEO accession IDs, which
we then used to create the <code>title</code> and
<code>characteristics</code> variables. If present, the
<code>characteristics</code> information can be used to create
additional metadata variables by parsing the <code>CharacterList</code>
in which it is stored. Since the input is free text, sometimes more than
one type of wording is used to describe the same information, meaning
that we might have to process that information in order to build a more
convenient variable, such as a factor vector.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## GEO information was absent for the SRP045638 data set</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"geo_accession"</span>, <span class="st">"title"</span>, <span class="st">"characteristics"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 72 rows and 3 columns</span></span>
<span><span class="co">##            geo_accession       title characteristics</span></span>
<span><span class="co">##              &lt;character&gt; &lt;character&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">## SRR2071341            NA          NA              NA</span></span>
<span><span class="co">## SRR2071345            NA          NA              NA</span></span>
<span><span class="co">## SRR2071346            NA          NA              NA</span></span>
<span><span class="co">## SRR2071347            NA          NA              NA</span></span>
<span><span class="co">## SRR2071348            NA          NA              NA</span></span>
<span><span class="co">## ...                  ...         ...             ...</span></span>
<span><span class="co">## SRR1554541            NA          NA              NA</span></span>
<span><span class="co">## SRR1554554            NA          NA              NA</span></span>
<span><span class="co">## SRR1554535            NA          NA              NA</span></span>
<span><span class="co">## SRR1554558            NA          NA              NA</span></span>
<span><span class="co">## SRR1554553            NA          NA              NA</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## GEO information for the SRP009615 data set</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">geo_accession</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "GSM836270" "GSM836271" "GSM836272" "GSM836273" "GSM847561" "GSM847562"</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">title</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep1." </span></span>
<span><span class="co">## [2] "K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep1."</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">characteristics</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 2</span></span>
<span><span class="co">## [[1]] cells: K562 shRNA expression: no treatment: Puromycin</span></span>
<span><span class="co">## [[2]] cells: K562 shRNA expression: yes, targeting SRF treatment: Puromycin, doxycycline</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Similar but not exactly the same wording used for two different samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">characteristics</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cells: K562"          "shRNA expression: no" "treatment: Puromycin"</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">characteristics</span><span class="op">[[</span><span class="fl">11</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cell line: K562"                      </span></span>
<span><span class="co">## [2] "shRNA expression: no shRNA expression"</span></span>
<span><span class="co">## [3] "treatment: Puromycin"</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the target information</span></span>
<span><span class="va">target</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">characteristics</span>, <span class="st">"["</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">target</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "shRNA expression: no"                             </span></span>
<span><span class="co">##  [2] "shRNA expression: yes, targeting SRF"             </span></span>
<span><span class="co">##  [3] "shRNA expression: no"                             </span></span>
<span><span class="co">##  [4] "shRNA expression: yes targeting SRF"              </span></span>
<span><span class="co">##  [5] "shRNA expression: no shRNA expression"            </span></span>
<span><span class="co">##  [6] "shRNA expression: expressing shRNA targeting EGR1"</span></span>
<span><span class="co">##  [7] "shRNA expression: no shRNA expression"            </span></span>
<span><span class="co">##  [8] "shRNA expression: expressing shRNA targeting EGR1"</span></span>
<span><span class="co">##  [9] "shRNA expression: no shRNA expression"            </span></span>
<span><span class="co">## [10] "shRNA expression: expressing shRNA targeting ATF3"</span></span>
<span><span class="co">## [11] "shRNA expression: no shRNA expression"            </span></span>
<span><span class="co">## [12] "shRNA expression: expressing shRNA targeting ATF3"</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Build a useful factor vector, set the reference level and append the result</span></span>
<span><span class="co">## to the colData() slot</span></span>
<span><span class="va">target_factor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">target</span>, <span class="st">"targeting "</span><span class="op">)</span>, <span class="st">"["</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">target_factor</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">target_factor</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"none"</span></span>
<span><span class="va">target_factor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">target_factor</span><span class="op">)</span></span>
<span><span class="va">target_factor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">target_factor</span>, <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">target_factor</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] none SRF  none SRF  none EGR1 none EGR1 none ATF3 none ATF3</span></span>
<span><span class="co">## Levels: none ATF3 EGR1 SRF</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_SRP009615</span><span class="op">)</span><span class="op">$</span><span class="va">target_factor</span> <span class="op">&lt;-</span> <span class="va">target_factor</span></span></code></pre></div>
<p>As shown in Figure @ref(fig:Figure2), we can expand the biological
metadata information by adding predictions based on RNA-seq data <span class="citation">(14)</span>. The predictions include information about
sex, sample source (cell line vs tissue), tissue and the sequencing
strategy used. To add the predictions, simply use the function
<code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html" class="external-link">add_predictions()</a></code> to expand the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData()</a></code>
slot.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Before adding predictions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72 21</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add the predictions</span></span>
<span><span class="va">rse_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/add_predictions.html" class="external-link">add_predictions</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 19:38:03.426619 downloading the predictions to C:\Users\fellg\AppData\Local\Temp\Rtmp2vyc2o/PredictedPhenotypes_v0.0.06.rda</span></span></code></pre>
<pre><code><span><span class="co">## Loading objects:</span></span>
<span><span class="co">##   PredictedPhenotypes</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## After adding the predictions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72 33</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the variables</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span>, <span class="fl">22</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 72 rows and 12 columns</span></span>
<span><span class="co">##            reported_sex predicted_sex accuracy_sex reported_samplesource</span></span>
<span><span class="co">##                &lt;factor&gt;      &lt;factor&gt;    &lt;numeric&gt;              &lt;factor&gt;</span></span>
<span><span class="co">## SRR2071341       female        female     0.862637                    NA</span></span>
<span><span class="co">## SRR2071345       male          male       0.862637                    NA</span></span>
<span><span class="co">## SRR2071346       male          male       0.862637                    NA</span></span>
<span><span class="co">## SRR2071347       female        female     0.862637                    NA</span></span>
<span><span class="co">## SRR2071348       female        female     0.862637                    NA</span></span>
<span><span class="co">## ...                 ...           ...          ...                   ...</span></span>
<span><span class="co">## SRR1554541       male          male       0.862637                    NA</span></span>
<span><span class="co">## SRR1554554       female        female     0.862637                    NA</span></span>
<span><span class="co">## SRR1554535       male          male       0.862637                    NA</span></span>
<span><span class="co">## SRR1554558       female        female     0.862637                    NA</span></span>
<span><span class="co">## SRR1554553       male          male       0.862637                    NA</span></span>
<span><span class="co">##            predicted_samplesource accuracy_samplesource reported_tissue</span></span>
<span><span class="co">##                          &lt;factor&gt;             &lt;numeric&gt;        &lt;factor&gt;</span></span>
<span><span class="co">## SRR2071341                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR2071345                 tissue               0.89235              NA</span></span>
<span><span class="co">## SRR2071346                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR2071347                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR2071348                 tissue                    NA              NA</span></span>
<span><span class="co">## ...                           ...                   ...             ...</span></span>
<span><span class="co">## SRR1554541                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR1554554                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR1554535                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR1554558                 tissue                    NA              NA</span></span>
<span><span class="co">## SRR1554553                 tissue               0.89235              NA</span></span>
<span><span class="co">##            predicted_tissue accuracy_tissue reported_sequencingstrategy</span></span>
<span><span class="co">##                    &lt;factor&gt;       &lt;numeric&gt;                    &lt;factor&gt;</span></span>
<span><span class="co">## SRR2071341            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR2071345            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR2071346            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR2071347            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR2071348            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## ...                     ...             ...                         ...</span></span>
<span><span class="co">## SRR1554541            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR1554554            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR1554535            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR1554558            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">## SRR1554553            Brain        0.518825                      PAIRED</span></span>
<span><span class="co">##            predicted_sequencingstrategy accuracy_sequencingstrategy</span></span>
<span><span class="co">##                                &lt;factor&gt;                   &lt;numeric&gt;</span></span>
<span><span class="co">## SRR2071341                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR2071345                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR2071346                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR2071347                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR2071348                       PAIRED                    0.908575</span></span>
<span><span class="co">## ...                                 ...                         ...</span></span>
<span><span class="co">## SRR1554541                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR1554554                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR1554535                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR1554558                       PAIRED                    0.908575</span></span>
<span><span class="co">## SRR1554553                       PAIRED                    0.908575</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="adding-more-information">Adding more information<a class="anchor" aria-label="anchor" href="#adding-more-information"></a>
</h5>
<p>Ultimately, more sample metadata information could be available
elsewhere, which can be useful for analyses. This information might be
provided in the paper describing the data, the SRA Run Selector or other
sources. As shown in Figure @ref(fig:Figure2), it is possible to append
information to the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData()</a></code> slot as long as there is a
unique sample identifier such as the SRA Run ID.</p>
<p>For our example use case, project SRP045638 has a few extra
biologically relevant variables via the SRA Run selector <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638" class="external-link uri">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638</a>.
We can download that information into text file named
<code>SraRunTable.txt</code> by default, then load it into R, sort it
appropriately and then append it to the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData()</a></code> slot.
Below we do so for the SRP045638 project.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Save the information from</span></span>
<span><span class="co">## https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638</span></span>
<span><span class="co">## to a table. We saved the file as SRP045638/SraRunTable.txt.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"SraRunTable.txt"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Read the table</span></span>
<span><span class="va">sra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"SraRunTable.txt"</span><span class="op">)</span>,</span>
<span>    header <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore it</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sra</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          Run     Age alignment_software..exp. Assay.Type AvgSpotLen       Bases</span></span>
<span><span class="co">## 1 SRR1554533 67.7800                  TopHat2    RNA-Seq        179  6766378000</span></span>
<span><span class="co">## 2 SRR1554534 40.4200                  TopHat2    RNA-Seq        179  6357169900</span></span>
<span><span class="co">## 3 SRR1554535 41.5800                  TopHat2    RNA-Seq        171  9118596900</span></span>
<span><span class="co">## 4 SRR1554536 44.1700                  TopHat2    RNA-Seq        184  4697126700</span></span>
<span><span class="co">## 5 SRR1554537 -0.3836                  TopHat2    RNA-Seq        182 12199232600</span></span>
<span><span class="co">## 6 SRR1554538 -0.4027                  TopHat2    RNA-Seq        185 14733333600</span></span>
<span><span class="co">##   Biomaterial_provider  BioProject    BioSample BioSampleModel      Bytes</span></span>
<span><span class="co">## 1                 LIBD PRJNA245228 SAMN02731372          Human 3744853436</span></span>
<span><span class="co">## 2                 LIBD PRJNA245228 SAMN02731373          Human 3019096844</span></span>
<span><span class="co">## 3                 LIBD PRJNA245228 SAMN02999518          Human 5204886934</span></span>
<span><span class="co">## 4                 LIBD PRJNA245228 SAMN02999519          Human 2771744419</span></span>
<span><span class="co">## 5                 LIBD PRJNA245228 SAMN02999520          Human 6485501388</span></span>
<span><span class="co">## 6                 LIBD PRJNA245228 SAMN02999521          Human 7505295858</span></span>
<span><span class="co">##                              Center.Name Consent DATASTORE.filetype</span></span>
<span><span class="co">## 1 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">## 2 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">## 3 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">## 4 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">## 5 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">## 6 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra</span></span>
<span><span class="co">##   DATASTORE.provider               DATASTORE.region disease Experiment fraction</span></span>
<span><span class="co">## 1         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 Control  SRX683791    total</span></span>
<span><span class="co">## 2         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 Control  SRX683792    total</span></span>
<span><span class="co">## 3         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683793    total</span></span>
<span><span class="co">## 4         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683794    total</span></span>
<span><span class="co">## 5         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683795    total</span></span>
<span><span class="co">## 6         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683796    total</span></span>
<span><span class="co">##            Instrument isolate                   Library.Name LibraryLayout</span></span>
<span><span class="co">## 1 Illumina HiSeq 2000   DLPFC R2835_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">## 2 Illumina HiSeq 2000   DLPFC R2857_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">## 3 Illumina HiSeq 2000   R2869 R2869_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">## 4 Illumina HiSeq 2000   R3098 R3098_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">## 5 Illumina HiSeq 2000   R3452 R3452_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">## 6 Illumina HiSeq 2000   R3462 R3462_DLPFC_polyA_RNAseq_total        PAIRED</span></span>
<span><span class="co">##   LibrarySelection  LibrarySource     Organism Platform race</span></span>
<span><span class="co">## 1             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">## 2             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">## 3             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">## 4             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">## 5             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">## 6             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA</span></span>
<span><span class="co">##            ReleaseDate rin Sample.Name    sex SRA.Study tissue     AssemblyName</span></span>
<span><span class="co">## 1 2014-11-13T00:00:00Z 8.3 R2835_DLPFC female SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">## 2 2014-11-13T00:00:00Z 8.4 R2857_DLPFC   male SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">## 3 2014-11-13T00:00:00Z 8.7 R2869_DLPFC   male SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">## 4 2014-11-13T00:00:00Z 5.3 R3098_DLPFC female SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">## 5 2014-11-13T00:00:00Z 9.6 R3452_DLPFC female SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">## 6 2014-11-13T00:00:00Z 6.4 R3462_DLPFC female SRP045638  DLPFC GCA_000001405.13</span></span>
<span><span class="co">##   dev_stage</span></span>
<span><span class="co">## 1     Fetal</span></span>
<span><span class="co">## 2          </span></span>
<span><span class="co">## 3          </span></span>
<span><span class="co">## 4          </span></span>
<span><span class="co">## 5          </span></span>
<span><span class="co">## 6</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set all column names in lower case</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sra</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sra</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Choose some variables we want to add</span></span>
<span><span class="va">sra_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"sex"</span>, <span class="st">"race"</span>, <span class="st">"rin"</span>, <span class="st">"age"</span>, <span class="st">"isolate"</span>, <span class="st">"disease"</span>,</span>
<span>    <span class="st">"tissue"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">sra_vars</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sra</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Re-organize the SRA table based on the SRA Run IDs we have</span></span>
<span><span class="va">sra</span> <span class="op">&lt;-</span> <span class="va">sra</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">run</span>, <span class="va">sra</span><span class="op">$</span><span class="va">run</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Double check the order</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">run</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sra</span><span class="op">$</span><span class="va">run</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Append the variables of interest</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>, <span class="va">sra</span><span class="op">[</span>, <span class="va">sra_vars</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Final dimensions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72 40</span></span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore result</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span>, <span class="fl">34</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 72 rows and 7 columns</span></span>
<span><span class="co">##                    sex        race       rin       age     isolate     disease</span></span>
<span><span class="co">##            &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## SRR2071341      female          AA       8.3   67.7800       DLPFC     Control</span></span>
<span><span class="co">## SRR2071345        male          AA       8.4   40.4200       DLPFC     Control</span></span>
<span><span class="co">## SRR2071346        male          AA       8.7   41.5800       R2869     control</span></span>
<span><span class="co">## SRR2071347      female          AA       5.3   44.1700       R3098     control</span></span>
<span><span class="co">## SRR2071348      female          AA       9.6   -0.3836       R3452     control</span></span>
<span><span class="co">## ...                ...         ...       ...       ...         ...         ...</span></span>
<span><span class="co">## SRR1554541        male          AA       5.7   -0.3836       R3485     control</span></span>
<span><span class="co">## SRR1554554      female          AA       8.1    0.3041       R3669     control</span></span>
<span><span class="co">## SRR1554535        male          AA       8.7   41.5800       R2869     control</span></span>
<span><span class="co">## SRR1554558      female        CAUC       9.1   16.7000       R4028     control</span></span>
<span><span class="co">## SRR1554553        male        CAUC       8.4    0.3918       R3652     control</span></span>
<span><span class="co">##                 tissue</span></span>
<span><span class="co">##            &lt;character&gt;</span></span>
<span><span class="co">## SRR2071341       DLPFC</span></span>
<span><span class="co">## SRR2071345       DLPFC</span></span>
<span><span class="co">## SRR2071346       DLPFC</span></span>
<span><span class="co">## SRR2071347       DLPFC</span></span>
<span><span class="co">## SRR2071348       DLPFC</span></span>
<span><span class="co">## ...                ...</span></span>
<span><span class="co">## SRR1554541       DLPFC</span></span>
<span><span class="co">## SRR1554554       DLPFC</span></span>
<span><span class="co">## SRR1554535       DLPFC</span></span>
<span><span class="co">## SRR1554558       DLPFC</span></span>
<span><span class="co">## SRR1554553       DLPFC</span></span></code></pre>
<p>Since we have the predicted sex as well as the reported sex via the
SRA Run Selector, we can check whether they match.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span></span>
<span>    <span class="st">"Predicted"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">predicted_sex</span>,</span>
<span>    <span class="st">"Observed"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">sex</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##             Observed</span></span>
<span><span class="co">## Predicted    female male</span></span>
<span><span class="co">##   female         24    4</span></span>
<span><span class="co">##   male            0   44</span></span>
<span><span class="co">##   Unassigned      0    0</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="de-setup">DE setup<a class="anchor" aria-label="anchor" href="#de-setup"></a>
</h4>
<p>Now that we have all the metadata available we can perform a DE
analysis. The original study for project SRP045638 <span class="citation">(13)</span> looked at differences between 6 age groups:
prenatal, infant, child, teen, adult and late life. The following code
creates these six age groups.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the original 6 age groups</span></span>
<span><span class="va">age_bins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html" class="external-link">cut</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">age</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">50</span>, <span class="cn">Inf</span><span class="op">)</span>,</span>
<span>    include.lowest <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">age_bins</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"prenatal"</span>, <span class="st">"infant"</span>, <span class="st">"child"</span>, <span class="st">"teen"</span>, <span class="st">"adult"</span>,</span>
<span>    <span class="st">"late life"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">age_group</span> <span class="op">&lt;-</span> <span class="va">age_bins</span></span></code></pre></div>
<p>Most of the DE signal from the original study was between the
prenatal and postnatal samples. To simplify the analysis, we will focus
on this comparison.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create prenatal factor</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">age_group</span> <span class="op">==</span> <span class="st">"prenatal"</span>, <span class="st">"prenatal"</span>,</span>
<span>        <span class="st">"postnatal"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prenatal"</span>, <span class="st">"postnatal"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>As we saw earlier in Figure @ref(fig:Figure9), it is important to
scale the coverage counts to read counts. To highlight the fact that we
scaled the counts, we will use a new object name and delete the previous
one. However, in practice we would simply overwrite the <code>rse</code>
object with the output of <code>scale_counts(rse)</code>.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Scale counts</span></span>
<span><span class="va">rse_gene_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To highlight that we scaled the counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span></span></code></pre></div>
<p>Having scaled the counts, we then filter out genes that are lowly
expressed and extract the count matrix.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract counts and filter out lowly expressed geens</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.5</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="de-analysis">DE analysis<a class="anchor" aria-label="anchor" href="#de-analysis"></a>
</h4>
<p>Now that we have scaled the counts, there are multiple DE packages we
could use, as described elsewhere <span class="citation">(2,3)</span>.
Since we have 12 samples per group, which is a moderate number, we will
use <code>limma-voom</code> <span class="citation">(16)</span> due to
its speed. The model we use tests for DE between prenatal and postnatal
samples adjusting for sex and RIN, which is a measure of quality of the
input sample. We check the data with multi-dimensional scaling plots
(Figures @ref(fig:limmade1) and @ref(fig:limmade2)) as well as the
mean-variance plot (Figure @ref(fig:limmade3)). In a real use case we
might have to explore the results with different models and perform
sensitivity analyses.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build DGEList object</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span><span class="op">[</span><span class="va">filter</span>, <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Calculate normalization factors</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="va">dge</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="limmade1-1.png" alt="Multi-dimensional scaling plot of the gene-level data by age group." width="70%"><p class="caption">
Multi-dimensional scaling plot of the gene-level data by age group.
</p>
</div>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="va">dge</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">sex</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="limmade2-1.png" alt="Multi-dimensional scaling plot of the gene-level data by sex." width="70%"><p class="caption">
Multi-dimensional scaling plot of the gene-level data by sex.
</p>
</div>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">rin</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span>,</span>
<span>    <span class="va">summary</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $prenatal</span></span>
<span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##   5.700   6.400   8.150   7.767   8.600   9.600 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $postnatal</span></span>
<span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##   5.300   8.100   8.300   8.197   8.700   9.100</span></span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Specify our design matrix</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">rin</span> <span class="op">+</span> <span class="va">prenatal</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run voom</span></span>
<span><span class="va">v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="limmade3-1.png" alt="voom mean-variance plot of the gene-level data." width="70%"><p class="caption">
voom mean-variance plot of the gene-level data.
</p>
</div>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run remaining parts of the DE analysis</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">v</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span></code></pre></div>
<p>Having run the DE analysis, we can explore some of the top results
either with an MA plot (Figure @ref(fig:limmaplots1)) and a volcano plot
Figure (@ref(fig:limmaplots2)). Both reveal very strong and widespread
DE signal.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visually explore DE results</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotma.html" class="external-link">plotMA</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="limmaplots1-1.png" alt="MA plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="70%"><p class="caption">
MA plot of the gene-level data. Testing for prenatal and postnatal DE
adjusting for sex and RIN.
</p>
</div>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html" class="external-link">volcanoplot</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="limmaplots2-1.png" alt="Volcano plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="70%"><p class="caption">
Volcano plot of the gene-level data. Testing for prenatal and postnatal
DE adjusting for sex and RIN.
</p>
</div>
</div>
<div class="section level4">
<h4 id="de-report">DE report<a class="anchor" aria-label="anchor" href="#de-report"></a>
</h4>
<p>Now that we have the DE results, we can use some of the tools with
the biocView <a href="http://bioconductor.org/packages/release/BiocViews.html#___ReportWriting" class="external-link">ReportWriting</a>
to create a report. One of them is <code>regionReport</code> <span class="citation">(17)</span>, which can create reports from
<code>DESeq2</code> <span class="citation">(18)</span> and
<code>edgeR</code> <span class="citation">(19)</span> results. It can
also handle <code>limma-voom</code> <span class="citation">(16)</span>
results by making them look like <code>DESeq2</code> results. To do so,
we need to extract the relevant information from the
<code>limma-voom</code> objects using <code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable()</a></code> and build
DESeqDataSet and DESeqResults objects as shown below. A similar
conversion is needed to use <code>ideal</code> <span class="citation">(20)</span>, which is another package in the
<em>ReportWriting</em> biocView category.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract data from limma-voom results</span></span>
<span><span class="va">top</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>,</span>
<span>    number <span class="op">=</span> <span class="cn">Inf</span>, sort.by <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    coef <span class="op">=</span> <span class="st">"prenatalpostnatal"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a DESeqDataSet with the count data and model we used</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">[</span><span class="va">filter</span>, <span class="op">]</span>, <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">rin</span> <span class="op">+</span> <span class="va">prenatal</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## converting counts to integer mode</span></span></code></pre>
<pre><code><span><span class="co">## Warning in DESeqDataSet(rse_gene_scaled[filter, ], ~sex + rin + prenatal): some</span></span>
<span><span class="co">## variables in design formula are characters, converting to factors</span></span></code></pre>
<pre><code><span><span class="co">##   the design formula contains one or more numeric variables that have mean or</span></span>
<span><span class="co">##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).</span></span>
<span><span class="co">##   Including numeric variables with large mean can induce collinearity with the intercept.</span></span>
<span><span class="co">##   Users should center and scale numeric variables in the design to improve GLM convergence.</span></span></code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add gene names keeping only the Ensembl part of the Gencode IDs</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a DESeqResults object with the relevant information</span></span>
<span><span class="co">## Note that we are transforming the baseMean so it will look ok</span></span>
<span><span class="co">## with DESeq2's plotting functions.</span></span>
<span><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html" class="external-link">DESeqResults</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span></span>
<span>    pvalue <span class="op">=</span> <span class="va">top</span><span class="op">[</span>, <span class="st">"P.Value"</span><span class="op">]</span>,</span>
<span>    log2FoldChange <span class="op">=</span> <span class="va">top</span><span class="op">[</span>, <span class="st">"logFC"</span><span class="op">]</span>,</span>
<span>    baseMean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="va">top</span><span class="op">[</span>, <span class="st">"AveExpr"</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    padj <span class="op">=</span> <span class="va">top</span><span class="op">[</span>, <span class="st">"adj.P.Val"</span><span class="op">]</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Specify FDR cutoff to use</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span><span class="op">[[</span><span class="st">"alpha"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0.001</span></span>
<span></span>
<span><span class="co">## Add gene symbols so they will be displayed in the report</span></span>
<span><span class="va">limma_res</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span><span class="op">[</span><span class="va">filter</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Some extra information used by the report function</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">limma_res</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span></span>
<span>    type <span class="op">=</span> <span class="st">"results"</span>,</span>
<span>    description <span class="op">=</span> <span class="st">"manual incomplete conversion from limma-voom to DESeq2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Having converted our <code>limma-voom</code> results to
<code>DESeq2</code> results, we can now create the report, which should
open automatically in a browser.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/regionReport" class="external-link">"regionReport"</a></span><span class="op">)</span></span>
<span><span class="co">## This takes about 20 minutes to run</span></span>
<span><span class="va">report</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html" class="external-link">DESeq2Report</a></span><span class="op">(</span><span class="va">dds</span>,</span>
<span>    project <span class="op">=</span> <span class="st">"SRP045638 gene results with limma-voom"</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"gene_report"</span>, outdir <span class="op">=</span> <span class="st">"SRP045638"</span>,</span>
<span>    intgroup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prenatal"</span>, <span class="st">"sex"</span><span class="op">)</span>, res <span class="op">=</span> <span class="va">limma_res</span>,</span>
<span>    software <span class="op">=</span> <span class="st">"limma"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="create_report-1.png" width="700"></p>
<p>If the report doesn’t open automatically, we can open it with
<code><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL()</a></code>. A pre-computed version is available as
Supplementary File 1.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"gene_report.html"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="go-enrichment">GO enrichment<a class="anchor" aria-label="anchor" href="#go-enrichment"></a>
</h4>
<p>Using <code>clusterProfiler</code> <span class="citation">(21)</span>
we can then perform several enrichment analyses using the Ensembl gene
IDs. Here we show how to perform an enrichment analysis using the
biological process ontology (Figure @ref(fig:goanalysis)).</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">"clusterProfiler"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Remember that limma_res had ENSEMBL IDs for the genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419" "ENSG00000000457" "ENSG00000000460" "ENSG00000000938"</span></span></code></pre>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform enrichment analysis for Biological Process (BP)</span></span>
<span><span class="co">## Note that the argument is keytype instead of keyType in Bioconductor 3.5</span></span>
<span><span class="va">enrich_go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span></span>
<span>    gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span><span class="op">[</span><span class="va">limma_res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">]</span>,</span>
<span>    OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, keyType <span class="op">=</span> <span class="st">"ENSEMBL"</span>, ont <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>    pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>, pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>, qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    universe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Visualize enrichment results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html" class="external-link">dotplot</a></span><span class="op">(</span><span class="va">enrich_go</span>, font.size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="goanalysis-1.png" alt="Biological processes enriched in the DE genes." width="70%"><p class="caption">
Biological processes enriched in the DE genes.
</p>
</div>
<p>Several other analyses can be performed with the resulting list of
differentially expressed genes as described previously <span class="citation">(2,3)</span>, although that is beyond the scope of this
workflow.</p>
</div>
</div>
<div class="section level3">
<h3 id="other-features">Other features<a class="anchor" aria-label="anchor" href="#other-features"></a>
</h3>
<p>As described in Figure @ref(fig:Figure1), recount2 provides data for
expression features beyond genes. In this section we perform a DE
analysis using exon data as well as the base-pair resolution
information.</p>
<div class="section level4">
<h4 id="exon-and-exon-exon-junctions">Exon and exon-exon junctions<a class="anchor" aria-label="anchor" href="#exon-and-exon-exon-junctions"></a>
</h4>
<p>The exon and exon-exon junction coverage count matrices are similar
to the gene-level one and can also be downloaded with
<code><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study()</a></code>. However, these coverage count matrices
are much larger than the gene one. Aggressive filtering of lowly
expressed exons or exon-exon junctions can reduce the matrix dimensions
if this impacts the performance of the DE software used.</p>
<p>Below we repeat the gene-level analysis for the disjoint exon data.
We first download the exon data, add the expanded metadata we
constructed for the gene analysis, explore the data (Figure1
@ref(fig:exondeanalysis1)), and then perform the DE analysis using
<code>limma-voom</code>.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the data if it is not there</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"rse_exon.Rdata"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, type <span class="op">=</span> <span class="st">"rse-exon"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 19:42:35.79495 downloading file rse_exon.Rdata to SRP045638</span></span></code></pre>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"rse_exon.Rdata"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Scale and add the metadata (it is in the same order)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span><span class="op">$</span><span class="va">run</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">run</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span></span>
<span><span class="va">rse_exon_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span></span>
<span><span class="co">## To highlight that we scaled the counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">rse_exon</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Filter lowly expressed exons</span></span>
<span><span class="va">filter_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_exon_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">filter_exon</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">filter_exon</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## filter_exon</span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 32.76 67.24</span></span></code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Build DGEList object</span></span>
<span><span class="va">dge_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span></span>
<span>    counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_exon_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="va">filter_exon</span>, <span class="op">]</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Calculate normalization factors</span></span>
<span><span class="va">dge_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge_exon</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run voom</span></span>
<span><span class="va">v_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span><span class="va">dge_exon</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="exondeanalysis1-1.png" alt="voom mean-variance plot of the exon-level data." width="70%"><p class="caption">
voom mean-variance plot of the exon-level data.
</p>
</div>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run remaining parts of the DE analysis</span></span>
<span><span class="va">fit_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">v_exon</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit_exon</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Visualize inspect results</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html" class="external-link">volcanoplot</a></span><span class="op">(</span><span class="va">fit_exon</span>, coef <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="exondeanalysis2-1.png" alt="Volcano plot of the exon-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="70%"><p class="caption">
Volcano plot of the exon-level data. Testing for prenatal and postnatal
DE adjusting for sex and RIN.
</p>
</div>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get p-values and other statistics</span></span>
<span><span class="va">top_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit_exon</span>,</span>
<span>    number <span class="op">=</span> <span class="cn">Inf</span>, sort.by <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    coef <span class="op">=</span> <span class="st">"prenatalpostnatal"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">top_exon</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  FALSE   TRUE </span></span>
<span><span class="co">## 192061 213482</span></span></code></pre>
<p>Just like at the gene-level, we see many exons differentially
expressed between prenatal and postnatal samples (Figure
@ref(fig:exondeanalysis2)). As a first step to integrate the results
from the two features, we can compare the list of genes that are
differentially expressed versus the genes that have at least one exon
differentially expressed.</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the gene IDs for genes that are DE at the gene-level or that have at</span></span>
<span><span class="co">## least one exon with DE signal.</span></span>
<span><span class="va">genes_w_de_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rse_exon_scaled</span><span class="op">)</span><span class="op">[</span><span class="va">top_exon</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span><span class="va">genes_de</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">filter</span><span class="op">)</span><span class="op">[</span><span class="va">top</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">]</span></span>
<span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Make a venn diagram</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/talgalili/gplots" class="external-link">"gplots"</a></span><span class="op">)</span></span>
<span><span class="va">vinfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gplots/man/venn.html" class="external-link">venn</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"genes"</span> <span class="op">=</span> <span class="va">genes_de</span>, <span class="st">"exons"</span> <span class="op">=</span> <span class="va">genes_w_de_exon</span><span class="op">)</span>,</span>
<span>    names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genes"</span>, <span class="st">"exons"</span><span class="op">)</span>, show.plot <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">vinfo</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="st">"Genes/exons with DE signal"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="geneexon-1.png" alt="Venn diagram of the overlap between DE genes and genes with at least one exon DE." width="70%"><p class="caption">
Venn diagram of the overlap between DE genes and genes with at least one
exon DE.
</p>
</div>
<pre><code><span><span class="co">## integer(0)</span></span></code></pre>
<p>Not all differentially expressed genes have differentially expressed
exons. Moreover, genes with at least one differentially expressed exon
are not necessarily differentially expressed, as shown in Figure
@ref(fig:geneexon). This is in line with what was described in Figure 2B
of Soneson et al., 2015 <span class="citation">(22)</span>.</p>
<p>This was just a quick example of how we can perform DE analyses at
the gene and exon feature levels. We envision that more involved
pipelines could be developed that leverage both feature levels, such as
in Jaffe et al., 2017 <span class="citation">(23)</span>. For instance,
we could focus on the differentially expressed genes with at least one
differentially expressed exon, and compare the direction of the DE
signal versus the gene-level signal as shown in figure
@ref(fig:geneexonmatch).</p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Keep only the DE exons that are from a gene that is also DE</span></span>
<span><span class="va">top_exon_de</span> <span class="op">&lt;-</span> <span class="va">top_exon</span><span class="op">[</span><span class="va">top_exon</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.001</span> <span class="op">&amp;</span></span>
<span>    <span class="va">top_exon</span><span class="op">$</span><span class="va">ID</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">vinfo</span>, <span class="st">"intersections"</span><span class="op">)</span><span class="op">[[</span><span class="st">"genes:exons"</span><span class="op">]</span><span class="op">]</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Find the fold change that is the most extreme among the DE exons of a gene</span></span>
<span><span class="va">exon_max_fc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">top_exon_de</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">top_exon_de</span><span class="op">$</span><span class="va">ID</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Keep only the DE genes that match the previous selection</span></span>
<span><span class="va">top_gene_de</span> <span class="op">&lt;-</span> <span class="va">top</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">exon_max_fc</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">top</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Make the plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">top_gene_de</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">exon_max_fc</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">20</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/adjustcolor.html" class="external-link">adjustcolor</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="fl">1</span> <span class="op">/</span> <span class="fl">5</span><span class="op">)</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"Most extreme exon log FC"</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"Gene log FC"</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"DE genes with at least one DE exon"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0</span>, b <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">"grey80"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="st">"grey80"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="geneexonmatch-1.png" alt="Log fold change (FC) for DE genes compared against the most extreme exon log FC among exons that are DE for the given gene." width="100%"><p class="caption">
Log fold change (FC) for DE genes compared against the most extreme exon
log FC among exons that are DE for the given gene.
</p>
</div>
<p>The fold change for most exons shown in Figure
@ref(fig:geneexonmatch) agrees with the gene-level fold change. However,
some of them have opposite directions and could be interesting to study
further.</p>
</div>
<div class="section level4">
<h4 id="base-pair-resolution">Base-pair resolution<a class="anchor" aria-label="anchor" href="#base-pair-resolution"></a>
</h4>
<p>recount2 provides bigWig coverage files (unscaled) for all samples,
as well as a mean bigWig coverage file per project where each sample was
scaled to 40 million 100 base-pair reads. The mean bigWig files are
exactly what is needed to start an <em>expressed regions</em> analysis
with <code>derfinder</code> <span class="citation">(8)</span>.
<code>recount</code> provides two related functions:
<code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html" class="external-link">expressed_regions()</a></code> which is used to define a set of
regions based on the mean bigWig file for a given project, and
<code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html" class="external-link">coverage_matrix()</a></code> which based on a set of regions builds a
count coverage matrix in a <em>RangedSummarizedExperiment</em> object
just like the ones that are provided for genes and exons. Both functions
ultimately use <code>import.bw()</code> from <code>rtracklayer</code>
<span class="citation">(24)</span> that was historically not supported
on Windows machines. While this presents a portability disadvantage, on
the other side it allows reading portions of bigWig files from the web
without having to fully download them. <code><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study()</a></code>
with <code>type = "mean"</code> or <code>type = "samples"</code> can be
used to download the bigWig files, which we recommend doing when working
with them extensively.</p>
<p>For illustrative purposes, we will use the data from chromosome 21
for the SRP045638 project. First, we obtain the expressed regions using
a relatively high mean cutoff of 5. We then filter the regions to keep
only the ones longer than 100 base-pairs to shorten the time needed for
running <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html" class="external-link">coverage_matrix()</a></code>.</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define expressed regions for study SRP045638, only for chromosome 21</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html" class="external-link">expressed_regions</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"chr21"</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">5L</span>,</span>
<span>    maxClusterGap <span class="op">=</span> <span class="fl">3000L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the resulting expressed regions</span></span>
<span><span class="va">regions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Keep only the ones that are at least 100 bp long</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="va">regions</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span></span></code></pre></div>
<p>Now that we have a set of regions to work with, we proceed to build a
<em>RangedSummarizedExperiment</em> object with the coverage counts, add
the expanded metadata we built for the gene-level, and scale the counts.
Note that <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html" class="external-link">coverage_matrix()</a></code> scales the base-pair coverage
counts by default, which we turn off in order to use use
<code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts()</a></code>.</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute coverage matrix for study SRP045638, only for chromosome 21</span></span>
<span><span class="co">## Takes about 4 minutes</span></span>
<span><span class="va">rse_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html" class="external-link">coverage_matrix</a></span><span class="op">(</span><span class="st">"SRP045638"</span>, <span class="st">"chr21"</span>, <span class="va">regions</span>,</span>
<span>    chunksize <span class="op">=</span> <span class="fl">2000</span>, verboseLoad <span class="op">=</span> <span class="cn">FALSE</span>, scale <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Use the expanded metadata we built for the gene model</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene_scaled</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Scale the coverage matrix</span></span>
<span><span class="va">rse_er_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts</a></span><span class="op">(</span><span class="va">rse_er</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To highlight that we scaled the counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">rse_er</span><span class="op">)</span></span></code></pre></div>
<p>Now that we have a scaled count matrix for the expressed regions, we
can proceed with the DE analysis just like we did at the gene and exon
feature levels (Figures @ref(fig:erdeanalysis1),
@ref(fig:erdeanalysis2), @ref(fig:erdeanalysis3), and
@ref(fig:erdeanalysis4)).</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Build DGEList object</span></span>
<span><span class="va">dge_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Calculate normalization factors</span></span>
<span><span class="va">dge_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge_er</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="va">dge_er</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="va">dge_er</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">sex</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run voom</span></span>
<span><span class="va">v_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span><span class="va">dge_er</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run remaining parts of the DE analysis</span></span>
<span><span class="va">fit_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">v_er</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit_er</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visually explore the results</span></span>
<span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html" class="external-link">volcanoplot</a></span><span class="op">(</span><span class="va">fit_er</span>, coef <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Number of DERs</span></span>
<span><span class="va">top_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit_er</span>,</span>
<span>    number <span class="op">=</span> <span class="cn">Inf</span>, sort.by <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    coef <span class="op">=</span> <span class="st">"prenatalpostnatal"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">top_er</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<p>Having identified the differentially expressed regions (DERs), we can
sort all regions by their adjusted p-value.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Sort regions by q-value</span></span>
<span><span class="va">regions_by_padj</span> <span class="op">&lt;-</span> <span class="va">regions</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">top_er</span><span class="op">$</span><span class="va">adj.P.Val</span>, decreasing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Look at the top 10</span></span>
<span><span class="va">regions_by_padj</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">regions_by_padj</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualize-regions">Visualize regions<a class="anchor" aria-label="anchor" href="#visualize-regions"></a>
</h4>
<p>Since the DERs do not necessarily match the annotation, it is
important to visualize them. The code for visualizing DERs can easily be
adapted to visualize other regions. Although, the width and number of
the regions will influence the computing resources needed to make the
plots.</p>
<p>Because the unscaled bigWig files are available in recount2, several
visualization packages can be used such as <code>epivizr</code> <span class="citation">(25)</span>, <code>wiggleplotr</code> <span class="citation">(26)</span> and <code>derfinderPlot</code> <span class="citation">(8)</span>. With all of them it is important to
remember to scale the data except when visualizing the mean bigWig file
for a given project.</p>
<p>First, we need to get the list of URLs for the bigWig files. We can
either manually construct them or search them inside the
<code>recount_url</code> table.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Construct the list of bigWig URLs</span></span>
<span><span class="co">## They have the following form:</span></span>
<span><span class="co">## http://duffel.rail.bio/recount/</span></span>
<span><span class="co">## project id</span></span>
<span><span class="co">## /bw/</span></span>
<span><span class="co">## sample run id</span></span>
<span><span class="co">## .bw</span></span>
<span><span class="va">bws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>    <span class="st">"http://duffel.rail.bio/recount/SRP045638/bw/"</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">bigwig_file</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Note that they are also present in the recount_url data.frame</span></span>
<span><span class="va">bws</span> <span class="op">&lt;-</span> <span class="va">recount_url</span><span class="op">$</span><span class="va">url</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">bigwig_file</span>,</span>
<span>    <span class="va">recount_url</span><span class="op">$</span><span class="va">file_name</span></span>
<span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Use the sample run IDs as the sample names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bws</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">run</span></span></code></pre></div>
<p>We visualize the DERs using <code>derfinderPlot</code>, similar to
what was done in the original publication <span class="citation">(13)</span>. However, we first add a little padding to
the regions: 100 base-pairs on each side.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add 100 bp padding on each side</span></span>
<span><span class="va">regions_resized</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">regions_by_padj</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">regions_by_padj</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fl">200</span>,</span>
<span>    fix <span class="op">=</span> <span class="st">"center"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Next, we obtain the base-pair coverage data for each DER and scale
the data to a library size of 40 million 100 base-pair reads, using the
coverage AUC information we have in the metadata.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the bp coverage data for the plots</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lcolladotor/derfinder" class="external-link">"derfinder"</a></span><span class="op">)</span></span>
<span><span class="va">regionCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html" class="external-link">getRegionCoverage</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions_resized</span>, files <span class="op">=</span> <span class="va">bws</span>,</span>
<span>    targetSize <span class="op">=</span> <span class="fl">40</span> <span class="op">*</span> <span class="fl">1e6</span> <span class="op">*</span> <span class="fl">100</span>,</span>
<span>    totalMapped <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">auc</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage()</a></code> requires several
pieces of annotation information for the plots that use a TxDb object.
For recount2 we used Gencode v25 hg38’s annotation, which means that we
need to process it manually instead of using a pre-computed TxDb
package. This is where the <code>GenomicState</code> <span class="citation">(27)</span> package comes into play as it has done the
heavy lifting for us already.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Import the Gencode v25 hg38 gene annotation</span></span>
<span><span class="co">## using GenomicState</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/GenomicState" class="external-link">"GenomicState"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create the TxDb object</span></span>
<span><span class="va">gencode_v25_hg38_txdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"25"</span>, genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"TxDb"</span></span>
<span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Explore the TxDb object</span></span>
<span><span class="va">gencode_v25_hg38_txdb</span></span></code></pre></div>
<p>Now that we have a TxDb object for Gencode v25 on hg38 coordinates,
we can use <code>bumphunter</code>’s <span class="citation">(28)</span>
annotation functions for annotating the original 10 regions we were
working with as well as the annotated genes that we can download using
<code>GenomicState</code>.</p>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download annotated transcripts for gencode v25</span></span>
<span><span class="va">ann_gencode_v25_hg38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"25"</span>, genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"AnnotatedGenes"</span></span>
<span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Annotate the regions of interest</span></span>
<span><span class="co">## Note that we are using the original regions, not the resized ones</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rafalab/bumphunter" class="external-link">"bumphunter"</a></span><span class="op">)</span></span>
<span><span class="va">nearest_ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html" class="external-link">matchGenes</a></span><span class="op">(</span><span class="va">regions_by_padj</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">ann_gencode_v25_hg38</span><span class="op">)</span></span></code></pre></div>
<p>The final piece we need to run <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage()</a></code> is
information about which base-pairs are exonic, intronic, etc. This is
done via the <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">annotateRegions()</a></code> function in
<code>derfinder</code>, which itself requires prior processing of the
TxDb information by <code><a href="https://rdrr.io/pkg/derfinder/man/makeGenomicState.html" class="external-link">makeGenomicState()</a></code> that we can download
with <code>GenomicState</code>.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the genomic state object for Gencode v25</span></span>
<span><span class="va">gs_gencode_v25_hg38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"25"</span>, genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"GenomicState"</span></span>
<span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Annotate the original regions</span></span>
<span><span class="va">regions_ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">annotateRegions</a></span><span class="op">(</span></span>
<span>    <span class="va">regions_resized</span>,</span>
<span>    <span class="va">gs_gencode_v25_hg38</span><span class="op">$</span><span class="va">fullGenome</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We can finally use <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage()</a></code> to visualize the
top 10 regions coloring by whether they are prenatal or postnatal
samples. Known exons are shown in dark blue, introns in light blue.</p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/derfinderPlot" class="external-link">"derfinderPlot"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"region_plots.pdf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions_resized</span>, regionCoverage <span class="op">=</span> <span class="va">regionCov</span>,</span>
<span>    groupInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span>,</span>
<span>    nearestAnnotation <span class="op">=</span> <span class="va">nearest_ann</span>,</span>
<span>    annotatedRegions <span class="op">=</span> <span class="va">regions_ann</span>,</span>
<span>    txdb <span class="op">=</span> <span class="va">gencode_v25_hg38_txdb</span>,</span>
<span>    scalefac <span class="op">=</span> <span class="fl">1</span>, ylab <span class="op">=</span> <span class="st">"Coverage (RP40M, 100bp)"</span>,</span>
<span>    ask <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Visualize DER #2</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions_resized</span>, regionCoverage <span class="op">=</span> <span class="va">regionCov</span>,</span>
<span>    groupInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_er_scaled</span><span class="op">)</span><span class="op">$</span><span class="va">prenatal</span>,</span>
<span>    nearestAnnotation <span class="op">=</span> <span class="va">nearest_ann</span>,</span>
<span>    annotatedRegions <span class="op">=</span> <span class="va">regions_ann</span>,</span>
<span>    txdb <span class="op">=</span> <span class="va">gencode_v25_hg38_txdb</span>,</span>
<span>    scalefac <span class="op">=</span> <span class="fl">1</span>, ylab <span class="op">=</span> <span class="st">"Coverage (RP40M, 100bp)"</span>,</span>
<span>    ask <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, whichRegions <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>In plots like Figure @ref(fig:regionplots) we can see that some DERs
match known exons (DERs 2, 8, 10), some are longer than known exons
(DERs 1, 7, 9), and others are exon fragments (DERs 3, 4, 5, 6) which
could be due to the cutoff used. Note that DER 7 could be shorter than a
known exon due to a coverage dip.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>In this workflow we described in detail the available data in
recount2, how the coverage count matrices were computed, the metadata
included in recount2 and how to get new phenotypic information from
other sources. We showed how to perform a DE analysis at the gene and
exon levels as well as use an annotation-agnostic approach. Finally, we
explained how to visualize the base-pair information for a given set of
regions. This workflow constitutes a strong basis to leverage the
recount2 data for human RNA-seq analyses.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This workflow was created using <code>BiocWorkflowTools</code> <span class="citation">(29)</span>. The session information is available in
Supplementary File 2. The most recent version of this workflow is
available via Bioconductor at <a href="http://bioconductor.org/help/workflows/" class="external-link uri">http://bioconductor.org/help/workflows/</a>.</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Final list of files created</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="st">"SRP045638"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "gene_report.bib"  "gene_report.html" "rse_exon.Rdata"   "rse_gene.Rdata"   "SraRunTable.txt"</span></span></code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Pandoc information</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'rmarkdown'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:BiocStyle':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     html_document, md_document, pdf_document</span></span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/pandoc_available.html" class="external-link">pandoc_version</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] '2.19.2'</span></span></code></pre>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Time for reproducing this workflow, in minutes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/proc.time.html" class="external-link">proc.time</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span>, <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## elapsed </span></span>
<span><span class="co">##     6.2</span></span></code></pre>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">"sessioninfo"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.3.0 (2023-04-21 ucrt)</span></span>
<span><span class="co">##  os       Windows 11 x64 (build 22621)</span></span>
<span><span class="co">##  system   x86_64, mingw32</span></span>
<span><span class="co">##  ui       RTerm</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  English_United States.utf8</span></span>
<span><span class="co">##  ctype    English_United States.utf8</span></span>
<span><span class="co">##  tz       America/New_York</span></span>
<span><span class="co">##  date     2023-05-07</span></span>
<span><span class="co">##  pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  AnnotationDbi          * 1.62.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationFilter         1.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationHub          * 3.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ape                      5.7-1     <span style="color: #555555;">2023-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  aplot                    0.1.10    <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  backports                1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  base64enc                0.1-3     <span style="color: #555555;">2015-07-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bibtex                   0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Biobase                * 2.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocFileCache          * 2.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocGenerics           * 0.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocIO                   1.10.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocManager              1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  BiocParallel             1.34.1    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocStyle              * 2.28.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocVersion              3.17.1    <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biomaRt                  2.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  Biostrings               2.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biovizBase               1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bit                      4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bit64                    4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bitops                   1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  blob                     1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bookdown                 0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  BSgenome                 1.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bslib                    0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bumphunter             * 1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  cachem                   1.0.8     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  caTools                  1.18.2    <span style="color: #555555;">2021-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  checkmate                2.2.0     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  cli                      3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  cluster                  2.1.4     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  clusterProfiler        * 4.8.1     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  codetools                0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  colorspace               2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  cowplot                  1.1.1     <span style="color: #555555;">2020-12-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  crayon                   1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  crosstalk                1.2.0     <span style="color: #555555;">2021-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  curl                     5.0.0     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  data.table               1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  DBI                      1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  dbplyr                 * 2.3.2     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  DEFormats                1.28.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  DelayedArray             0.26.2    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinder              * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderHelper          1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderPlot          * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  desc                     1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  DESeq2                 * 1.40.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  dichromat                2.0-0.1   <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  digest                   0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  doRNG                    1.8.6     <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  DOSE                     3.26.1    <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  downloader               0.4       <span style="color: #555555;">2015-07-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  dplyr                    1.1.2     <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  DT                     * 0.27      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  edgeR                  * 3.42.2    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ellipsis                 0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  enrichplot               1.20.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ensembldb                2.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  evaluate                 0.21      <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fansi                    1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  farver                   2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fastmap                  1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fastmatch                1.1-3     <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fgsea                    1.26.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  filelock                 1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  foreach                * 1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  foreign                  0.8-84    <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Formula                  1.2-5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fs                       1.6.2     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  generics                 0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  GenomeInfoDb           * 1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomeInfoDbData         1.2.10    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicAlignments        1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFeatures          1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFiles             1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicRanges          * 1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicState           * 0.99.15   <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GEOquery                 2.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GGally                   2.1.2     <span style="color: #555555;">2021-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggbio                    1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ggforce                  0.4.1     <span style="color: #555555;">2022-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggfun                    0.0.9     <span style="color: #555555;">2022-11-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggplot2                * 3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggplotify                0.1.0     <span style="color: #555555;">2021-09-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggraph                   2.1.0     <span style="color: #555555;">2022-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggrepel                  0.9.3     <span style="color: #555555;">2023-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggtree                   3.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  glue                     1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  GO.db                    3.17.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GOSemSim                 2.26.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  gplots                 * 3.1.3     <span style="color: #555555;">2022-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  graph                    1.78.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  graphlayouts             1.0.0     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gridExtra                2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gridGraphics             0.5-1     <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gson                     0.1.0     <span style="color: #555555;">2023-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gtable                   0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gtools                   3.9.4     <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  HDO.db                   0.99.1    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  highr                    0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Hmisc                    5.0-1     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  hms                      1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  htmlTable                2.4.1     <span style="color: #555555;">2022-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  htmltools                0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  htmlwidgets              1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  httpuv                   1.6.9     <span style="color: #555555;">2023-02-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  httr                     1.4.5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  igraph                   1.4.2     <span style="color: #555555;">2023-04-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  interactiveDisplayBase   1.38.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  IRanges                * 2.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  iterators              * 1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  jquerylib                0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  jsonlite                 1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  KEGGREST                 1.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  KernSmooth               2.23-20   <span style="color: #555555;">2021-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  knitr                  * 1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  knitrBootstrap           1.0.2     <span style="color: #555555;">2018-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  labeling                 0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  later                    1.3.1     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lattice                  0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lazyeval                 0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lifecycle                1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  limma                  * 3.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  locfit                 * 1.5-9.7   <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lubridate                1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  magrittr                 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  markdown                 1.6       <span style="color: #555555;">2023-04-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  MASS                     7.3-58.4  <span style="color: #555555;">2023-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Matrix                   1.5-4     <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  MatrixGenerics         * 1.12.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  matrixStats            * 0.63.0    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  memoise                  2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  mime                     0.12      <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  munsell                  0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  nlme                     3.1-162   <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  nnet                     7.3-18    <span style="color: #555555;">2022-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  org.Hs.eg.db           * 3.17.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  OrganismDbi              1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  patchwork                1.1.2     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  pheatmap               * 1.0.12    <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  pillar                   1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgconfig                2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgdown                  2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  plyr                     1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  png                      0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  polyclip                 1.10-4    <span style="color: #555555;">2022-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  prettyunits              1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  progress                 1.2.2     <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  promises                 1.2.0.1   <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ProtGenerics             1.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  purrr                    1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  qvalue                   2.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  R6                       2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ragg                     1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rappdirs                 0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  RBGL                     1.76.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RColorBrewer           * 1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Rcpp                     1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  RCurl                    1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  readr                    2.1.4     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  recount                * 1.26.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RefManageR               1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  regionReport           * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  rentrez                  1.2.3     <span style="color: #555555;">2020-11-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  reshape                  0.8.9     <span style="color: #555555;">2022-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  reshape2                 1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  restfulr                 0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rjson                    0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rlang                    1.1.1     <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rmarkdown              * 2.21      <span style="color: #555555;">2023-03-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rngtools                 1.5.2     <span style="color: #555555;">2021-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rpart                    4.1.19    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rprojroot                2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Rsamtools                2.16.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RSQLite                  2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rstudioapi               0.14      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rtracklayer              1.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Arrays                 1.0.1     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Vectors              * 0.38.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sass                     0.4.6     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  scales                   1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  scatterpie               0.1.9     <span style="color: #555555;">2023-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  sessioninfo            * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  shadowtext               0.1.2     <span style="color: #555555;">2022-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  shiny                    1.7.4     <span style="color: #555555;">2022-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  stringi                  1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  stringr                  1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  SummarizedExperiment   * 1.30.1    <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  systemfonts              1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  textshaping              0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tibble                   3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tidygraph                1.2.3     <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tidyr                    1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tidyselect               1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tidytree                 0.4.2     <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  timechange               0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  treeio                   1.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  tweenr                   2.0.2     <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  tzdb                     0.3.0     <span style="color: #555555;">2022-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  utf8                     1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  VariantAnnotation        1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  vctrs                    0.6.2     <span style="color: #555555;">2023-04-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  viridis                  0.6.3     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  viridisLite              0.4.2     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  withr                    2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  xfun                     0.39      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  XML                      3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  xml2                     1.3.4     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  xtable                   1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  XVector                  0.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  yaml                     2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  yulab.utils              0.0.6     <span style="color: #555555;">2022-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  zlibbioc                 1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] C:/R/R-4.3.0bioc3.17/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="competing-interests">Competing interests<a class="anchor" aria-label="anchor" href="#competing-interests"></a>
</h2>
<p>No competing interests were disclosed.</p>
</div>
<div class="section level2">
<h2 id="grant-information">Grant information<a class="anchor" aria-label="anchor" href="#grant-information"></a>
</h2>
<p>LCT and AEJ were supported by the National Institutes of Health
(grant R21 MH109956-01). LCT and AN were supported by the National
Institutes of Health (grant R01 GM105705).</p>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>We would like to acknowledge the members of Andrew Jaffe (Lieber
Institute for Brain Development, Johns Hopkins Medical Campus) and
Alexis Battle (Department of Computer Science, Whiting School of
Engineering at Johns Hopkins University) labs for feedback on the
explanatory figures.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-huber2015" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Huber W, Carey VJ, Gentleman R, Anders S,
Carlson M, Carvalho BS, et al. <a href="https://doi.org/10.1038/nmeth.3252" class="external-link">Orchestrating high-throughput
genomic analysis with bioconductor</a>. Nature methods.
2015;12(2):115–21. </div>
</div>
<div id="ref-Law2016a" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Law
CW, Alhamdoosh M, Su S, Smyth GK, Ritchie ME. <a href="https://doi.org/10.12688/f1000research.9005.2" class="external-link"><span class="nocase">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and
edgeR.</span></a> F1000Research. 2016;5(May):1408. </div>
</div>
<div id="ref-Love2016" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Love
MI, Anders S, Kim V, Huber W. <a href="https://doi.org/10.12688/f1000research.7035.2" class="external-link"><span class="nocase">RNA-Seq workflow: gene-level exploratory analysis and
differential expression</span></a>. F1000Research. 2016;4(May):1070.
</div>
</div>
<div id="ref-Chen2016" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Chen
Y, Lun ATL, Smyth GK. <a href="https://doi.org/10.12688/f1000research.8987.2" class="external-link"><span class="nocase">From reads to genes to pathways: differential expression
analysis of RNA-Seq experiments using Rsubread and the edgeR
quasi-likelihood pipeline.</span></a> F1000Research. 2016;5(May):1438.
</div>
</div>
<div id="ref-frazee2011" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Frazee AC, Langmead B, Leek JT. <a href="https://doi.org/10.1186/1471-2105-12-449" class="external-link"><span>ReCount</span>: A
multi-experiment resource of analysis-ready <span>RNA</span>-seq gene
count datasets</a>. BMC bioinformatics. 2011;12:449. </div>
</div>
<div id="ref-airway" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">Himes, E. B, Jiang, X., Wagner, P., et al.
<span class="nocase"><span>R</span><span>N</span><span>A</span>-<span>S</span>eq
<span>T</span>ranscriptome <span>P</span>rofiling
<span>I</span>dentifies
<span>C</span><span>R</span><span>I</span><span>S</span><span>P</span><span>L</span><span>D</span>2
as a <span>G</span>lucocorticoid <span>R</span>esponsive
<span>G</span>ene that <span>M</span>odulates <span>C</span>ytokine
<span>F</span>unction in <span>A</span>irway <span>S</span>mooth
<span>M</span>uscle <span>C</span>ells</span>. PLoS ONE.
2014;9(6):e99625. </div>
</div>
<div id="ref-collado2017b" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">Collado-Torres L, Nellore A, Kammers K, Ellis
SE, Taub MA, Hansen KD, et al. <a href="https://doi.org/10.1038/nbt.3838" class="external-link">Reproducible
<span>RNA</span>-seq analysis using recount2</a>. Nature Biotechnology.
2017 Apr;35(4):319–21. </div>
</div>
<div id="ref-collado2017a" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">Collado-Torres L, Nellore A, Frazee AC, Wilks
C, Love MI, Langmead B, et al. <a href="https://doi.org/10.1093/nar/gkw852" class="external-link">Flexible expressed region
analysis for <span>RNA</span>-seq with derfinder</a>. Nucleic Acids
Research. 2017 Jan;45(2):e9–9. </div>
</div>
<div id="ref-summarizedexperiment" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">Morgan M, Obenchain V, Hester J, Pagès H.
SummarizedExperiment: SummarizedExperiment container. 2017. </div>
</div>
<div id="ref-Wilks2017" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">Wilks C, Gaddipati P, Nellore A, Langmead B. <a href="https://doi.org/10.1101/097881" class="external-link"><span class="nocase">Snaptron:
querying and visualizing splicing across tens of thousands of RNA-seq
samples</span></a>. bioRxiv. 2017; </div>
</div>
<div id="ref-rail2016" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">Nellore A, Collado-Torres L, Jaffe AE,
Alquicira-Hernández J, Wilks C, Pritt J, et al. <a href="https://doi.org/10.1093/bioinformatics/btw575" class="external-link">Rail-<span>RNA</span>:
Scalable analysis of <span>RNA</span>-seq splicing and coverage</a>.
Bioinformatics (Oxford, England). 2016 Sep; </div>
</div>
<div id="ref-lawrence2013" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">Lawrence M, Huber W, Pagès H, Aboyoun P,
Carlson M, Gentleman R, et al. <a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">Software for
computing and annotating genomic ranges</a>. PLoS computational biology.
2013;9(8):e1003118. </div>
</div>
<div id="ref-jaffe2015" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">Jaffe AE, Shin J, Collado-Torres L, Leek JT,
Tao R, Li C, et al. <a href="https://doi.org/10.1038/nn.3898" class="external-link">Developmental regulation of human
cortex transcription and its clinical relevance at single base
resolution</a>. Nature Neuroscience. 2015 Jan;18(1):154–61. </div>
</div>
<div id="ref-Ellis2017" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">Ellis SE, Collado-Torres L, Leek J. <a href="https://doi.org/10.1101/145656" class="external-link">Improving the value of public
RNA-seq expression data by phenotype prediction</a>. bioRxiv. 2017;
</div>
</div>
<div id="ref-TCGAbiolinks" class="csl-entry">
<div class="csl-left-margin">15. </div>
<div class="csl-right-inline">Colaprico A, Silva TC, Olsen C, Garofano L,
Cava C, Garolini D, et al. <a href="https://doi.org/10.1093/nar/gkv1507" class="external-link">TCGAbiolinks: An
r/bioconductor package for integrative analysis of TCGA data</a>.
Nucleic Acids Research. 2015; </div>
</div>
<div id="ref-law2014voom" class="csl-entry">
<div class="csl-left-margin">16. </div>
<div class="csl-right-inline">Law
CW, Chen Y, Shi W, Smyth GK. Voom: Precision weights unlock linear model
analysis tools for RNA-seq read counts. Genome Biol. 2014;15(2):R29.
</div>
</div>
<div id="ref-collado2016regionreport" class="csl-entry">
<div class="csl-left-margin">17. </div>
<div class="csl-right-inline">Collado-Torres L, Jaffe AE, Leek JT. <a href="https://doi.org/10.12688/f1000research.6379.2" class="external-link"><span class="nocase">regionReport</span>: Interactive reports for region-level
and feature-level genomic analyses [version2; referees: 2 approved, 1
approved with reservations]</a>. F1000Research. 2016 Jun;4:1–10. </div>
</div>
<div id="ref-love2014moderated" class="csl-entry">
<div class="csl-left-margin">18. </div>
<div class="csl-right-inline">Love MI, Huber W, Anders S. Moderated
estimation of fold change and dispersion for RNA-seq data with DESeq2.
Genome biology. 2014;15(12):1–21. </div>
</div>
<div id="ref-robinson2010" class="csl-entry">
<div class="csl-left-margin">19. </div>
<div class="csl-right-inline">Robinson MD, McCarthy DJ, Smyth GK. <a href="https://doi.org/10.1093/bioinformatics/btp616" class="external-link"><span class="nocase">edgeR:</span> A bioconductor package for differential
expression analysis of digital gene expression data</a>. Bioinformatics
(Oxford, England). 2010 Jan;26(1):139–40. </div>
</div>
<div id="ref-ideal" class="csl-entry">
<div class="csl-left-margin">20. </div>
<div class="csl-right-inline">Marini F. Ideal: Interactive differential
expression AnaLysis. 2017. </div>
</div>
<div id="ref-clusterProfiler" class="csl-entry">
<div class="csl-left-margin">21. </div>
<div class="csl-right-inline">Yu
G, Wang LG, Han Y, He QY. <a href="https://doi.org/10.1089/omi.2011.0118" class="external-link">clusterProfiler: An r
package for comparing biological themes among gene clusters</a>. OMICS:
A Journal of Integrative Biology. 2012;16(5):284–7. </div>
</div>
<div id="ref-Soneson2015" class="csl-entry">
<div class="csl-left-margin">22. </div>
<div class="csl-right-inline">Soneson C, Love MI, Robinson MD. <a href="https://doi.org/10.12688/f1000research.7563.2" class="external-link"><span class="nocase">Differential analyses for RNA-seq: transcript-level
estimates improve gene-level inferences. [version 2; referees: 2
approved]</span></a>. F1000Research. 2015;4(0):1521. </div>
</div>
<div id="ref-jaffe2017" class="csl-entry">
<div class="csl-left-margin">23. </div>
<div class="csl-right-inline">Jaffe AE, Straub R, Shin JH, Tao R, Gao Y,
Collado-Torres L, et al. <span class="nocase">Developmental and genetic
regulation of the human cortex transcriptome in schizophrenia</span>.
bioRxiv. 2017; </div>
</div>
<div id="ref-rtracklayer" class="csl-entry">
<div class="csl-left-margin">24. </div>
<div class="csl-right-inline">Lawrence M, Gentleman R, Carey V. <a href="https://doi.org/10.1093/bioinformatics/btp328" class="external-link">Rtracklayer: An r
package for interfacing with genome browsers</a>. Bioinformatics.
2009;25:1841–2. </div>
</div>
<div id="ref-epivizr" class="csl-entry">
<div class="csl-left-margin">25. </div>
<div class="csl-right-inline">Bravo HC, Chelaru F, Smith L, Goldstein N,
Kancherla J, Walter M, et al. Epivizr: R interface to epiviz web app.
2017. </div>
</div>
<div id="ref-wiggleplotr" class="csl-entry">
<div class="csl-left-margin">26. </div>
<div class="csl-right-inline">Alasoo K. Wiggleplotr: Make read coverage plots
from BigWig files. 2017. </div>
</div>
<div id="ref-GenomicState" class="csl-entry">
<div class="csl-left-margin">27. </div>
<div class="csl-right-inline">Collado-Torres L. GenomicState: Build and
access GenomicState objects for use with derfinder tools from sources
like gencode [Internet]. 2019. Available from: <a href="https://github.com/LieberInstitute/GenomicState" class="external-link">https://github.com/LieberInstitute/GenomicState</a>
</div>
</div>
<div id="ref-bumphunter" class="csl-entry">
<div class="csl-left-margin">28. </div>
<div class="csl-right-inline">Jaffe AE, Murakami P, Lee H, Leek JT, Fallin
DM, Feinberg AP, et al. <a href="https://doi.org/10.1093/ije/dyr238" class="external-link">Bump hunting to identify
differentially methylated regions in epigenetic epidemiology
studies</a>. International journal of epidemiology. 2012;41(1):200–9.
</div>
</div>
<div id="ref-biocWorkflow" class="csl-entry">
<div class="csl-left-margin">29. </div>
<div class="csl-right-inline">Smith M, Oleś A. BiocWorkflowTools: Tools to
aid the development of bioconductor workflow packages. 2017. </div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
