<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor • recountWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor">
<meta property="og:description" content="recountWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recountWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.13.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/recount-workflow.html">recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/recountWorkflow/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>recount workflow: accessing over 70,000 human RNA-seq samples with Bioconductor</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus, Baltimore, MD, 21205, USA<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 class="author">Abhinav Nellore</h4>
            <address class="author_afil">
      Department of Biomedical Engineering, Oregon Health and Science University, Portland, OR, 97239, USA<br><h4 class="author">Andrew E. Jaffe</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus, Baltimore, MD, 21205, USA<br><small class="dont-index">Source: <a href="https://github.com/LieberInstitute/recountWorkflow/blob/master/vignettes/recount-workflow.Rmd"><code>vignettes/recount-workflow.Rmd</code></a></small>
      <div class="hidden name"><code>recount-workflow.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      The recount2 resource is composed of over 70,000 uniformly processed human RNA-seq samples spanning TCGA and SRA, including GTEx. The processed data can be accessed via the recount2 website and the <code>recount</code> Bioconductor package. This workflow explains in detail how to use the <code>recount</code> package and how to integrate it with other Bioconductor packages for several analyses that can be carried out with the recount2 resource. In particular, we describe how the coverage count matrices were computed in recount2 as well as different ways of obtaining public metadata, which can facilitate downstream analyses. Step-by-step directions show how to do a gene-level differential expression analysis, visualize base-level genome coverage data, and perform an analyses at multiple feature levels. This workflow thus provides further information to understand the data in recount2 and a compendium of R code to use the data.
    </div>
    
<p><strong>R version</strong>: R version 4.0.0 (2020-04-24)</p>
<p><strong>Bioconductor version</strong>: 3.12</p>
<p><strong>Package</strong>: 1.13.1</p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>RNA sequencing (RNA-seq) is now the most widely used high-throughput assay for measuring gene expression. In a typical RNA-seq experiment, several million reads are sequenced per sample. The reads are often aligned to the reference genome using a splice-aware aligner to identify where reads originated. Resulting alignment files are then used to compute count matrices for several analyses such as identifying differentially expressed genes. The Bioconductor project <span class="citation">(1)</span> has many contributed packages that specialize in analyzing this type of data and previous workflows have explained how to use them <span class="citation">(2–4)</span>. Initial steps are typically focused on generating the count matrices. Some pre-computed matrices have been made available via the ReCount project <span class="citation">(5)</span> or Bioconductor Experiment data packages such as the <code>airway</code> dataset <span class="citation">(6)</span>. The pre-computed count matrices in ReCount have been useful to RNA-seq methods developers and to researchers seeking to avoid the computationally intensive process of creating these matrices. In the years since ReCount was published, hundreds of new RNA-seq projects have been carried out, and researchers have shared the data publicly.</p>
<p>We recently uniformly processed over 70,000 publicly available human RNA-seq samples, and made the data available via the recount2 resource <span class="citation">(7)</span> at <a href="https://jhubiostatistics.shinyapps.io/recount/">jhubiostatistics.shinyapps.io/recount/</a>. Samples in recount2 are grouped by project (over 2,000) originating from the Sequence Read Archive, the Genotype-Tissue Expression study (GTEx) and the Cancer Genome Atlas (TCGA). The processed data can be accessed via the <code>recount</code> Bioconductor package available at <a href="http://bioconductor.org/packages/recount">bioconductor.org/packages/recount</a>. Together, recount2 and the <code>recount</code> Bioconductor package should be considered a successor to ReCount.</p>
<p>Due to space constraints, the recount2 publication <span class="citation">(7)</span> did not cover how to use the <code>recount</code> package and other useful information for carrying out analyses with recount2 data. We describe how the count matrices in recount2 were generated. We also review the R code necessary for using the recount2 data, whose details are important because some of this code involves multiple Bioconductor packages and changing default options. We further show: (a) how to augment metadata that comes with datasets with metadata learned from natural language processing of associated papers as well as expression data (b) how to perform differential expression analyses, and (c) how to visualize the base-pair data available from recount2.</p>
</div>
<div id="analysis-of-rna-seq-data-available-at-recount2" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-of-rna-seq-data-available-at-recount2" class="anchor"></a>Analysis of RNA-seq data available at recount2</h1>
<div id="recount2-overview" class="section level2">
<h2 class="hasAnchor">
<a href="#recount2-overview" class="anchor"></a>recount2 overview</h2>
<p>The recount2 resource provides expression data summarized at different feature levels to enable novel cross-study analyses. Generally when investigators use the term <em>expression</em>, they think about gene expression. But more information can be extracted from RNA-seq data. Once RNA-seq reads have been aligned to the reference genome it is possible to determine the number of aligned reads overlapping each base-pair resulting in the genome base-pair coverage curve as shown in Figure @ref(fig:Figure1). In the example shown in Figure @ref(fig:Figure1), most of the reads overlap known exons from a gene. Those reads can be used to compute a count matrix at the exon or gene feature levels. Some reads span exon-exon junctions (jx) and while most match the annotation, some do not (jx 3 and 4). An exon-exon junction count matrix can be used to identify differentially expressed junctions, which can show which isoforms are differentially expressed given sufficient coverage. For example, junctions 2 and 5 are unique to isoform 2, while junction 6 is unique to isoform 1. The genome base-pair coverage data can be used with <code>derfinder</code> <span class="citation">(8)</span> to identify expressed regions; some of them could be unannotated exons, which together with the exon-exon junction data could help establish new isoforms.</p>
<div class="figure" style="text-align: center">
<img src="Figure1.png" alt="Overview of the data available in recount2. Reads (pink boxes) aligned to the reference genome can be used to compute a base-pair coverage curve and identify exon-exon junctions (split reads). Gene and exon count matrices are generated using annotation information providing the gene (green boxes) and exon (blue boxes) coordinates together with the base-level coverage curve. The reads spanning exon-exon junctions (jx) are used to compute a third count matrix that might include unannotated junctions (jx 3 and 4). Without using annotation information, expressed regions (orange box) can be determined from the base-level coverage curve to then construct data-driven count matrices." width="100%"><p class="caption">
Overview of the data available in recount2. Reads (pink boxes) aligned to the reference genome can be used to compute a base-pair coverage curve and identify exon-exon junctions (split reads). Gene and exon count matrices are generated using annotation information providing the gene (green boxes) and exon (blue boxes) coordinates together with the base-level coverage curve. The reads spanning exon-exon junctions (jx) are used to compute a third count matrix that might include unannotated junctions (jx 3 and 4). Without using annotation information, expressed regions (orange box) can be determined from the base-level coverage curve to then construct data-driven count matrices.
</p>
</div>
<p>recount2 provides gene, exon, and exon-exon junction count matrices both in text format and <em>RangedSummarizedExperiment</em> objects (rse) <span class="citation">(9)</span> as shown in Figure @ref(fig:Figure2). These rse objects provide information about the expression features (for example gene IDs) and the samples. In this workflow we will explain how to add metadata to the rse objects in recount2 in order to ask biological questions. recount2 also provides coverage data in the form of bigWig files. All four features can be accessed with the <code>recount</code> Bioconductor package <span class="citation">(7)</span>. <code>recount</code> also allows sending queries to <code>snaptron</code> <span class="citation">(10)</span> to search for specific exon-exon junctions.</p>
<div class="figure" style="text-align: center">
<img src="Figure2.png" alt="recount2 provides coverage count matrices in RangedSummarizedExperiment (rse) objects. Once the rse object has been downloaded and loaded into R, the feature information is accessed with rowRanges(rse) (blue box), the counts with assays(rse)\$counts (pink box) and the sample metadata with colData(rse) (green box). The sample metadata can be expanded using add\_predictions(rse) (orange box) or with custom code (brown box) matching by a unique sample identifier such as the SRA Run ID. The rse object is inside the purple box and matching data is highlighted in each box." width="100%"><p class="caption">
recount2 provides coverage count matrices in RangedSummarizedExperiment (rse) objects. Once the rse object has been downloaded and loaded into R, the feature information is accessed with rowRanges(rse) (blue box), the counts with assays(rse)$counts (pink box) and the sample metadata with colData(rse) (green box). The sample metadata can be expanded using add_predictions(rse) (orange box) or with custom code (brown box) matching by a unique sample identifier such as the SRA Run ID. The rse object is inside the purple box and matching data is highlighted in each box.
</p>
</div>
</div>
<div id="packages-used-in-the-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#packages-used-in-the-workflow" class="anchor"></a>Packages used in the workflow</h2>
<p>In this workflow we will use several Bioconductor packages. To reproduce the entirety of this workflow, install the packages using the following code after installing R 3.4.x from CRAN in order to use Bioconductor version 3.5 or newer.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">## Install packages from Bioconductor</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) {
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
  }
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"recount"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"limma"</span>, <span class="st">"edgeR"</span>, <span class="st">"DESeq2"</span>,
    <span class="st">"regionReport"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"gplots"</span>,
    <span class="st">"derfinder"</span>, <span class="st">"GenomicState"</span>, <span class="st">"bumphunter"</span>, <span class="st">"derfinderPlot"</span>, <span class="st">"sessioninfo"</span>
))</pre></body></html></div>
<p>Once they are installed, load all the packages with the following code.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"limma"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"edgeR"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"regionReport"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"clusterProfiler"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"org.Hs.eg.db"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"gplots"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinder"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicState"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"bumphunter"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderPlot"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"sessioninfo"</span>)</pre></body></html></div>
</div>
<div id="coverage-counts-provided-by-recount2" class="section level2">
<h2 class="hasAnchor">
<a href="#coverage-counts-provided-by-recount2" class="anchor"></a>Coverage counts provided by recount2</h2>
<p>The most accessible features are the gene, exon and exon-exon junction count matrices. This section explains them in greater detail. Figure @ref(fig:Figure3) shows 16 RNA-seq reads, each 3 base-pairs long, and a reference genome.</p>
<div class="figure" style="text-align: center">
<img src="Figure3.png" alt="RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs long are shown (pink boxes) alongside a reference genome that is 16 base-pairs long (white box)." width="100%"><p class="caption">
RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs long are shown (pink boxes) alongside a reference genome that is 16 base-pairs long (white box).
</p>
</div>
<p>Reads in the recount2 resource were aligned with the splice-aware Rail-RNA aligner <span class="citation">(11)</span>. Figure @ref(fig:Figure4) shows the reads aligned to the reference genome. Some of the reads are split as they span an exon-exon junction. Two of the reads were soft clipped meaning that just a portion of the reads aligned (top left in purple).</p>
<div class="figure" style="text-align: center">
<img src="Figure4.png" alt="Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are able to find the coordinates to which the reads map, even if they span exon-exon junctions (connected boxes). Rail-RNA soft clips some reads (purple boxes with rough edges) such that a portion of these reads align to the reference genome." width="100%"><p class="caption">
Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are able to find the coordinates to which the reads map, even if they span exon-exon junctions (connected boxes). Rail-RNA soft clips some reads (purple boxes with rough edges) such that a portion of these reads align to the reference genome.
</p>
</div>
<p>In order to compute the gene and exon count matrices we first have to process the annotation, which for recount2 is Gencode v25 (CHR regions) with hg38 coordinates. Although <code>recount</code> can generate count matrices for other annotations using hg38 coordinates. Figure @ref(fig:Figure5) shows two isoforms for a gene composed of 3 different exons.</p>
<div class="figure" style="text-align: center">
<img src="Figure5.png" alt="Gene annotation. A single gene with two isoforms composed by three distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the first five base-pairs while exon 2 is common to both isoforms." width="100%"><p class="caption">
Gene annotation. A single gene with two isoforms composed by three distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the first five base-pairs while exon 2 is common to both isoforms.
</p>
</div>
<p>The coverage curve is at base-pair resolution, so if we are interested in gene counts we have to be careful not to double count base-pairs 1 through 5 that are shared by exons 1 and 3 (Figure @ref(fig:Figure5)). Using the function <code><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">disjoin()</a></code> from <code>GenomicRanges</code> <span class="citation">(12)</span> we identified the distinct exonic sequences (disjoint exons). The following code defines the exon coordinates that match Figure @ref(fig:Figure5) and the resulting disjoint exons for our example gene. The resulting disjoint exons are shown in Figure @ref(fig:Figure6).</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="no">exons</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="st">"seq"</span>, <span class="fu">IRanges</span>(<span class="kw">start</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">13</span>), <span class="kw">end</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>, <span class="fl">8</span>, <span class="fl">15</span>)))
<span class="no">exons</span></pre></body></html></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]      seq       1-5      *
##   [2]      seq       1-8      *
##   [3]      seq     13-15      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">disjoin</a></span>(<span class="no">exons</span>)</pre></body></html></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]      seq       1-5      *
##   [2]      seq       6-8      *
##   [3]      seq     13-15      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="figure" style="text-align: center">
<img src="Figure6.png" alt="Disjoint exons. Windows of distinct exonic sequence for the example gene. Disjoint exons 1 and 2 form exon 1." width="100%"><p class="caption">
Disjoint exons. Windows of distinct exonic sequence for the example gene. Disjoint exons 1 and 2 form exon 1.
</p>
</div>
<p>Now that we have disjoint exons, we can compute the base-pair coverage for each of them as shown in Figure @ref(fig:Figure7). That is, for each base-pair that corresponds to exonic sequence, we compute the number of reads overlapping that given base-pair. For example, the first base-pair is covered by 3 different reads and it does not matter whether the reads themselves were soft clipped. Not all reads or bases of a read contribute information to this step, as some do not overlap known exonic sequence (light pink in Figure @ref(fig:Figure7)).</p>
<div class="figure" style="text-align: center">
<img src="Figure7.png" alt="Base-pair coverage counting for exonic base-pairs. At each exonic base-pair we compute the number of reads overlapping that given base-pair. The first base (orange arrow) has 3 reads overlapping that base-pair. Base-pair 11 has a coverage of 3 but does not overlap known exonic sequence, so that information is not used for the gene and exon count matrices (grey arrow). If a read partially overlaps exonic sequence, only the portion that overlaps is used in the computation (see right most read)." width="100%"><p class="caption">
Base-pair coverage counting for exonic base-pairs. At each exonic base-pair we compute the number of reads overlapping that given base-pair. The first base (orange arrow) has 3 reads overlapping that base-pair. Base-pair 11 has a coverage of 3 but does not overlap known exonic sequence, so that information is not used for the gene and exon count matrices (grey arrow). If a read partially overlaps exonic sequence, only the portion that overlaps is used in the computation (see right most read).
</p>
</div>
<p>With base-pair coverage for the exonic sequences computed, the coverage count for each distinct exon is simply the sum of the base-pair coverage for each base in a given distinct exon. For example, the coverage count for disjoint exon 2 is <span class="math inline">\(2 + 2 + 3 = 7\)</span> as shown in Figure @ref(fig:Figure8). The gene coverage count is then <span class="math inline">\(\sum_i^n \texttt{coverage}_i\)</span> where <span class="math inline">\(n\)</span> is the number of exonic base-pairs for the gene and is equal to the sum of the coverage counts for its disjoint exons as shown in Figure @ref(fig:Figure8).</p>
<div class="figure" style="text-align: center">
<img src="Figure8.png" alt="Exon and gene coverage counts. The coverage counts for each disjoint exon are the sum of the base-pair coverage. The gene coverage count is the sum of the disjoint exons coverage counts." width="100%"><p class="caption">
Exon and gene coverage counts. The coverage counts for each disjoint exon are the sum of the base-pair coverage. The gene coverage count is the sum of the disjoint exons coverage counts.
</p>
</div>
<p>For the exons, recount2 provides the disjoint exons coverage count matrix. It is possible to reconstruct the exon coverage count matrix by summing the coverage count for the disjoint exons that compose each exon. For example, the coverage count for exon 1 would be the sum of the coverage counts for disjoint exons 1 and 2, that is <span class="math inline">\(19 + 7 = 26\)</span>. Some methods might assume that double counting of the shared base-pairs was performed while others assume or recommend the opposite.</p>
<div id="scaling-coverage-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#scaling-coverage-counts" class="anchor"></a>Scaling coverage counts</h3>
<p>The coverage counts described previously are the ones actually included in the rse objects in recount2 instead of typical read count matrices. This is an important difference to keep in mind as most methods were developed for read count matrices. Part of the sample metadata available from recount2 includes the read length and number of mapped reads. Given a target library size (40 million reads by default), the coverage counts in recount2 can be scaled to read counts for a given library size as shown in Equation @ref(eq:scale). Note that the resulting scaled read counts are not necessarily integers so it might be necessary to round them if a differential expression (DE) method assumes integer data.</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{Read Length}} * \frac{\text{target}}{\text{mapped}} = \text{scaled read counts}
  (\#eq:scale)
\end{equation}\]</span></p>
<p>From Figure @ref(fig:Figure4) we know that Rail-RNA soft clipped some reads, so a more precise measure than the denominator of Equation @ref(eq:scale) is the area under coverage (AUC) which is the sum of the coverage for all base-pairs of the genome, regardless of the annotation as shown in Figure @ref(fig:Figure9). Without soft clipping reads, the AUC would be equal to the number of reads mapped multiplied by the read length. So for our example gene, the scaled counts for a library size of 20 reads would be <span class="math inline">\(\frac{36}{45} * 20 = 16\)</span> and in general calculated with Equation @ref(eq:scale2). The following code shows how to compute the AUC given a set of aligned reads and reproduce a portion of Figure @ref(fig:Figure9).</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{AUC}} * \text{target} = \text{scaled read counts}
  (\#eq:scale2)
\end{equation}\]</span></p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Take the example and translate it to R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="no">reads</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="st">"seq"</span>, <span class="fu">IRanges</span>(
    <span class="kw">start</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">13</span>, <span class="fl">14</span>),
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">1</span>)
    ), <span class="kw">width</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span>),
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>)
    )
))
<span class="co">## Get the base-level genome coverage curve</span>
<span class="no">cov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/coverage-methods.html">coverage</a></span>(<span class="no">reads</span>)$<span class="no">seq</span>)

<span class="co">## AUC</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">cov</span>)</pre></body></html></div>
<pre><code>## [1] 45</code></pre>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">## Code for reproducing the bottom portion of Figure 8.</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"base_pair_coverage.pdf"</span>, <span class="kw">width</span> <span class="kw">=</span> <span class="fl">20</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">4</span>, <span class="fl">2</span>) + <span class="fl">0.1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">cov</span>,
    <span class="kw">type</span> <span class="kw">=</span> <span class="st">"o"</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"violetred1"</span>, <span class="kw">lwd</span> <span class="kw">=</span> <span class="fl">10</span>, <span class="kw">ylim</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">5</span>),
    <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"Genome"</span>, <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Coverage"</span>, <span class="kw">cex.axis</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">cex.lab</span> <span class="kw">=</span> <span class="fl">3</span>,
    <span class="kw">bty</span> <span class="kw">=</span> <span class="st">"n"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/polygon.html">polygon</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">cov</span>)), <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">cov</span>)), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="no">cov</span>, <span class="fl">0</span>),
    <span class="kw">border</span> <span class="kw">=</span> <span class="fl">NA</span>, <span class="kw">density</span> <span class="kw">=</span> -<span class="fl">1</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"light blue"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">cov</span>)), <span class="no">cov</span>,
    <span class="kw">col</span> <span class="kw">=</span> <span class="st">"violetred1"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"o"</span>,
    <span class="kw">lwd</span> <span class="kw">=</span> <span class="fl">10</span>
)
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="Figure9.png" alt="Area under coverage (AUC). The area under coverage is the sum of the base-pair coverage for all positions in the genome regardless of the annotation. It is the area under the base-level coverage curve shown as the light blue area under the pink curve." width="100%"><p class="caption">
Area under coverage (AUC). The area under coverage is the sum of the base-pair coverage for all positions in the genome regardless of the annotation. It is the area under the base-level coverage curve shown as the light blue area under the pink curve.
</p>
</div>
<p>The <code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code> computes the scaled read counts for a target library size of 40 million reads and we highly recommend using it before doing other analyses. The following code shows how to use <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code> and that the resulting read counts per sample can be lower than the target size (40 million). This happens when not all mapped reads overlap known exonic base-pairs of the genome. In our example, the gene has a scaled count of 16 reads for a library size of 20 reads, meaning that 4 reads did not overlap exonic sequences.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## Check that the number of reads is less than or equal to 40 million</span>
<span class="co">## after scaling.</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)
<span class="no">rse_scaled</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="no">rse_gene_SRP009615</span>, <span class="kw">round</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(<span class="fu">assays</span>(<span class="no">rse_scaled</span>)$<span class="no">counts</span>)) / <span class="fl">1e6</span></pre></body></html></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   22.62   29.97   34.00   31.96   34.86   36.78</code></pre>
</div>
<div id="enriching-the-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#enriching-the-annotation" class="anchor"></a>Enriching the annotation</h3>
<p>Data in recount2 can be used for annotation-agnostic analyses and enriching the known annotation. Just like exon and gene coverage count matrices, recount2 provides exon-exon junction count matrices. These matrices can be used to identify new isoforms (Figure @ref(fig:Figure10)) or identify differentially expressed isoforms. For example, exon-exon junctions 2, 5 and 6 in Figure @ref(fig:Figure1) are only present in one annotated isoform. <code>Snaptron</code> <span class="citation">(10)</span> allows programatic and high-level queries of the exon-exon junction information and its graphical user interface is specially useful for visualizing this data. Inside R, the <code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/snaptron_query.html">snaptron_query()</a></code> can be used for searching specific exon-exon junctions in recount2.</p>
<div class="figure" style="text-align: center">
<img src="Figure10.png" alt="Exon-exon junctions go beyond the annotation. Reads spanning exon-exon junctions are highlighted and compared against the annotation. Three of them match the annotated junctions, but one (blue and orange read) spans an unannotated exon-exon junction with the left end matching the annotation and the right end hinting at a possible new isoform for this gene (blue and orange isoform)." width="100%"><p class="caption">
Exon-exon junctions go beyond the annotation. Reads spanning exon-exon junctions are highlighted and compared against the annotation. Three of them match the annotated junctions, but one (blue and orange read) spans an unannotated exon-exon junction with the left end matching the annotation and the right end hinting at a possible new isoform for this gene (blue and orange isoform).
</p>
</div>
<p>The base-pair coverage data from recount2 can be used together with <code>derfinder</code> <span class="citation">(8)</span> to identify expressed regions of the genome, providing another annotation-agnostic analysis of the expression data. Using the function <code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions()</a></code> we can identify regions of expression based on a given data set in recount2. These regions might overlap known exons but can also provide information about intron retention events (Figure @ref(fig:Figure11)), improve detection of exon boundaries (Figure @ref(fig:Figure12)), and help identify new exons (Fig @ref(fig:Figure1)) or expressed sequences in intergenic regions. Using <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> we can compute a coverage matrix based on the expressed regions or another set of genomic intervals. The resulting matrix can then be used for a DE analysis, just like the exon, gene and exon-exon junction matrices.</p>
<div class="figure" style="text-align: center">
<img src="Figure11.png" alt="Intron retention events. Some reads might align with known intronic segments of the genome and provide information for exploring intron retention events (pink read). Some might support an intron retention event or a new isoform when coupled with exon-exon junction data (orange read)." width="100%"><p class="caption">
Intron retention events. Some reads might align with known intronic segments of the genome and provide information for exploring intron retention events (pink read). Some might support an intron retention event or a new isoform when coupled with exon-exon junction data (orange read).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="Figure12.png" alt="Exon boundaries. Reads that go beyond the known exon boundaries can inform us of whether the annotated boundaries are correct or if there was a run-off transcription event." width="100%"><p class="caption">
Exon boundaries. Reads that go beyond the known exon boundaries can inform us of whether the annotated boundaries are correct or if there was a run-off transcription event.
</p>
</div>
</div>
</div>
<div id="gene-level-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-level-analysis" class="anchor"></a>Gene-level analysis</h2>
<p>Having reviewed how the coverage counts in recount2 were produced, we can now do a DE analysis. We will use data from 72 individuals spanning the human lifespan, split into 6 age groups with SRA accession SRP045638 <span class="citation">(13)</span>. The function <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> requires a SRA accession which can be found using <code><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search()</a></code>. <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> can then be used to download the gene coverage count data as well as other expression features. The files are saved in a directory named after the SRA accession, in this case SRP045638.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"recount"</span>)

<span class="co">## Find the project ID by searching abstracts of studies</span>
<span class="fu"><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search</a></span>(<span class="st">"human brain development by age"</span>)</pre></body></html></div>
<pre><code>##      number_samples species
## 1296             72   human
##                                                                             abstract
## 1296 RNAseq data of 36 samples across human brain development by age group from LIBD
##        project
## 1296 SRP045638</code></pre>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co">## Download the data if it is not there</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span>))) {
    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span>(<span class="st">"SRP045638"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"rse-gene"</span>)
}</pre></body></html></div>
<pre><code>## 2020-05-27 06:31:50 downloading file rse_gene.Rdata to SRP045638</code></pre>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="co">## Check that the file was downloaded</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"rse_gene.Rdata"</span>))</pre></body></html></div>
<p>The coverage count matrices are provided as <em>RangedSummarizedExperiment</em> objects (rse) <span class="citation">(9)</span>. These objects store information at the feature level, the samples and the actual count matrix as shown in Figure 1 of Love et al., 2016 <span class="citation">(3)</span>. Figure @ref(fig:Figure2) shows the actual rse objects provided by recount2 and how to access the different portions of the data. Using a unique sample ID such as the SRA Run ID it is possible to expand the sample metadata. This can be done using the predicted phenotype provided by <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code> <span class="citation">(14)</span>, pulling information from GEO via <code><a href="https://rdrr.io/pkg/recount/man/find_geo.html">find_geo()</a></code> and <code><a href="https://rdrr.io/pkg/recount/man/geo_characteristics.html">geo_characteristics()</a></code>, or with custom code.</p>
<div id="metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#metadata" class="anchor"></a>Metadata</h3>
<p>Using the <code>colData()</code> function we can access sample metadata. More information on these metadata is provided in the supplementary material of the recount2 paper <span class="citation">(7)</span>, and we provide a brief review here. The rse objects for SRA data sets include 21 columns with mostly technical information. The GTEx and TCGA rse objects include additional metadata as available from the raw sources. In particular, we compiled metadata for GTEx using the v6 phenotype information available at <a href="gtexportal.org">gtexportal.org</a>, and we put together a large table of TCGA case and sample information by combining information accumulated across <a href="http://www.cancergenomicscloud.org/">Seven Bridges’ Cancer Genomics Cloud</a> and <code>TCGAbiolinks</code> <span class="citation">(15)</span>.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="co">## One row per sample, one column per phenotype variable</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))</pre></body></html></div>
<pre><code>## [1] 72 21</code></pre>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="co">## Mostly technical variables are included</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))</pre></body></html></div>
<pre><code>##  [1] "project"                                       
##  [2] "sample"                                        
##  [3] "experiment"                                    
##  [4] "run"                                           
##  [5] "read_count_as_reported_by_sra"                 
##  [6] "reads_downloaded"                              
##  [7] "proportion_of_reads_reported_by_sra_downloaded"
##  [8] "paired_end"                                    
##  [9] "sra_misreported_paired_end"                    
## [10] "mapped_read_count"                             
## [11] "auc"                                           
## [12] "sharq_beta_tissue"                             
## [13] "sharq_beta_cell_type"                          
## [14] "biosample_submission_date"                     
## [15] "biosample_publication_date"                    
## [16] "biosample_update_date"                         
## [17] "avg_read_length"                               
## [18] "geo_accession"                                 
## [19] "bigwig_file"                                   
## [20] "title"                                         
## [21] "characteristics"</code></pre>
<div id="technical-variables" class="section level4">
<h4 class="hasAnchor">
<a href="#technical-variables" class="anchor"></a>Technical variables</h4>
<p>Several of these technical variables include the number of reads as reported by SRA, the actual number of reads Rail-RNA was able to download (which might be lower in some cases), the number of reads mapped by Rail-RNA, whether the sample is paired-end or not, the coverage AUC and the average read length (times 2 for paired-end samples). Note that the sample with SRA Run ID SRR2071341 has about 240.8 million reads as reported by SRA, while it has 120.4 million spots reported in <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341</a>; that is because it is a paired-end sample (2 reads per spot). These details are important for those interested in writing alternative scaling functions to <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code>.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="co">## Input reads: number reported by SRA might be larger than number</span>
<span class="co">## of reads Rail-RNA downloaded</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)[
    ,
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"read_count_as_reported_by_sra"</span>, <span class="st">"reads_downloaded"</span>)
]</pre></body></html></div>
<pre><code>## DataFrame with 72 rows and 2 columns
##            read_count_as_reported_by_sra reads_downloaded
##                                &lt;integer&gt;        &lt;integer&gt;
## SRR2071341                     240797206        240797206
## SRR2071345                      82266652         82266652
## SRR2071346                     132911310        132911310
## SRR2071347                      74051302         74051302
## SRR2071348                     250259914        250259914
## ...                                  ...              ...
## SRR1554541                     186250218        162403466
## SRR1554554                     140038024        121793680
## SRR1554535                     106244496         91185969
## SRR1554558                     200687480        170754145
## SRR1554553                      90579486         51803404</code></pre>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(
    <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">proportion_of_reads_reported_by_sra_downloaded</span>
)</pre></body></html></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.5719  0.9165  0.9788  0.9532  1.0000  1.0000</code></pre>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="co">## AUC information used by scale_counts() by default</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">auc</span>)</pre></body></html></div>
<pre><code>## [1] 22950214241  7553726235 12018044330  7041243857 24062460144 45169026301</code></pre>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co">## Alternatively, scale_scounts() can use the number of mapped reads</span>
<span class="co">## and other information</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"mapped_read_count"</span>, <span class="st">"paired_end"</span>,
    <span class="st">"avg_read_length"</span>
)]</pre></body></html></div>
<pre><code>## DataFrame with 72 rows and 3 columns
##            mapped_read_count paired_end avg_read_length
##                    &lt;integer&gt;  &lt;logical&gt;       &lt;integer&gt;
## SRR2071341         232970536       TRUE             200
## SRR2071345          78431778       TRUE             200
## SRR2071346         124493632       TRUE             200
## SRR2071347          71742875       TRUE             200
## SRR2071348         242992735       TRUE             200
## ...                      ...        ...             ...
## SRR1554541         162329325       TRUE             174
## SRR1554554         121738246       TRUE             173
## SRR1554535          91120421       TRUE             171
## SRR1554558         170648458       TRUE             170
## SRR1554553          51684462       TRUE             114</code></pre>
</div>
<div id="biological-information" class="section level4">
<h4 class="hasAnchor">
<a href="#biological-information" class="anchor"></a>Biological information</h4>
<p>Other metadata variables included provide more biological information, such as the <a href="http://www.cs.cmu.edu/~ckingsf/sharq/">SHARQ beta</a> tissue and cell type predictions, which are based on processing the abstract of papers. This information is available for some of the SRA projects.</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="co">## SHARQ tissue predictions: not present for all studies</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">sharq_beta_tissue</span>)</pre></body></html></div>
<pre><code>## [1] NA NA NA NA NA NA</code></pre>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">sharq_beta_tissue</span>)</pre></body></html></div>
<pre><code>## [1] "blood" "blood" "blood" "blood" "blood" "blood"</code></pre>
<p>For some data sets we were able to find the GEO accession IDs, which we then used to create the <code>title</code> and <code>characteristics</code> variables. If present, the <code>characteristics</code> information can be used to create additional metadata variables by parsing the <code>CharacterList</code> in which it is stored. Since the input is free text, sometimes more than one type of wording is used to describe the same information, meaning that we might have to process that information in order to build a more convenient variable, such as a factor vector.</p>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="co">## GEO information was absent for the SRP045638 data set</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"geo_accession"</span>, <span class="st">"title"</span>, <span class="st">"characteristics"</span>)]</pre></body></html></div>
<pre><code>## DataFrame with 72 rows and 3 columns
##            geo_accession       title characteristics
##              &lt;character&gt; &lt;character&gt; &lt;CharacterList&gt;
## SRR2071341            NA          NA              NA
## SRR2071345            NA          NA              NA
## SRR2071346            NA          NA              NA
## SRR2071347            NA          NA              NA
## SRR2071348            NA          NA              NA
## ...                  ...         ...             ...
## SRR1554541            NA          NA              NA
## SRR1554554            NA          NA              NA
## SRR1554535            NA          NA              NA
## SRR1554558            NA          NA              NA
## SRR1554553            NA          NA              NA</code></pre>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="co">## GEO information for the SRP009615 data set</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">geo_accession</span>)</pre></body></html></div>
<pre><code>## [1] "GSM836270" "GSM836271" "GSM836272" "GSM836273" "GSM847561" "GSM847562"</code></pre>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">title</span>, <span class="fl">2</span>)</pre></body></html></div>
<pre><code>## [1] "K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep1." 
## [2] "K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep1."</code></pre>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">characteristics</span>, <span class="fl">2</span>)</pre></body></html></div>
<pre><code>## CharacterList of length 2
## [[1]] cells: K562 shRNA expression: no treatment: Puromycin
## [[2]] cells: K562 shRNA expression: yes, targeting SRF treatment: Puromycin, doxycycline</code></pre>
<div class="sourceCode" id="cb43"><html><body><pre class="r"><span class="co">## Similar but not exactly the same wording used for two different samples</span>
<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">characteristics</span><span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<pre><code>## [1] "cells: K562"          "shRNA expression: no" "treatment: Puromycin"</code></pre>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">characteristics</span><span class="kw">[[</span><span class="fl">11</span>]]</pre></body></html></div>
<pre><code>## [1] "cell line: K562"                      
## [2] "shRNA expression: no shRNA expression"
## [3] "treatment: Puromycin"</code></pre>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="co">## Extract the target information</span>
<span class="no">target</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">characteristics</span>, <span class="st">"["</span>, <span class="fl">2</span>)
<span class="no">target</span></pre></body></html></div>
<pre><code>##  [1] "shRNA expression: no"                             
##  [2] "shRNA expression: yes, targeting SRF"             
##  [3] "shRNA expression: no"                             
##  [4] "shRNA expression: yes targeting SRF"              
##  [5] "shRNA expression: no shRNA expression"            
##  [6] "shRNA expression: expressing shRNA targeting EGR1"
##  [7] "shRNA expression: no shRNA expression"            
##  [8] "shRNA expression: expressing shRNA targeting EGR1"
##  [9] "shRNA expression: no shRNA expression"            
## [10] "shRNA expression: expressing shRNA targeting ATF3"
## [11] "shRNA expression: no shRNA expression"            
## [12] "shRNA expression: expressing shRNA targeting ATF3"</code></pre>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="co">## Build a useful factor vector, set the reference level and append the result</span>
<span class="co">## to the colData() slot</span>
<span class="no">target_factor</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">target</span>, <span class="st">"targeting "</span>), <span class="st">"["</span>, <span class="fl">2</span>)
<span class="no">target_factor</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">target_factor</span>)] <span class="kw">&lt;-</span> <span class="st">"none"</span>
<span class="no">target_factor</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">target_factor</span>)
<span class="no">target_factor</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span>(<span class="no">target_factor</span>, <span class="st">"none"</span>)
<span class="no">target_factor</span></pre></body></html></div>
<pre><code>##  [1] none SRF  none SRF  none EGR1 none EGR1 none ATF3 none ATF3
## Levels: none ATF3 EGR1 SRF</code></pre>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="fu">colData</span>(<span class="no">rse_gene_SRP009615</span>)$<span class="no">target_factor</span> <span class="kw">&lt;-</span> <span class="no">target_factor</span></pre></body></html></div>
<p>As shown in Figure @ref(fig:Figure2), we can expand the biological metadata information by adding predictions based on RNA-seq data <span class="citation">(14)</span>. The predictions include information about sex, sample source (cell line vs tissue), tissue and the sequencing strategy used. To add the predictions, simply use the function <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code> to expand the <code>colData()</code> slot.</p>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="co">## Before adding predictions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))</pre></body></html></div>
<pre><code>## [1] 72 21</code></pre>
<div class="sourceCode" id="cb54"><html><body><pre class="r"><span class="co">## Add the predictions</span>
<span class="no">rse_gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions</a></span>(<span class="no">rse_gene</span>)</pre></body></html></div>
<pre><code>## 2020-05-27 06:31:53 downloading the predictions to /tmp/Rtmpj4G2Ve/PredictedPhenotypes_v0.0.06.rda</code></pre>
<pre><code>## Loading objects:
##   PredictedPhenotypes</code></pre>
<div class="sourceCode" id="cb57"><html><body><pre class="r"><span class="co">## After adding the predictions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))</pre></body></html></div>
<pre><code>## [1] 72 33</code></pre>
<div class="sourceCode" id="cb59"><html><body><pre class="r"><span class="co">## Explore the variables</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)[, <span class="fl">22</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))]</pre></body></html></div>
<pre><code>## DataFrame with 72 rows and 12 columns
##            reported_sex predicted_sex accuracy_sex reported_samplesource
##                &lt;factor&gt;      &lt;factor&gt;    &lt;numeric&gt;              &lt;factor&gt;
## SRR2071341       female        female     0.862637                    NA
## SRR2071345       male          male       0.862637                    NA
## SRR2071346       male          male       0.862637                    NA
## SRR2071347       female        female     0.862637                    NA
## SRR2071348       female        female     0.862637                    NA
## ...                 ...           ...          ...                   ...
## SRR1554541       male          male       0.862637                    NA
## SRR1554554       female        female     0.862637                    NA
## SRR1554535       male          male       0.862637                    NA
## SRR1554558       female        female     0.862637                    NA
## SRR1554553       male          male       0.862637                    NA
##            predicted_samplesource accuracy_samplesource reported_tissue
##                          &lt;factor&gt;             &lt;numeric&gt;        &lt;factor&gt;
## SRR2071341                 tissue                    NA              NA
## SRR2071345                 tissue               0.89235              NA
## SRR2071346                 tissue                    NA              NA
## SRR2071347                 tissue                    NA              NA
## SRR2071348                 tissue                    NA              NA
## ...                           ...                   ...             ...
## SRR1554541                 tissue                    NA              NA
## SRR1554554                 tissue                    NA              NA
## SRR1554535                 tissue                    NA              NA
## SRR1554558                 tissue                    NA              NA
## SRR1554553                 tissue               0.89235              NA
##            predicted_tissue accuracy_tissue reported_sequencingstrategy
##                    &lt;factor&gt;       &lt;numeric&gt;                    &lt;factor&gt;
## SRR2071341            Brain        0.518825                      PAIRED
## SRR2071345            Brain        0.518825                      PAIRED
## SRR2071346            Brain        0.518825                      PAIRED
## SRR2071347            Brain        0.518825                      PAIRED
## SRR2071348            Brain        0.518825                      PAIRED
## ...                     ...             ...                         ...
## SRR1554541            Brain        0.518825                      PAIRED
## SRR1554554            Brain        0.518825                      PAIRED
## SRR1554535            Brain        0.518825                      PAIRED
## SRR1554558            Brain        0.518825                      PAIRED
## SRR1554553            Brain        0.518825                      PAIRED
##            predicted_sequencingstrategy accuracy_sequencingstrategy
##                                &lt;factor&gt;                   &lt;numeric&gt;
## SRR2071341                       PAIRED                    0.908575
## SRR2071345                       PAIRED                    0.908575
## SRR2071346                       PAIRED                    0.908575
## SRR2071347                       PAIRED                    0.908575
## SRR2071348                       PAIRED                    0.908575
## ...                                 ...                         ...
## SRR1554541                       PAIRED                    0.908575
## SRR1554554                       PAIRED                    0.908575
## SRR1554535                       PAIRED                    0.908575
## SRR1554558                       PAIRED                    0.908575
## SRR1554553                       PAIRED                    0.908575</code></pre>
</div>
<div id="adding-more-information" class="section level4">
<h4 class="hasAnchor">
<a href="#adding-more-information" class="anchor"></a>Adding more information</h4>
<p>Ultimately, more sample metadata information could be available elsewhere, which can be useful for analyses. This information might be provided in the paper describing the data, the SRA Run Selector or other sources. As shown in Figure @ref(fig:Figure2), it is possible to append information to the <code>colData()</code> slot as long as there is a unique sample identifier such as the SRA Run ID.</p>
<p>For our example use case, project SRP045638 has a few extra biologically relevant variables via the SRA Run selector <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638</a>. We can download that information into text file named <code>SraRunTable.txt</code> by default, then load it into R, sort it appropriately and then append it to the <code>colData()</code> slot. Below we do so for the SRP045638 project.</p>
<div class="sourceCode" id="cb61"><html><body><pre class="r"><span class="co">## Save the information from</span>
<span class="co">## https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP045638</span>
<span class="co">## to a table. We saved the file as SRP045638/SraRunTable.txt.</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"SraRunTable.txt"</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb63"><html><body><pre class="r"><span class="co">## Read the table</span>
<span class="no">sra</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"SraRunTable.txt"</span>),
    <span class="kw">header</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)

<span class="co">## Explore it</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sra</span>)</pre></body></html></div>
<pre><code>##          Run     Age alignment_software..exp. Assay.Type AvgSpotLen       Bases
## 1 SRR1554533 67.7800                  TopHat2    RNA-Seq        179  6766378000
## 2 SRR1554534 40.4200                  TopHat2    RNA-Seq        179  6357169900
## 3 SRR1554535 41.5800                  TopHat2    RNA-Seq        171  9118596900
## 4 SRR1554536 44.1700                  TopHat2    RNA-Seq        184  4697126700
## 5 SRR1554537 -0.3836                  TopHat2    RNA-Seq        182 12199232600
## 6 SRR1554538 -0.4027                  TopHat2    RNA-Seq        185 14733333600
##   Biomaterial_provider  BioProject    BioSample BioSampleModel      Bytes
## 1                 LIBD PRJNA245228 SAMN02731372          Human 3744853436
## 2                 LIBD PRJNA245228 SAMN02731373          Human 3019096844
## 3                 LIBD PRJNA245228 SAMN02999518          Human 5204886934
## 4                 LIBD PRJNA245228 SAMN02999519          Human 2771744419
## 5                 LIBD PRJNA245228 SAMN02999520          Human 6485501388
## 6                 LIBD PRJNA245228 SAMN02999521          Human 7505295858
##                              Center.Name Consent DATASTORE.filetype
## 1 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
## 2 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
## 3 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
## 4 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
## 5 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
## 6 LIEBER INSTITUTE FOR BRAIN DEVELOPMENT  public                sra
##   DATASTORE.provider               DATASTORE.region disease Experiment fraction
## 1         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 Control  SRX683791    total
## 2         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 Control  SRX683792    total
## 3         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683793    total
## 4         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683794    total
## 5         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683795    total
## 6         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 control  SRX683796    total
##            Instrument isolate                   Library.Name LibraryLayout
## 1 Illumina HiSeq 2000   DLPFC R2835_DLPFC_polyA_RNAseq_total        PAIRED
## 2 Illumina HiSeq 2000   DLPFC R2857_DLPFC_polyA_RNAseq_total        PAIRED
## 3 Illumina HiSeq 2000   R2869 R2869_DLPFC_polyA_RNAseq_total        PAIRED
## 4 Illumina HiSeq 2000   R3098 R3098_DLPFC_polyA_RNAseq_total        PAIRED
## 5 Illumina HiSeq 2000   R3452 R3452_DLPFC_polyA_RNAseq_total        PAIRED
## 6 Illumina HiSeq 2000   R3462 R3462_DLPFC_polyA_RNAseq_total        PAIRED
##   LibrarySelection  LibrarySource     Organism Platform race
## 1             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
## 2             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
## 3             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
## 4             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
## 5             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
## 6             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA   AA
##            ReleaseDate rin Sample.Name    sex SRA.Study tissue     AssemblyName
## 1 2014-11-13T00:00:00Z 8.3 R2835_DLPFC female SRP045638  DLPFC GCA_000001405.13
## 2 2014-11-13T00:00:00Z 8.4 R2857_DLPFC   male SRP045638  DLPFC GCA_000001405.13
## 3 2014-11-13T00:00:00Z 8.7 R2869_DLPFC   male SRP045638  DLPFC GCA_000001405.13
## 4 2014-11-13T00:00:00Z 5.3 R3098_DLPFC female SRP045638  DLPFC GCA_000001405.13
## 5 2014-11-13T00:00:00Z 9.6 R3452_DLPFC female SRP045638  DLPFC GCA_000001405.13
## 6 2014-11-13T00:00:00Z 6.4 R3462_DLPFC female SRP045638  DLPFC GCA_000001405.13
##   dev_stage
## 1     Fetal
## 2          
## 3          
## 4          
## 5          
## 6</code></pre>
<div class="sourceCode" id="cb65"><html><body><pre class="r"><span class="co">## Set all column names in lower case</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sra</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sra</span>))

<span class="co">## Choose some variables we want to add</span>
<span class="no">sra_vars</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"sex"</span>, <span class="st">"race"</span>, <span class="st">"rin"</span>, <span class="st">"age"</span>, <span class="st">"isolate"</span>, <span class="st">"disease"</span>,
    <span class="st">"tissue"</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="no">sra_vars</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sra</span>)))

<span class="co">## Re-organize the SRA table based on the SRA Run IDs we have</span>
<span class="no">sra</span> <span class="kw">&lt;-</span> <span class="no">sra</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">run</span>, <span class="no">sra</span>$<span class="no">run</span>), ]

<span class="co">## Double check the order</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">run</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">sra</span>$<span class="no">run</span>)))

<span class="co">## Append the variables of interest</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>), <span class="no">sra</span>[, <span class="no">sra_vars</span>])

<span class="co">## Final dimensions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))</pre></body></html></div>
<pre><code>## [1] 72 40</code></pre>
<div class="sourceCode" id="cb67"><html><body><pre class="r"><span class="co">## Explore result</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)[, <span class="fl">34</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>))]</pre></body></html></div>
<pre><code>## DataFrame with 72 rows and 7 columns
##                    sex        race       rin       age     isolate     disease
##            &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;
## SRR2071341      female          AA       8.3   67.7800       DLPFC     Control
## SRR2071345        male          AA       8.4   40.4200       DLPFC     Control
## SRR2071346        male          AA       8.7   41.5800       R2869     control
## SRR2071347      female          AA       5.3   44.1700       R3098     control
## SRR2071348      female          AA       9.6   -0.3836       R3452     control
## ...                ...         ...       ...       ...         ...         ...
## SRR1554541        male          AA       5.7   -0.3836       R3485     control
## SRR1554554      female          AA       8.1    0.3041       R3669     control
## SRR1554535        male          AA       8.7   41.5800       R2869     control
## SRR1554558      female        CAUC       9.1   16.7000       R4028     control
## SRR1554553        male        CAUC       8.4    0.3918       R3652     control
##                 tissue
##            &lt;character&gt;
## SRR2071341       DLPFC
## SRR2071345       DLPFC
## SRR2071346       DLPFC
## SRR2071347       DLPFC
## SRR2071348       DLPFC
## ...                ...
## SRR1554541       DLPFC
## SRR1554554       DLPFC
## SRR1554535       DLPFC
## SRR1554558       DLPFC
## SRR1554553       DLPFC</code></pre>
<p>Since we have the predicted sex as well as the reported sex via the SRA Run Selector, we can check whether they match.</p>
<div class="sourceCode" id="cb69"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(
    <span class="st">"Predicted"</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">predicted_sex</span>,
    <span class="st">"Observed"</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">sex</span>
)</pre></body></html></div>
<pre><code>##             Observed
## Predicted    female male
##   female         24    4
##   male            0   44
##   Unassigned      0    0</code></pre>
</div>
</div>
<div id="de-setup" class="section level3">
<h3 class="hasAnchor">
<a href="#de-setup" class="anchor"></a>DE setup</h3>
<p>Now that we have all the metadata available we can perform a DE analysis. The original study for project SRP045638 <span class="citation">(13)</span> looked at differences between 6 age groups: prenatal, infant, child, teen, adult and late life. The following code creates these six age groups.</p>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="co">## Create the original 6 age groups</span>
<span class="no">age_bins</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">age</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(-<span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">50</span>, <span class="fl">Inf</span>),
    <span class="kw">include.lowest</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">age_bins</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"prenatal"</span>, <span class="st">"infant"</span>, <span class="st">"child"</span>, <span class="st">"teen"</span>, <span class="st">"adult"</span>,
    <span class="st">"late life"</span>
)
<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">age_group</span> <span class="kw">&lt;-</span> <span class="no">age_bins</span></pre></body></html></div>
<p>Most of the DE signal from the original study was between the prenatal and postnatal samples. To simplify the analysis, we will focus on this comparison.</p>
<div class="sourceCode" id="cb72"><html><body><pre class="r"><span class="co">## Create prenatal factor</span>
<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">prenatal</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene</span>)$<span class="no">age_group</span> <span class="kw">==</span> <span class="st">"prenatal"</span>, <span class="st">"prenatal"</span>,
        <span class="st">"postnatal"</span>
    ),
    <span class="kw">levels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"prenatal"</span>, <span class="st">"postnatal"</span>)
)</pre></body></html></div>
<p>As we saw earlier in Figure @ref(fig:Figure9), it is important to scale the coverage counts to read counts. To highlight the fact that we scaled the counts, we will use a new object name and delete the previous one. However, in practice we would simply overwrite the <code>rse</code> object with the output of <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts(rse)</a></code>.</p>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="co">## Scale counts</span>
<span class="no">rse_gene_scaled</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="no">rse_gene</span>)

<span class="co">## To highlight that we scaled the counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">rse_gene</span>)</pre></body></html></div>
<p>Having scaled the counts, we then filter out genes that are lowly expressed and extract the count matrix.</p>
<div class="sourceCode" id="cb74"><html><body><pre class="r"><span class="co">## Extract counts and filter out lowly expressed geens</span>
<span class="no">counts</span> <span class="kw">&lt;-</span> <span class="fu">assays</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">counts</span>
<span class="no">filter</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">counts</span>) <span class="kw">&gt;</span> <span class="fl">0.5</span></pre></body></html></div>
</div>
<div id="de-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#de-analysis" class="anchor"></a>DE analysis</h3>
<p>Now that we have scaled the counts, there are multiple DE packages we could use, as described elsewhere <span class="citation">(2,3)</span>. Since we have 12 samples per group, which is a moderate number, we will use <code>limma-voom</code> <span class="citation">(16)</span> due to its speed. The model we use tests for DE between prenatal and postnatal samples adjusting for sex and RIN, which is a measure of quality of the input sample. We check the data with multi-dimensional scaling plots (Figures @ref(fig:limmade1) and @ref(fig:limmade2)) as well as the mean-variance plot (Figure @ref(fig:limmade3)). In a real use case we might have to explore the results with different models and perform sensitivity analyses.</p>
<div class="sourceCode" id="cb75"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"limma"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"edgeR"</span>)

<span class="co">## Build DGEList object</span>
<span class="no">dge</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="no">counts</span>[<span class="no">filter</span>, ])

<span class="co">## Calculate normalization factors</span>
<span class="no">dge</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span>(<span class="no">dge</span>)

<span class="co">## Explore the data</span>
<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="no">dge</span>, <span class="kw">labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">prenatal</span>, <span class="fl">1</span>, <span class="fl">2</span>))</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="limmade1-1.png" alt="Multi-dimensional scaling plot of the gene-level data by age group." width="100%"><p class="caption">
Multi-dimensional scaling plot of the gene-level data by age group.
</p>
</div>
<div class="sourceCode" id="cb76"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="no">dge</span>, <span class="kw">labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">sex</span>, <span class="fl">1</span>, <span class="fl">1</span>))</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="limmade2-1.png" alt="Multi-dimensional scaling plot of the gene-level data by sex." width="100%"><p class="caption">
Multi-dimensional scaling plot of the gene-level data by sex.
</p>
</div>
<div class="sourceCode" id="cb77"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/tapply.html">tapply</a></span>(
    <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">rin</span>, <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">prenatal</span>,
    <span class="no">summary</span>
)</pre></body></html></div>
<pre><code>## $prenatal
##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   5.700   6.400   8.150   7.767   8.600   9.600 
## 
## $postnatal
##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   5.300   8.100   8.300   8.197   8.700   9.100</code></pre>
<div class="sourceCode" id="cb79"><html><body><pre class="r"><span class="co">## Specify our design matrix</span>
<span class="no">design</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(
    <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>),
    <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">sex</span> + <span class="no">rin</span> + <span class="no">prenatal</span>)
)</pre></body></html></div>
<div class="sourceCode" id="cb80"><html><body><pre class="r"><span class="co">## Run voom</span>
<span class="no">v</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html">voom</a></span>(<span class="no">dge</span>, <span class="no">design</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="limmade3-1.png" alt="voom mean-variance plot of the gene-level data." width="100%"><p class="caption">
voom mean-variance plot of the gene-level data.
</p>
</div>
<div class="sourceCode" id="cb81"><html><body><pre class="r"><span class="co">## Run remaining parts of the DE analysis</span>
<span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">v</span>, <span class="no">design</span>)
<span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit</span>)</pre></body></html></div>
<p>Having run the DE analysis, we can explore some of the top results either with an MA plot (Figure @ref(fig:limmaplots1)) and a volcano plot Figure (@ref(fig:limmaplots2)). Both reveal very strong and widespread DE signal.</p>
<div class="sourceCode" id="cb82"><html><body><pre class="r"><span class="co">## Visually explore DE results</span>
<span class="kw pkg">limma</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotma.html">plotMA</a></span>(<span class="no">fit</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">4</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="limmaplots1-1.png" alt="MA plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="100%"><p class="caption">
MA plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN.
</p>
</div>
<div class="sourceCode" id="cb83"><html><body><pre class="r"><span class="kw pkg">limma</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html">volcanoplot</a></span>(<span class="no">fit</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">4</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="limmaplots2-1.png" alt="Volcano plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="100%"><p class="caption">
Volcano plot of the gene-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN.
</p>
</div>
</div>
<div id="de-report" class="section level3">
<h3 class="hasAnchor">
<a href="#de-report" class="anchor"></a>DE report</h3>
<p>Now that we have the DE results, we can use some of the tools with the biocView <a href="http://bioconductor.org/packages/release/BiocViews.html#___ReportWriting">ReportWriting</a> to create a report. One of them is <code>regionReport</code> <span class="citation">(17)</span>, which can create reports from <code>DESeq2</code> <span class="citation">(18)</span> and <code>edgeR</code> <span class="citation">(19)</span> results. It can also handle <code>limma-voom</code> <span class="citation">(16)</span> results by making them look like <code>DESeq2</code> results. To do so, we need to extract the relevant information from the <code>limma-voom</code> objects using <code><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable()</a></code> and build DESeqDataSet and DESeqResults objects as shown below. A similar conversion is needed to use <code>ideal</code> <span class="citation">(20)</span>, which is another package in the <em>ReportWriting</em> biocView category.</p>
<div class="sourceCode" id="cb84"><html><body><pre class="r"><span class="co">## Extract data from limma-voom results</span>
<span class="no">top</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit</span>,
    <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>, <span class="kw">sort.by</span> <span class="kw">=</span> <span class="st">"none"</span>,
    <span class="kw">coef</span> <span class="kw">=</span> <span class="st">"prenatalpostnatal"</span>
)

<span class="co">## Build a DESeqDataSet with the count data and model we used</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="no">rse_gene_scaled</span>[<span class="no">filter</span>, ], ~ <span class="no">sex</span> + <span class="no">rin</span> + <span class="no">prenatal</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(rse_gene_scaled[filter, ], ~sex + rin + prenatal): some
## variables in design formula are characters, converting to factors</code></pre>
<pre><code>##   the design formula contains one or more numeric variables that have mean or
##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).
##   it is generally a good idea to center and scale numeric variables in the design
##   to improve GLM convergence.</code></pre>
<div class="sourceCode" id="cb88"><html><body><pre class="r"><span class="co">## Add gene names keeping only the Ensembl part of the Gencode IDs</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">dds</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">dds</span>))

<span class="co">## Build a DESeqResults object with the relevant information</span>
<span class="co">## Note that we are transforming the baseMean so it will look ok</span>
<span class="co">## with DESeq2's plotting functions.</span>
<span class="no">limma_res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html">DESeqResults</a></span>(<span class="fu">DataFrame</span>(
    <span class="kw">pvalue</span> <span class="kw">=</span> <span class="no">top</span>[, <span class="st">"P.Value"</span>],
    <span class="kw">log2FoldChange</span> <span class="kw">=</span> <span class="no">top</span>[, <span class="st">"logFC"</span>],
    <span class="kw">baseMean</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span>(<span class="no">top</span>[, <span class="st">"AveExpr"</span>]),
    <span class="kw">padj</span> <span class="kw">=</span> <span class="no">top</span>[, <span class="st">"adj.P.Val"</span>]
))
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">limma_res</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">dds</span>)

<span class="co">## Specify FDR cutoff to use</span>
<span class="fu">metadata</span>(<span class="no">limma_res</span>)<span class="kw">[[</span><span class="st">"alpha"</span>]] <span class="kw">&lt;-</span> <span class="fl">0.001</span>

<span class="co">## Add gene symbols so they will be displayed in the report</span>
<span class="no">limma_res</span>$<span class="no">symbol</span> <span class="kw">&lt;-</span> <span class="fu">rowRanges</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">symbol</span>[<span class="no">filter</span>]

<span class="co">## Some extra information used by the report function</span>
<span class="fu">mcols</span>(<span class="no">dds</span>) <span class="kw">&lt;-</span> <span class="no">limma_res</span>
<span class="fu">mcols</span>(<span class="fu">mcols</span>(<span class="no">dds</span>)) <span class="kw">&lt;-</span> <span class="fu">DataFrame</span>(
    <span class="kw">type</span> <span class="kw">=</span> <span class="st">"results"</span>,
    <span class="kw">description</span> <span class="kw">=</span> <span class="st">"manual incomplete conversion from limma-voom to DESeq2"</span>
)</pre></body></html></div>
<p>Having converted our <code>limma-voom</code> results to <code>DESeq2</code> results, we can now create the report, which should open automatically in a browser.</p>
<div class="sourceCode" id="cb89"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"regionReport"</span>)
<span class="co">## This takes about 20 minutes to run</span>
<span class="no">report</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="no">dds</span>,
    <span class="kw">project</span> <span class="kw">=</span> <span class="st">"SRP045638 gene results with limma-voom"</span>,
    <span class="kw">output</span> <span class="kw">=</span> <span class="st">"gene_report"</span>, <span class="kw">outdir</span> <span class="kw">=</span> <span class="st">"SRP045638"</span>,
    <span class="kw">intgroup</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"prenatal"</span>, <span class="st">"sex"</span>), <span class="kw">res</span> <span class="kw">=</span> <span class="no">limma_res</span>,
    <span class="kw">software</span> <span class="kw">=</span> <span class="st">"limma"</span>
)</pre></body></html></div>
<p>If the report doesn’t open automatically, we can open it with <code><a href="https://rdrr.io/r/utils/browseURL.html">browseURL()</a></code>. A pre-computed version is available as Supplementary File 1.</p>
<div class="sourceCode" id="cb90"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html">browseURL</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"gene_report.html"</span>))</pre></body></html></div>
</div>
<div id="go-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h3>
<p>Using <code>clusterProfiler</code> <span class="citation">(21)</span> we can then perform several enrichment analyses using the Ensembl gene IDs. Here we show how to perform an enrichment analysis using the biological process ontology (Figure @ref(fig:goanalysis)).</p>
<div class="sourceCode" id="cb91"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"clusterProfiler"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"org.Hs.eg.db"</span>)

<span class="co">## Remember that limma_res had ENSEMBL IDs for the genes</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">limma_res</span>))</pre></body></html></div>
<pre><code>## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419" "ENSG00000000457"
## [5] "ENSG00000000460" "ENSG00000000938"</code></pre>
<div class="sourceCode" id="cb93"><html><body><pre class="r"><span class="co">## Perform enrichment analysis for Biological Process (BP)</span>
<span class="co">## Note that the argument is keytype instead of keyType in Bioconductor 3.5</span>
<span class="no">enrich_go</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span>(
    <span class="kw">gene</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">limma_res</span>)[<span class="no">limma_res</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="fl">0.001</span>],
    <span class="kw">OrgDb</span> <span class="kw">=</span> <span class="no">org.Hs.eg.db</span>, <span class="kw">keyType</span> <span class="kw">=</span> <span class="st">"ENSEMBL"</span>, <span class="kw">ont</span> <span class="kw">=</span> <span class="st">"BP"</span>,
    <span class="kw">pAdjustMethod</span> <span class="kw">=</span> <span class="st">"BH"</span>, <span class="kw">pvalueCutoff</span> <span class="kw">=</span> <span class="fl">0.01</span>, <span class="kw">qvalueCutoff</span> <span class="kw">=</span> <span class="fl">0.05</span>,
    <span class="kw">universe</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">limma_res</span>)
)

<span class="co">## Visualize enrichment results</span>
<span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/reexports.html">dotplot</a></span>(<span class="no">enrich_go</span>, <span class="kw">font.size</span> <span class="kw">=</span> <span class="fl">7</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="goanalysis-1.png" alt="Biological processes enriched in the DE genes." width="100%"><p class="caption">
Biological processes enriched in the DE genes.
</p>
</div>
<p>Several other analyses can be performed with the resulting list of differentially expressed genes as described previously <span class="citation">(2,3)</span>, although that is beyond the scope of this workflow.</p>
</div>
</div>
<div id="other-features" class="section level2">
<h2 class="hasAnchor">
<a href="#other-features" class="anchor"></a>Other features</h2>
<p>As described in Figure @ref(fig:Figure1), recount2 provides data for expression features beyond genes. In this section we perform a DE analysis using exon data as well as the base-pair resolution information.</p>
<div id="exon-and-exon-exon-junctions" class="section level3">
<h3 class="hasAnchor">
<a href="#exon-and-exon-exon-junctions" class="anchor"></a>Exon and exon-exon junctions</h3>
<p>The exon and exon-exon junction coverage count matrices are similar to the gene-level one and can also be downloaded with <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code>. However, these coverage count matrices are much larger than the gene one. Aggressive filtering of lowly expressed exons or exon-exon junctions can reduce the matrix dimensions if this impacts the performance of the DE software used.</p>
<p>Below we repeat the gene-level analysis for the disjoint exon data. We first download the exon data, add the expanded metadata we constructed for the gene analysis, explore the data (Figure1 @ref(fig:exondeanalysis1)), and then perform the DE analysis using <code>limma-voom</code>.</p>
<div class="sourceCode" id="cb94"><html><body><pre class="r"><span class="co">## Download the data if it is not there</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"rse_exon.Rdata"</span>))) {
    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span>(<span class="st">"SRP045638"</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"rse-exon"</span>)
}</pre></body></html></div>
<pre><code>## 2020-05-27 06:33:32 downloading file rse_exon.Rdata to SRP045638</code></pre>
<div class="sourceCode" id="cb96"><html><body><pre class="r"><span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"rse_exon.Rdata"</span>))

<span class="co">## Scale and add the metadata (it is in the same order)</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu">colData</span>(<span class="no">rse_exon</span>)$<span class="no">run</span>, <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)$<span class="no">run</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb98"><html><body><pre class="r"><span class="fu">colData</span>(<span class="no">rse_exon</span>) <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)
<span class="no">rse_exon_scaled</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="no">rse_exon</span>)
<span class="co">## To highlight that we scaled the counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">rse_exon</span>)

<span class="co">## Filter lowly expressed exons</span>
<span class="no">filter_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="fu">assays</span>(<span class="no">rse_exon_scaled</span>)$<span class="no">counts</span>) <span class="kw">&gt;</span> <span class="fl">0.5</span>
<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">filter_exon</span>) / <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">filter_exon</span>) * <span class="fl">100</span>, <span class="fl">2</span>)</pre></body></html></div>
<pre><code>## filter_exon
## FALSE  TRUE 
## 32.76 67.24</code></pre>
<div class="sourceCode" id="cb100"><html><body><pre class="r"><span class="co">## Build DGEList object</span>
<span class="no">dge_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(
    <span class="kw">counts</span> <span class="kw">=</span> <span class="fu">assays</span>(<span class="no">rse_exon_scaled</span>)$<span class="no">counts</span>[<span class="no">filter_exon</span>, ]
)

<span class="co">## Calculate normalization factors</span>
<span class="no">dge_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span>(<span class="no">dge_exon</span>)

<span class="co">## Run voom</span>
<span class="no">v_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html">voom</a></span>(<span class="no">dge_exon</span>, <span class="no">design</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="exondeanalysis1-1.png" alt="voom mean-variance plot of the exon-level data." width="100%"><p class="caption">
voom mean-variance plot of the exon-level data.
</p>
</div>
<div class="sourceCode" id="cb101"><html><body><pre class="r"><span class="co">## Run remaining parts of the DE analysis</span>
<span class="no">fit_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">v_exon</span>, <span class="no">design</span>)
<span class="no">fit_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit_exon</span>)

<span class="co">## Visualize inspect results</span>
<span class="kw pkg">limma</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html">volcanoplot</a></span>(<span class="no">fit_exon</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">4</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="exondeanalysis2-1.png" alt="Volcano plot of the exon-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="100%"><p class="caption">
Volcano plot of the exon-level data. Testing for prenatal and postnatal DE adjusting for sex and RIN.
</p>
</div>
<div class="sourceCode" id="cb102"><html><body><pre class="r"><span class="co">## Get p-values and other statistics</span>
<span class="no">top_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit_exon</span>,
    <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>, <span class="kw">sort.by</span> <span class="kw">=</span> <span class="st">"none"</span>,
    <span class="kw">coef</span> <span class="kw">=</span> <span class="st">"prenatalpostnatal"</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">top_exon</span>$<span class="no">adj.P.Val</span> <span class="kw">&lt;</span> <span class="fl">0.001</span>)</pre></body></html></div>
<pre><code>## 
##  FALSE   TRUE 
## 192061 213482</code></pre>
<p>Just like at the gene-level, we see many exons differentially expressed between prenatal and postnatal samples (Figure @ref(fig:exondeanalysis2)). As a first step to integrate the results from the two features, we can compare the list of genes that are differentially expressed versus the genes that have at least one exon differentially expressed.</p>
<div class="sourceCode" id="cb104"><html><body><pre class="r"><span class="co">## Get the gene IDs for genes that are DE at the gene-level or that have at</span>
<span class="co">## least one exon with DE signal.</span>
<span class="no">genes_w_de_exon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">rse_exon_scaled</span>)[<span class="no">top_exon</span>$<span class="no">adj.P.Val</span> <span class="kw">&lt;</span> <span class="fl">0.001</span>]
)
<span class="no">genes_de</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">rse_gene_scaled</span>)[
    <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">filter</span>)[<span class="no">top</span>$<span class="no">adj.P.Val</span> <span class="kw">&lt;</span> <span class="fl">0.001</span>]
]

<span class="co">## Make a venn diagram</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"gplots"</span>)
<span class="no">vinfo</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gplots/man/venn.html">venn</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"genes"</span> <span class="kw">=</span> <span class="no">genes_de</span>, <span class="st">"exons"</span> <span class="kw">=</span> <span class="no">genes_w_de_exon</span>),
    <span class="kw">names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"genes"</span>, <span class="st">"exons"</span>), <span class="kw">show.plot</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">vinfo</span>) +
    <span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Genes/exons with DE signal"</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="geneexon-1.png" alt="Venn diagram of the overlap between DE genes and genes with at least one exon DE." width="100%"><p class="caption">
Venn diagram of the overlap between DE genes and genes with at least one exon DE.
</p>
</div>
<pre><code>## integer(0)</code></pre>
<p>Not all differentially expressed genes have differentially expressed exons. Moreover, genes with at least one differentially expressed exon are not necessarily differentially expressed, as shown in Figure @ref(fig:geneexon). This is in line with what was described in Figure 2B of Soneson et al., 2015 <span class="citation">(22)</span>.</p>
<p>This was just a quick example of how we can perform DE analyses at the gene and exon feature levels. We envision that more involved pipelines could be developed that leverage both feature levels, such as in Jaffe et al., 2017 <span class="citation">(23)</span>. For instance, we could focus on the differentially expressed genes with at least one differentially expressed exon, and compare the direction of the DE signal versus the gene-level signal as shown in figure @ref(fig:geneexonmatch).</p>
<div class="sourceCode" id="cb106"><html><body><pre class="r"><span class="co">## Keep only the DE exons that are from a gene that is also DE</span>
<span class="no">top_exon_de</span> <span class="kw">&lt;-</span> <span class="no">top_exon</span>[<span class="no">top_exon</span>$<span class="no">adj.P.Val</span> <span class="kw">&lt;</span> <span class="fl">0.001</span> <span class="kw">&amp;</span>
    <span class="no">top_exon</span>$<span class="no">ID</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span>(<span class="no">vinfo</span>, <span class="st">"intersections"</span>)<span class="kw">[[</span><span class="st">"genes:exons"</span>]], ]

<span class="co">## Find the fold change that is the most extreme among the DE exons of a gene</span>
<span class="no">exon_max_fc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html">tapply</a></span>(<span class="no">top_exon_de</span>$<span class="no">logFC</span>, <span class="no">top_exon_de</span>$<span class="no">ID</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="no">x</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">x</span>))]
})

<span class="co">## Keep only the DE genes that match the previous selection</span>
<span class="no">top_gene_de</span> <span class="kw">&lt;-</span> <span class="no">top</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">exon_max_fc</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">top</span>)), ]

<span class="co">## Make the plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">top_gene_de</span>$<span class="no">logFC</span>, <span class="no">exon_max_fc</span>,
    <span class="kw">pch</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/adjustcolor.html">adjustcolor</a></span>(<span class="st">"black"</span>, <span class="fl">1</span> / <span class="fl">5</span>),
    <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Most extreme exon log FC"</span>,
    <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"Gene log FC"</span>,
    <span class="kw">main</span> <span class="kw">=</span> <span class="st">"DE genes with at least one DE exon"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">a</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">b</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">h</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"grey80"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">v</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"grey80"</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="geneexonmatch-1.png" alt="Log fold change (FC) for DE genes compared against the most extreme exon log FC among exons that are DE for the given gene." width="100%"><p class="caption">
Log fold change (FC) for DE genes compared against the most extreme exon log FC among exons that are DE for the given gene.
</p>
</div>
<p>The fold change for most exons shown in Figure @ref(fig:geneexonmatch) agrees with the gene-level fold change. However, some of them have opposite directions and could be interesting to study further.</p>
</div>
<div id="base-pair-resolution" class="section level3">
<h3 class="hasAnchor">
<a href="#base-pair-resolution" class="anchor"></a>Base-pair resolution</h3>
<p>recount2 provides bigWig coverage files (unscaled) for all samples, as well as a mean bigWig coverage file per project where each sample was scaled to 40 million 100 base-pair reads. The mean bigWig files are exactly what is needed to start an <em>expressed regions</em> analysis with <code>derfinder</code> <span class="citation">(8)</span>. <code>recount</code> provides two related functions: <code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions()</a></code> which is used to define a set of regions based on the mean bigWig file for a given project, and <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> which based on a set of regions builds a count coverage matrix in a <em>RangedSummarizedExperiment</em> object just like the ones that are provided for genes and exons. Both functions ultimately use <code>import.bw()</code> from <code>rtracklayer</code> <span class="citation">(24)</span> that was historically not supported on Windows machines. While this presents a portability disadvantage, on the other side it allows reading portions of bigWig files from the web without having to fully download them. <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> with <code>type = "mean"</code> or <code>type = "samples"</code> can be used to download the bigWig files, which we recommend doing when working with them extensively.</p>
<p>For illustrative purposes, we will use the data from chromosome 21 for the SRP045638 project. First, we obtain the expressed regions using a relatively high mean cutoff of 5. We then filter the regions to keep only the ones longer than 100 base-pairs to shorten the time needed for running <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code>.</p>
<div class="sourceCode" id="cb107"><html><body><pre class="r"><span class="co">## Define expressed regions for study SRP045638, only for chromosome 21</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"chr21"</span>,
    <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">5L</span>,
    <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000L</span>
)</pre></body></html></div>
<pre><code>## 2020-05-27 06:36:08 loadCoverage: loading BigWig file http://duffel.rail.bio/recount/SRP045638/bw/mean_SRP045638.bw</code></pre>
<pre><code>## 2020-05-27 06:36:16 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-27 06:36:41 filterData: originally there were 46709983 rows, now there are 46709983 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>## 2020-05-27 06:36:41 findRegions: identifying potential segments</code></pre>
<pre><code>## 2020-05-27 06:36:41 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-27 06:36:41 .getSegmentsRle: segmenting with cutoff(s) 5</code></pre>
<pre><code>## 2020-05-27 06:36:44 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-27 06:36:45 findRegions: identifying region clusters</code></pre>
<div class="sourceCode" id="cb116"><html><body><pre class="r"><span class="co">## Explore the resulting expressed regions</span>
<span class="no">regions</span></pre></body></html></div>
<pre><code>## GRanges object with 3853 ranges and 6 metadata columns:
##        seqnames            ranges strand |     value      area indexStart
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##      1    chr21   5026549-5026630      * |   6.48181   531.509    5026549
##      2    chr21   5027935-5027961      * |   6.19690   167.316    5027935
##      3    chr21   5028108-5028225      * |   8.99329  1061.208    5028108
##      4    chr21   5032053-5032117      * |   7.06828   459.438    5032053
##      5    chr21   5032148-5032217      * |   6.48833   454.183    5032148
##    ...      ...               ...    ... .       ...       ...        ...
##   3849    chr21          46695774      * |   5.02902   5.02902   46695774
##   3850    chr21 46695784-46695843      * |   5.38047 322.82838   46695784
##   3851    chr21 46695865-46695869      * |   5.11283  25.56414   46695865
##   3852    chr21 46696463-46696486      * |   5.25689 126.16540   46696463
##   3853    chr21 46696508-46696534      * |   5.22988 141.20686   46696508
##         indexEnd cluster clusterL
##        &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##      1   5026630       1     1677
##      2   5027961       1     1677
##      3   5028225       1     1677
##      4   5032117       2     8283
##      5   5032217       2     8283
##    ...       ...     ...      ...
##   3849  46695774     708     5708
##   3850  46695843     708     5708
##   3851  46695869     708     5708
##   3852  46696486     708     5708
##   3853  46696534     708     5708
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<div class="sourceCode" id="cb118"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu">width</span>(<span class="no">regions</span>))</pre></body></html></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     1.0     6.0    68.0   186.2   151.0 11709.0</code></pre>
<div class="sourceCode" id="cb120"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">width</span>(<span class="no">regions</span>) <span class="kw">&gt;=</span> <span class="fl">100</span>)</pre></body></html></div>
<pre><code>## 
## FALSE  TRUE 
##  2284  1569</code></pre>
<div class="sourceCode" id="cb122"><html><body><pre class="r"><span class="co">## Keep only the ones that are at least 100 bp long</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="no">regions</span>[<span class="fu">width</span>(<span class="no">regions</span>) <span class="kw">&gt;=</span> <span class="fl">100</span>]
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">regions</span>)</pre></body></html></div>
<pre><code>## [1] 1569</code></pre>
<p>Now that we have a set of regions to work with, we proceed to build a <em>RangedSummarizedExperiment</em> object with the coverage counts, add the expanded metadata we built for the gene-level, and scale the counts. Note that <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> scales the base-pair coverage counts by default, which we turn off in order to use use <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code>.</p>
<div class="sourceCode" id="cb124"><html><body><pre class="r"><span class="co">## Compute coverage matrix for study SRP045638, only for chromosome 21</span>
<span class="co">## Takes about 4 minutes</span>
<span class="no">rse_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix</a></span>(<span class="st">"SRP045638"</span>, <span class="st">"chr21"</span>, <span class="no">regions</span>,
    <span class="kw">chunksize</span> <span class="kw">=</span> <span class="fl">2000</span>, <span class="kw">verboseLoad</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>## 2020-05-27 06:36:46 railMatrix: processing regions 1 to 1569</code></pre>
<div class="sourceCode" id="cb126"><html><body><pre class="r"><span class="co">## Use the expanded metadata we built for the gene model</span>
<span class="fu">colData</span>(<span class="no">rse_er</span>) <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">rse_gene_scaled</span>)

<span class="co">## Scale the coverage matrix</span>
<span class="no">rse_er_scaled</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="no">rse_er</span>)

<span class="co">## To highlight that we scaled the counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">rse_er</span>)</pre></body></html></div>
<p>Now that we have a scaled count matrix for the expressed regions, we can proceed with the DE analysis just like we did at the gene and exon feature levels (Figures @ref(fig:erdeanalysis1), @ref(fig:erdeanalysis2), @ref(fig:erdeanalysis3), and @ref(fig:erdeanalysis4)).</p>
<div class="sourceCode" id="cb127"><html><body><pre class="r"><span class="co">## Build DGEList object</span>
<span class="no">dge_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="fu">assays</span>(<span class="no">rse_er_scaled</span>)$<span class="no">counts</span>)

<span class="co">## Calculate normalization factors</span>
<span class="no">dge_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span>(<span class="no">dge_er</span>)

<span class="co">## Explore the data</span>
<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="no">dge_er</span>, <span class="kw">labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">prenatal</span>, <span class="fl">1</span>, <span class="fl">2</span>))</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="erdeanalysis1-1.png" alt="Multi-dimensional scaling plot of the expressed regions level data by age group." width="100%"><p class="caption">
Multi-dimensional scaling plot of the expressed regions level data by age group.
</p>
</div>
<div class="sourceCode" id="cb128"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="no">dge_er</span>, <span class="kw">labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">sex</span>, <span class="fl">1</span>, <span class="fl">1</span>))</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="erdeanalysis2-1.png" alt="Multi-dimensional scaling plot of the expressed regions level data by sex." width="100%"><p class="caption">
Multi-dimensional scaling plot of the expressed regions level data by sex.
</p>
</div>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="co">## Run voom</span>
<span class="no">v_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html">voom</a></span>(<span class="no">dge_er</span>, <span class="no">design</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="erdeanalysis3-1.png" alt="voom mean-variance plot of the expressed regions level data." width="100%"><p class="caption">
voom mean-variance plot of the expressed regions level data.
</p>
</div>
<div class="sourceCode" id="cb130"><html><body><pre class="r"><span class="co">## Run remaining parts of the DE analysis</span>
<span class="no">fit_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">v_er</span>, <span class="no">design</span>)
<span class="no">fit_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit_er</span>)</pre></body></html></div>
<div class="sourceCode" id="cb131"><html><body><pre class="r"><span class="co">## Visually explore the results</span>
<span class="kw pkg">limma</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/volcanoplot.html">volcanoplot</a></span>(<span class="no">fit_er</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">4</span>)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="erdeanalysis4-1.png" alt="Volcano plot of the expressed regions level data. Testing for prenatal and postnatal DE adjusting for sex and RIN." width="100%"><p class="caption">
Volcano plot of the expressed regions level data. Testing for prenatal and postnatal DE adjusting for sex and RIN.
</p>
</div>
<div class="sourceCode" id="cb132"><html><body><pre class="r"><span class="co">## Number of DERs</span>
<span class="no">top_er</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit_er</span>,
    <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>, <span class="kw">sort.by</span> <span class="kw">=</span> <span class="st">"none"</span>,
    <span class="kw">coef</span> <span class="kw">=</span> <span class="st">"prenatalpostnatal"</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">top_er</span>$<span class="no">adj.P.Val</span> <span class="kw">&lt;</span> <span class="fl">0.001</span>)</pre></body></html></div>
<pre><code>## 
## FALSE  TRUE 
##   609   960</code></pre>
<p>Having identified the differentially expressed regions (DERs), we can sort all regions by their adjusted p-value.</p>
<div class="sourceCode" id="cb134"><html><body><pre class="r"><span class="co">## Sort regions by q-value</span>
<span class="no">regions_by_padj</span> <span class="kw">&lt;-</span> <span class="no">regions</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">top_er</span>$<span class="no">adj.P.Val</span>, <span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">FALSE</span>)]

<span class="co">## Look at the top 10</span>
<span class="no">regions_by_padj</span>[<span class="fl">1</span>:<span class="fl">10</span>]</pre></body></html></div>
<pre><code>## GRanges object with 10 ranges and 6 metadata columns:
##        seqnames            ranges strand |     value       area indexStart
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;  &lt;numeric&gt;  &lt;integer&gt;
##   2998    chr21 44441692-44442678      * |  34.73978  34288.160   44441692
##   2144    chr21 38822674-38824916      * |  85.56379 191919.577   38822674
##   3033    chr21 44458772-44459070      * |   8.44090   2523.830   44458772
##   3029    chr21 44458526-44458644      * |   5.80784    691.133   44458526
##   3505    chr21 46250498-46250780      * |   5.68433   1608.666   46250498
##   3045    chr21 44461331-44461480      * |   5.82022    873.033   44461331
##   1356    chr21 33070821-33072413      * | 190.20982 303004.244   33070821
##   1714    chr21 36225565-36225667      * |  11.56453   1191.146   36225565
##   3773    chr21 46598568-46599629      * | 301.85950 320574.784   46598568
##   2254    chr21 39928983-39929390      * | 233.01399  95069.710   39928983
##         indexEnd cluster clusterL
##        &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##   2998  44442678     607    14072
##   2144  38824916     435    14882
##   3033  44459070     608     4968
##   3029  44458644     608     4968
##   3505  46250780     678    30649
##   3045  44461480     608     4968
##   1356  33072413     292     2261
##   1714  36225667     375     9845
##   3773  46599629     704     6544
##   2254  39929390     464     3344
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<div class="sourceCode" id="cb136"><html><body><pre class="r"><span class="fu">width</span>(<span class="no">regions_by_padj</span>[<span class="fl">1</span>:<span class="fl">10</span>])</pre></body></html></div>
<pre><code>##  [1]  987 2243  299  119  283  150 1593  103 1062  408</code></pre>
</div>
<div id="visualize-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-regions" class="anchor"></a>Visualize regions</h3>
<p>Since the DERs do not necessarily match the annotation, it is important to visualize them. The code for visualizing DERs can easily be adapted to visualize other regions. Although, the width and number of the regions will influence the computing resources needed to make the plots.</p>
<p>Because the unscaled bigWig files are available in recount2, several visualization packages can be used such as <code>epivizr</code> <span class="citation">(25)</span>, <code>wiggleplotr</code> <span class="citation">(26)</span> and <code>derfinderPlot</code> <span class="citation">(8)</span>. With all of them it is important to remember to scale the data except when visualizing the mean bigWig file for a given project.</p>
<p>First, we need to get the list of URLs for the bigWig files. We can either manually construct them or search them inside the <code>recount_url</code> table.</p>
<div class="sourceCode" id="cb138"><html><body><pre class="r"><span class="co">## Construct the list of bigWig URLs</span>
<span class="co">## They have the following form:</span>
<span class="co">## http://duffel.rail.bio/recount/</span>
<span class="co">## project id</span>
<span class="co">## /bw/</span>
<span class="co">## sample run id</span>
<span class="co">## .bw</span>
<span class="no">bws</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(
    <span class="st">"http://duffel.rail.bio/recount/SRP045638/bw/"</span>,
    <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">bigwig_file</span>
)

<span class="co">## Note that they are also present in the recount_url data.frame</span>
<span class="no">bws</span> <span class="kw">&lt;-</span> <span class="no">recount_url</span>$<span class="no">url</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(
    <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">bigwig_file</span>,
    <span class="no">recount_url</span>$<span class="no">file_name</span>
)]

<span class="co">## Use the sample run IDs as the sample names</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">bws</span>) <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">run</span></pre></body></html></div>
<p>We visualize the DERs using <code>derfinderPlot</code>, similar to what was done in the original publication <span class="citation">(13)</span>. However, we first add a little padding to the regions: 100 base-pairs on each side.</p>
<div class="sourceCode" id="cb139"><html><body><pre class="r"><span class="co">## Add 100 bp padding on each side</span>
<span class="no">regions_resized</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">resize</a></span>(<span class="no">regions_by_padj</span>[<span class="fl">1</span>:<span class="fl">10</span>],
    <span class="fu">width</span>(<span class="no">regions_by_padj</span>[<span class="fl">1</span>:<span class="fl">10</span>]) + <span class="fl">200</span>,
    <span class="kw">fix</span> <span class="kw">=</span> <span class="st">"center"</span>
)</pre></body></html></div>
<p>Next, we obtain the base-pair coverage data for each DER and scale the data to a library size of 40 million 100 base-pair reads, using the coverage AUC information we have in the metadata.</p>
<div class="sourceCode" id="cb140"><html><body><pre class="r"><span class="co">## Get the bp coverage data for the plots</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinder"</span>)
<span class="no">regionCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html">getRegionCoverage</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions_resized</span>, <span class="kw">files</span> <span class="kw">=</span> <span class="no">bws</span>,
    <span class="kw">targetSize</span> <span class="kw">=</span> <span class="fl">40</span> * <span class="fl">1e6</span> * <span class="fl">100</span>,
    <span class="kw">totalMapped</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">auc</span>,
    <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<p>The function <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> requires several pieces of annotation information for the plots that use a TxDb object. For recount2 we used Gencode v25 hg38’s annotation, which means that we need to process it manually instead of using a pre-computed TxDb package. This is where the <code>GenomicState</code> <span class="citation">(27)</span> package comes into play as it has done the heavy lifting for us already.</p>
<div class="sourceCode" id="cb141"><html><body><pre class="r"><span class="co">## Import the Gencode v25 hg38 gene annotation</span>
<span class="co">## using GenomicState</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicState"</span>)

<span class="co">## Create the TxDb object</span>
<span class="no">gencode_v25_hg38_txdb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    <span class="kw">version</span> <span class="kw">=</span> <span class="st">"25"</span>, <span class="kw">genome</span> <span class="kw">=</span> <span class="st">"hg38"</span>,
    <span class="kw">filetype</span> <span class="kw">=</span> <span class="st">"TxDb"</span>
)<span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<pre><code>## using temporary cache /tmp/Rtmpj4G2Ve/BiocFileCache</code></pre>
<pre><code>## snapshotDate(): 2020-05-23</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Loading required package: GenomicFeatures</code></pre>
<div class="sourceCode" id="cb148"><html><body><pre class="r"><span class="co">## Explore the TxDb object</span>
<span class="no">gencode_v25_hg38_txdb</span></pre></body></html></div>
<pre><code>## TxDb object:
## # Db type: TxDb
## # Supporting package: GenomicFeatures
## # Data source: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_25/gencode.v25.annotation.gtf.gz
## # Organism: Homo sapiens
## # Taxonomy ID: 9606
## # miRBase build ID: NA
## # Genome: hg38
## # transcript_nrow: 198093
## # exon_nrow: 678580
## # cds_nrow: 270623
## # Db created by: GenomicFeatures package from Bioconductor
## # Creation time: 2019-10-07 09:59:57 -0400 (Mon, 07 Oct 2019)
## # GenomicFeatures version at creation time: 1.36.4
## # RSQLite version at creation time: 2.1.2
## # DBSCHEMAVERSION: 1.2</code></pre>
<p>Now that we have a TxDb object for Gencode v25 on hg38 coordinates, we can use <code>bumphunter</code>’s <span class="citation">(28)</span> annotation functions for annotating the original 10 regions we were working with as well as the annotated genes that we can download using <code>GenomicState</code>.</p>
<div class="sourceCode" id="cb150"><html><body><pre class="r"><span class="co">## Download annotated transcripts for gencode v25</span>
<span class="no">ann_gencode_v25_hg38</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    <span class="kw">version</span> <span class="kw">=</span> <span class="st">"25"</span>, <span class="kw">genome</span> <span class="kw">=</span> <span class="st">"hg38"</span>,
    <span class="kw">filetype</span> <span class="kw">=</span> <span class="st">"AnnotatedGenes"</span>
)<span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<pre><code>## snapshotDate(): 2020-05-23</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb155"><html><body><pre class="r"><span class="co">## Annotate the regions of interest</span>
<span class="co">## Note that we are using the original regions, not the resized ones</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"bumphunter"</span>)
<span class="no">nearest_ann</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">matchGenes</a></span>(<span class="no">regions_by_padj</span>[<span class="fl">1</span>:<span class="fl">10</span>], <span class="no">ann_gencode_v25_hg38</span>)</pre></body></html></div>
<p>The final piece we need to run <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> is information about which base-pairs are exonic, intronic, etc. This is done via the <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions()</a></code> function in <code>derfinder</code>, which itself requires prior processing of the TxDb information by <code><a href="https://rdrr.io/pkg/derfinder/man/makeGenomicState.html">makeGenomicState()</a></code> that we can download with <code>GenomicState</code>.</p>
<div class="sourceCode" id="cb156"><html><body><pre class="r"><span class="co">## Download the genomic state object for Gencode v25</span>
<span class="no">gs_gencode_v25_hg38</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    <span class="kw">version</span> <span class="kw">=</span> <span class="st">"25"</span>, <span class="kw">genome</span> <span class="kw">=</span> <span class="st">"hg38"</span>,
    <span class="kw">filetype</span> <span class="kw">=</span> <span class="st">"GenomicState"</span>
)<span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<pre><code>## snapshotDate(): 2020-05-23</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb161"><html><body><pre class="r"><span class="co">## Annotate the original regions</span>
<span class="no">regions_ann</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions</a></span>(
    <span class="no">regions_resized</span>,
    <span class="no">gs_gencode_v25_hg38</span>$<span class="no">fullGenome</span>
)</pre></body></html></div>
<pre><code>## 2020-05-27 06:39:54 annotateRegions: counting</code></pre>
<pre><code>## 2020-05-27 06:39:55 annotateRegions: annotating</code></pre>
<p>We can finally use <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> to visualize the top 10 regions coloring by whether they are prenatal or postnatal samples. Known exons are shown in dark blue, introns in light blue.</p>
<div class="sourceCode" id="cb164"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderPlot"</span>)
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"region_plots.pdf"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions_resized</span>, <span class="kw">regionCoverage</span> <span class="kw">=</span> <span class="no">regionCov</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">prenatal</span>,
    <span class="kw">nearestAnnotation</span> <span class="kw">=</span> <span class="no">nearest_ann</span>,
    <span class="kw">annotatedRegions</span> <span class="kw">=</span> <span class="no">regions_ann</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">gencode_v25_hg38_txdb</span>,
    <span class="kw">scalefac</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Coverage (RP40M, 100bp)"</span>,
    <span class="kw">ask</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</pre></body></html></div>
<pre><code>## png 
##   2</code></pre>
<div class="sourceCode" id="cb166"><html><body><pre class="r"><span class="co">## Visualize DER #2</span>
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions_resized</span>, <span class="kw">regionCoverage</span> <span class="kw">=</span> <span class="no">regionCov</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">rse_er_scaled</span>)$<span class="no">prenatal</span>,
    <span class="kw">nearestAnnotation</span> <span class="kw">=</span> <span class="no">nearest_ann</span>,
    <span class="kw">annotatedRegions</span> <span class="kw">=</span> <span class="no">regions_ann</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">gencode_v25_hg38_txdb</span>,
    <span class="kw">scalefac</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Coverage (RP40M, 100bp)"</span>,
    <span class="kw">ask</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">whichRegions</span> <span class="kw">=</span> <span class="fl">2</span>
)</pre></body></html></div>
<div class="figure" style="text-align: center">
<img src="regionplots-1.png" alt="Base-pair resolution plot of differentially expressed region 2." width="100%"><p class="caption">
Base-pair resolution plot of differentially expressed region 2.
</p>
</div>
<p>In plots like Figure @ref(fig:regionplots) we can see that some DERs match known exons (DERs 2, 8, 10), some are longer than known exons (DERs 1, 7, 9), and others are exon fragments (DERs 3, 4, 5, 6) which could be due to the cutoff used. Note that DER 7 could be shorter than a known exon due to a coverage dip.</p>
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<p>In this workflow we described in detail the available data in recount2, how the coverage count matrices were computed, the metadata included in recount2 and how to get new phenotypic information from other sources. We showed how to perform a DE analysis at the gene and exon levels as well as use an annotation-agnostic approach. Finally, we explained how to visualize the base-pair information for a given set of regions. This workflow constitutes a strong basis to leverage the recount2 data for human RNA-seq analyses.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<p>This workflow was created using <code>BiocWorkflowTools</code> <span class="citation">(29)</span>. The session information is available in Supplementary File 2. The most recent version of this workflow is available via Bioconductor at <a href="http://bioconductor.org/help/workflows/" class="uri">http://bioconductor.org/help/workflows/</a>. The published version of this workflow <span class="citation">(30)</span> is available via F1000Research at <a href="https://f1000research.com/articles/6-1558/v1">f1000research.com/articles/6-1558/v1</a>.</p>
<div class="sourceCode" id="cb167"><html><body><pre class="r"><span class="co">## Final list of files created</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="st">"SRP045638"</span>)</pre></body></html></div>
<pre><code>## [1] "rse_exon.Rdata"  "rse_gene.Rdata"  "SraRunTable.txt"</code></pre>
<div class="sourceCode" id="cb169"><html><body><pre class="r"><span class="co">## Pandoc information</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rmarkdown"</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'rmarkdown'</code></pre>
<pre><code>## The following objects are masked from 'package:BiocStyle':
## 
##     html_document, md_document, pdf_document</code></pre>
<div class="sourceCode" id="cb172"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/pandoc_available.html">pandoc_version</a></span>()</pre></body></html></div>
<pre><code>## [1] '2.3.1'</code></pre>
<div class="sourceCode" id="cb174"><html><body><pre class="r"><span class="co">## Time for reproducing this workflow, in minutes</span>
<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()[<span class="fl">3</span>] / <span class="fl">60</span>, <span class="fl">1</span>)</pre></body></html></div>
<pre><code>## elapsed 
##     8.8</code></pre>
<div class="sourceCode" id="cb176"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="kw">width</span> <span class="kw">=</span> <span class="fl">100</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"sessioninfo"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span>()</pre></body></html></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.0 (2020-04-24)
##  os       Ubuntu 18.04.4 LTS          
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       UTC                         
##  date     2020-05-27                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────
##  package                * version    date       lib source                        
##  acepack                  1.4.1      2016-10-29 [1] RSPM (R 4.0.0)                
##  annotate                 1.67.0     2020-04-27 [1] Bioconductor                  
##  AnnotationDbi          * 1.51.0     2020-04-27 [1] Bioconductor                  
##  AnnotationFilter         1.13.0     2020-04-27 [1] Bioconductor                  
##  AnnotationHub          * 2.21.0     2020-05-01 [1] Bioconductor                  
##  askpass                  1.1        2019-01-13 [2] RSPM (R 4.0.0)                
##  assertthat               0.2.1      2019-03-21 [2] RSPM (R 4.0.0)                
##  backports                1.1.7      2020-05-13 [2] RSPM (R 4.0.0)                
##  base64enc                0.1-3      2015-07-28 [1] RSPM (R 4.0.0)                
##  bibtex                   0.4.2.2    2020-01-02 [1] RSPM (R 4.0.0)                
##  Biobase                * 2.49.0     2020-04-27 [1] Bioconductor                  
##  BiocFileCache          * 1.13.0     2020-04-27 [1] Bioconductor                  
##  BiocGenerics           * 0.35.2     2020-05-06 [1] Bioconductor                  
##  BiocManager              1.30.10    2019-11-16 [2] CRAN (R 4.0.0)                
##  BiocParallel             1.23.0     2020-04-27 [1] Bioconductor                  
##  BiocStyle              * 2.17.0     2020-04-27 [1] Bioconductor                  
##  BiocVersion              3.12.0     2020-04-27 [2] Bioconductor                  
##  biomaRt                  2.45.0     2020-04-27 [1] Bioconductor                  
##  Biostrings               2.57.1     2020-05-21 [1] Bioconductor                  
##  biovizBase               1.37.0     2020-04-27 [1] Bioconductor                  
##  bit                      1.1-15.2   2020-02-10 [1] RSPM (R 4.0.0)                
##  bit64                    0.9-7      2017-05-08 [1] RSPM (R 4.0.0)                
##  bitops                   1.0-6      2013-08-17 [1] RSPM (R 4.0.0)                
##  blob                     1.2.1      2020-01-20 [1] RSPM (R 4.0.0)                
##  bookdown                 0.19       2020-05-15 [1] RSPM (R 4.0.0)                
##  BSgenome                 1.57.0     2020-04-27 [1] Bioconductor                  
##  bumphunter             * 1.31.0     2020-04-28 [1] Bioconductor                  
##  caTools                  1.18.0     2020-01-17 [1] RSPM (R 4.0.0)                
##  checkmate                2.0.0      2020-02-06 [1] RSPM (R 4.0.0)                
##  cli                      2.0.2      2020-02-28 [2] RSPM (R 4.0.0)                
##  cluster                  2.1.0      2019-06-19 [3] CRAN (R 4.0.0)                
##  clusterProfiler        * 3.17.0     2020-04-27 [1] Bioconductor                  
##  codetools                0.2-16     2018-12-24 [3] CRAN (R 4.0.0)                
##  colorspace               1.4-1      2019-03-18 [1] RSPM (R 4.0.0)                
##  cowplot                  1.0.0      2019-07-11 [1] RSPM (R 4.0.0)                
##  crayon                   1.3.4      2017-09-16 [2] RSPM (R 4.0.0)                
##  curl                     4.3        2019-12-02 [2] RSPM (R 4.0.0)                
##  data.table               1.12.8     2019-12-09 [1] RSPM (R 4.0.0)                
##  DBI                      1.1.0      2019-12-15 [1] RSPM (R 4.0.0)                
##  dbplyr                 * 1.4.3      2020-04-19 [1] RSPM (R 4.0.0)                
##  DEFormats                1.17.0     2020-04-27 [1] Bioconductor                  
##  DelayedArray           * 0.15.1     2020-05-01 [1] Bioconductor                  
##  derfinder              * 1.23.1     2020-05-10 [1] Bioconductor                  
##  derfinderHelper          1.23.1     2020-05-10 [1] Bioconductor                  
##  derfinderPlot          * 1.23.1     2020-05-10 [1] Bioconductor                  
##  desc                     1.2.0      2018-05-01 [2] RSPM (R 4.0.0)                
##  DESeq2                 * 1.29.4     2020-05-16 [1] Bioconductor                  
##  dichromat                2.0-0      2013-01-24 [1] RSPM (R 4.0.0)                
##  digest                   0.6.25     2020-02-23 [2] RSPM (R 4.0.0)                
##  DO.db                    2.9        2020-05-27 [1] Bioconductor                  
##  doRNG                    1.8.2      2020-01-27 [1] RSPM (R 4.0.0)                
##  DOSE                     3.15.0     2020-04-27 [1] Bioconductor                  
##  downloader               0.4        2015-07-09 [1] RSPM (R 4.0.0)                
##  dplyr                    0.8.5      2020-03-07 [1] RSPM (R 4.0.0)                
##  edgeR                  * 3.31.0     2020-04-27 [1] Bioconductor                  
##  ellipsis                 0.3.1      2020-05-15 [2] RSPM (R 4.0.0)                
##  enrichplot               1.9.1      2020-04-29 [1] Bioconductor                  
##  ensembldb                2.13.1     2020-05-06 [1] Bioconductor                  
##  europepmc                0.3        2018-04-20 [1] RSPM (R 4.0.0)                
##  evaluate                 0.14       2019-05-28 [2] RSPM (R 4.0.0)                
##  fansi                    0.4.1      2020-01-08 [2] RSPM (R 4.0.0)                
##  farver                   2.0.3      2020-01-16 [1] RSPM (R 4.0.0)                
##  fastmap                  1.0.1      2019-10-08 [1] RSPM (R 4.0.0)                
##  fastmatch                1.1-0      2017-01-28 [1] RSPM (R 4.0.0)                
##  fgsea                    1.15.0     2020-04-27 [1] Bioconductor                  
##  foreach                * 1.5.0      2020-03-30 [1] RSPM (R 4.0.0)                
##  foreign                  0.8-78     2020-04-13 [3] CRAN (R 4.0.0)                
##  Formula                  1.2-3      2018-05-03 [1] RSPM (R 4.0.0)                
##  fs                       1.4.1      2020-04-04 [2] RSPM (R 4.0.0)                
##  gdata                    2.18.0     2017-06-06 [1] RSPM (R 4.0.0)                
##  genefilter               1.71.0     2020-04-27 [1] Bioconductor                  
##  geneplotter              1.67.0     2020-04-27 [1] Bioconductor                  
##  generics                 0.0.2      2018-11-29 [1] RSPM (R 4.0.0)                
##  GenomeInfoDb           * 1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomeInfoDbData         1.2.3      2020-05-27 [1] Bioconductor                  
##  GenomicAlignments        1.25.1     2020-05-21 [1] Bioconductor                  
##  GenomicFeatures        * 1.41.0     2020-04-27 [1] Bioconductor                  
##  GenomicFiles             1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicRanges          * 1.41.1     2020-05-02 [1] Bioconductor                  
##  GenomicState           * 0.99.9     2020-05-27 [1] Bioconductor                  
##  GEOquery                 2.57.0     2020-04-27 [1] Bioconductor                  
##  GGally                   1.5.0      2020-03-25 [1] RSPM (R 4.0.0)                
##  ggbio                    1.37.0     2020-04-27 [1] Bioconductor                  
##  ggforce                  0.3.1      2019-08-20 [1] RSPM (R 4.0.0)                
##  ggplot2                  3.3.0      2020-03-05 [1] RSPM (R 4.0.0)                
##  ggplotify                0.0.5      2020-03-12 [1] RSPM (R 4.0.0)                
##  ggraph                   2.0.3      2020-05-20 [1] RSPM (R 4.0.0)                
##  ggrepel                  0.8.2      2020-03-08 [1] RSPM (R 4.0.0)                
##  ggridges                 0.5.2      2020-01-12 [1] RSPM (R 4.0.0)                
##  glue                     1.4.1      2020-05-13 [2] RSPM (R 4.0.0)                
##  GO.db                    3.11.4     2020-05-27 [1] Bioconductor                  
##  GOSemSim                 2.15.0     2020-04-27 [1] Bioconductor                  
##  gplots                 * 3.0.3      2020-02-25 [1] RSPM (R 4.0.0)                
##  graph                    1.67.0     2020-04-27 [1] Bioconductor                  
##  graphlayouts             0.7.0      2020-04-25 [1] RSPM (R 4.0.0)                
##  gridExtra                2.3        2017-09-09 [1] RSPM (R 4.0.0)                
##  gridGraphics             0.5-0      2020-02-25 [1] RSPM (R 4.0.0)                
##  gtable                   0.3.0      2019-03-25 [1] RSPM (R 4.0.0)                
##  gtools                   3.8.2      2020-03-31 [1] RSPM (R 4.0.0)                
##  highr                    0.8        2019-03-20 [2] RSPM (R 4.0.0)                
##  Hmisc                    4.4-0      2020-03-23 [1] RSPM (R 4.0.0)                
##  hms                      0.5.3      2020-01-08 [1] RSPM (R 4.0.0)                
##  htmlTable                1.13.3     2019-12-04 [1] RSPM (R 4.0.0)                
##  htmltools                0.4.0      2019-10-04 [2] RSPM (R 4.0.0)                
##  htmlwidgets              1.5.1      2019-10-08 [2] RSPM (R 4.0.0)                
##  httpuv                   1.5.2      2019-09-11 [1] RSPM (R 4.0.0)                
##  httr                     1.4.1      2019-08-05 [2] RSPM (R 4.0.0)                
##  igraph                   1.2.5      2020-03-19 [1] RSPM (R 4.0.0)                
##  interactiveDisplayBase   1.27.2     2020-05-21 [1] Bioconductor                  
##  IRanges                * 2.23.5     2020-05-21 [1] Bioconductor                  
##  iterators              * 1.0.12     2019-07-26 [1] RSPM (R 4.0.0)                
##  jpeg                     0.1-8.1    2019-10-24 [1] RSPM (R 4.0.0)                
##  jsonlite                 1.6.1      2020-02-02 [2] RSPM (R 4.0.0)                
##  KernSmooth               2.23-16    2019-10-15 [3] CRAN (R 4.0.0)                
##  knitcitations            1.0.10     2019-09-15 [1] RSPM (R 4.0.0)                
##  knitr                    1.28       2020-02-06 [2] RSPM (R 4.0.0)                
##  knitrBootstrap           1.0.2      2018-05-24 [1] RSPM (R 4.0.0)                
##  labeling                 0.3        2014-08-23 [1] RSPM (R 4.0.0)                
##  later                    1.0.0      2019-10-04 [2] RSPM (R 4.0.0)                
##  lattice                  0.20-41    2020-04-02 [3] CRAN (R 4.0.0)                
##  latticeExtra             0.6-29     2019-12-19 [1] RSPM (R 4.0.0)                
##  lazyeval                 0.2.2      2019-03-15 [2] RSPM (R 4.0.0)                
##  lifecycle                0.2.0      2020-03-06 [2] RSPM (R 4.0.0)                
##  limma                  * 3.45.0     2020-04-27 [1] Bioconductor                  
##  locfit                 * 1.5-9.4    2020-03-25 [1] RSPM (R 4.0.0)                
##  lubridate                1.7.8      2020-04-06 [1] RSPM (R 4.0.0)                
##  magrittr                 1.5        2014-11-22 [2] RSPM (R 4.0.0)                
##  markdown                 1.1        2019-08-07 [2] RSPM (R 4.0.0)                
##  MASS                     7.3-51.5   2019-12-20 [3] CRAN (R 4.0.0)                
##  Matrix                   1.2-18     2019-11-27 [3] CRAN (R 4.0.0)                
##  matrixStats            * 0.56.0     2020-03-13 [1] RSPM (R 4.0.0)                
##  memoise                  1.1.0      2017-04-21 [2] RSPM (R 4.0.0)                
##  mime                     0.9        2020-02-04 [2] RSPM (R 4.0.0)                
##  munsell                  0.5.0      2018-06-12 [1] RSPM (R 4.0.0)                
##  nnet                     7.3-13     2020-02-25 [3] CRAN (R 4.0.0)                
##  openssl                  1.4.1      2019-07-18 [2] RSPM (R 4.0.0)                
##  org.Hs.eg.db           * 3.11.4     2020-05-27 [1] Bioconductor                  
##  OrganismDbi              1.31.0     2020-04-27 [1] Bioconductor                  
##  pillar                   1.4.4      2020-05-05 [2] RSPM (R 4.0.0)                
##  pkgconfig                2.0.3      2019-09-22 [2] RSPM (R 4.0.0)                
##  pkgdown                  1.5.1.9000 2020-05-27 [1] Github (r-lib/pkgdown@ac78596)
##  plyr                     1.8.6      2020-03-03 [1] RSPM (R 4.0.0)                
##  png                      0.1-7      2013-12-03 [1] RSPM (R 4.0.0)                
##  polyclip                 1.10-0     2019-03-14 [1] RSPM (R 4.0.0)                
##  prettyunits              1.1.1      2020-01-24 [2] RSPM (R 4.0.0)                
##  progress                 1.2.2      2019-05-16 [1] RSPM (R 4.0.0)                
##  promises                 1.1.0      2019-10-04 [2] RSPM (R 4.0.0)                
##  ProtGenerics             1.21.0     2020-04-27 [1] Bioconductor                  
##  purrr                    0.3.4      2020-04-17 [2] RSPM (R 4.0.0)                
##  qvalue                   2.21.0     2020-04-27 [1] Bioconductor                  
##  R6                       2.4.1      2019-11-12 [2] RSPM (R 4.0.0)                
##  rappdirs                 0.3.1      2016-03-28 [1] RSPM (R 4.0.0)                
##  RBGL                     1.65.0     2020-04-27 [1] Bioconductor                  
##  RColorBrewer             1.1-2      2014-12-07 [1] RSPM (R 4.0.0)                
##  Rcpp                     1.0.4.6    2020-04-09 [2] RSPM (R 4.0.0)                
##  RCurl                    1.98-1.2   2020-04-18 [1] RSPM (R 4.0.0)                
##  readr                    1.3.1      2018-12-21 [1] RSPM (R 4.0.0)                
##  recount                * 1.15.1     2020-05-12 [1] Bioconductor                  
##  RefManageR               1.2.12     2019-04-03 [1] RSPM (R 4.0.0)                
##  regionReport           * 1.23.2     2020-05-10 [1] Bioconductor                  
##  rentrez                  1.2.2      2019-05-02 [1] RSPM (R 4.0.0)                
##  reshape                  0.8.8      2018-10-23 [1] RSPM (R 4.0.0)                
##  reshape2                 1.4.4      2020-04-09 [1] RSPM (R 4.0.0)                
##  rlang                    0.4.6      2020-05-02 [2] RSPM (R 4.0.0)                
##  rmarkdown              * 2.1        2020-01-20 [1] RSPM (R 4.0.0)                
##  rngtools                 1.5        2020-01-23 [1] RSPM (R 4.0.0)                
##  rpart                    4.1-15     2019-04-12 [3] CRAN (R 4.0.0)                
##  rprojroot                1.3-2      2018-01-03 [2] RSPM (R 4.0.0)                
##  Rsamtools                2.5.1      2020-05-21 [1] Bioconductor                  
##  RSQLite                  2.2.0      2020-01-07 [1] RSPM (R 4.0.0)                
##  rstudioapi               0.11       2020-02-07 [2] RSPM (R 4.0.0)                
##  rtracklayer              1.49.2     2020-05-21 [1] Bioconductor                  
##  rvcheck                  0.1.8      2020-03-01 [1] RSPM (R 4.0.0)                
##  S4Vectors              * 0.27.8     2020-05-24 [1] Bioconductor                  
##  scales                   1.1.1      2020-05-11 [1] RSPM (R 4.0.0)                
##  scatterpie               0.1.4      2019-11-08 [1] RSPM (R 4.0.0)                
##  sessioninfo            * 1.1.1      2018-11-05 [2] RSPM (R 4.0.0)                
##  shiny                    1.4.0.2    2020-03-13 [1] RSPM (R 4.0.0)                
##  stringi                  1.4.6      2020-02-17 [2] RSPM (R 4.0.0)                
##  stringr                  1.4.0      2019-02-10 [2] RSPM (R 4.0.0)                
##  SummarizedExperiment   * 1.19.4     2020-05-17 [1] Bioconductor                  
##  survival                 3.1-12     2020-04-10 [3] CRAN (R 4.0.0)                
##  tibble                   3.0.1      2020-04-20 [2] RSPM (R 4.0.0)                
##  tidygraph                1.2.0      2020-05-12 [1] RSPM (R 4.0.0)                
##  tidyr                    1.1.0      2020-05-20 [1] RSPM (R 4.0.0)                
##  tidyselect               1.1.0      2020-05-11 [1] RSPM (R 4.0.0)                
##  triebeard                0.3.0      2016-08-04 [1] RSPM (R 4.0.0)                
##  tweenr                   1.0.1      2018-12-14 [1] RSPM (R 4.0.0)                
##  urltools                 1.7.3      2019-04-14 [1] RSPM (R 4.0.0)                
##  VariantAnnotation        1.35.0     2020-04-27 [1] Bioconductor                  
##  vctrs                    0.3.0      2020-05-11 [2] RSPM (R 4.0.0)                
##  viridis                  0.5.1      2018-03-29 [1] RSPM (R 4.0.0)                
##  viridisLite              0.3.0      2018-02-01 [1] RSPM (R 4.0.0)                
##  withr                    2.2.0      2020-04-20 [2] RSPM (R 4.0.0)                
##  xfun                     0.14       2020-05-20 [2] RSPM (R 4.0.0)                
##  XML                      3.99-0.3   2020-01-20 [1] RSPM (R 4.0.0)                
##  xml2                     1.3.2      2020-04-23 [2] RSPM (R 4.0.0)                
##  xtable                   1.8-4      2019-04-21 [1] RSPM (R 4.0.0)                
##  XVector                  0.29.1     2020-05-21 [1] Bioconductor                  
##  yaml                     2.2.1      2020-02-01 [2] RSPM (R 4.0.0)                
##  zlibbioc                 1.35.0     2020-04-27 [1] Bioconductor                  
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="competing-interests" class="section level1">
<h1 class="hasAnchor">
<a href="#competing-interests" class="anchor"></a>Competing interests</h1>
<p>No competing interests were disclosed.</p>
</div>
<div id="grant-information" class="section level1">
<h1 class="hasAnchor">
<a href="#grant-information" class="anchor"></a>Grant information</h1>
<p>LCT and AEJ were supported by the National Institutes of Health (grant R21 MH109956-01). LCT and AN were supported by the National Institutes of Health (grant R01 GM105705).</p>
</div>
<div id="acknowledgments" class="section level1">
<h1 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h1>
<p>We would like to acknowledge the members of Andrew Jaffe (Lieber Institute for Brain Development, Johns Hopkins Medical Campus) and Alexis Battle (Department of Computer Science, Whiting School of Engineering at Johns Hopkins University) labs for feedback on the explanatory figures.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-huber2015">
<p>1. Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS, et al. Orchestrating high-throughput genomic analysis with bioconductor. Nature methods. 2015;12(2):115–21. </p>
</div>
<div id="ref-Law2016a">
<p>2. Law CW, Alhamdoosh M, Su S, Smyth GK, Ritchie ME. RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR. F1000Research. 2016;5(May):1408. </p>
</div>
<div id="ref-Love2016">
<p>3. Love MI, Anders S, Kim V, Huber W. RNA-Seq workflow: gene-level exploratory analysis and differential expression. F1000Research. 2016;4(May):1070. </p>
</div>
<div id="ref-Chen2016">
<p>4. Chen Y, Lun ATL, Smyth GK. From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline. F1000Research. 2016;5(May):1438. </p>
</div>
<div id="ref-frazee2011">
<p>5. Frazee AC, Langmead B, Leek JT. ReCount: A multi-experiment resource of analysis-ready RNA-seq gene count datasets. BMC bioinformatics. 2011;12:449. </p>
</div>
<div id="ref-airway">
<p>6. Himes, E. B, Jiang, X., Wagner, P., et al. RNA-Seq Transcriptome Profiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that Modulates Cytokine Function in Airway Smooth Muscle Cells. PLoS ONE. 2014;9(6):e99625. </p>
</div>
<div id="ref-collado2017b">
<p>7. Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD, et al. Reproducible RNA-seq analysis using recount2. Nature Biotechnology. 2017 Apr;35(4):319–21. </p>
</div>
<div id="ref-collado2017a">
<p>8. Collado-Torres L, Nellore A, Frazee AC, Wilks C, Love MI, Langmead B, et al. Flexible expressed region analysis for RNA-seq with derfinder. Nucleic Acids Research. 2017 Jan;45(2):e9–9. </p>
</div>
<div id="ref-summarizedexperiment">
<p>9. Morgan M, Obenchain V, Hester J, Pagès H. SummarizedExperiment: SummarizedExperiment container. 2017. </p>
</div>
<div id="ref-Wilks2017">
<p>10. Wilks C, Gaddipati P, Nellore A, Langmead B. Snaptron: querying and visualizing splicing across tens of thousands of RNA-seq samples. bioRxiv. 2017; </p>
</div>
<div id="ref-rail2016">
<p>11. Nellore A, Collado-Torres L, Jaffe AE, Alquicira-Hernández J, Wilks C, Pritt J, et al. Rail-RNA: Scalable analysis of RNA-seq splicing and coverage. Bioinformatics (Oxford, England). 2016 Sep; </p>
</div>
<div id="ref-lawrence2013">
<p>12. Lawrence M, Huber W, Pagès H, Aboyoun P, Carlson M, Gentleman R, et al. Software for computing and annotating genomic ranges. PLoS computational biology. 2013;9(8):e1003118. </p>
</div>
<div id="ref-jaffe2015">
<p>13. Jaffe AE, Shin J, Collado-Torres L, Leek JT, Tao R, Li C, et al. Developmental regulation of human cortex transcription and its clinical relevance at single base resolution. Nature Neuroscience. 2015 Jan;18(1):154–61. </p>
</div>
<div id="ref-Ellis2017">
<p>14. Ellis SE, Collado-Torres L, Leek J. Improving the value of public rna-seq expression data by phenotype prediction. bioRxiv. 2017; </p>
</div>
<div id="ref-TCGAbiolinks">
<p>15. Colaprico A, Silva TC, Olsen C, Garofano L, Cava C, Garolini D, et al. TCGAbiolinks: An r/bioconductor package for integrative analysis of tcga data. Nucleic Acids Research. 2015; </p>
</div>
<div id="ref-law2014voom">
<p>16. Law CW, Chen Y, Shi W, Smyth GK. Voom: Precision weights unlock linear model analysis tools for rna-seq read counts. Genome Biol. 2014;15(2):R29. </p>
</div>
<div id="ref-collado2016regionreport">
<p>17. Collado-Torres L, Jaffe AE, Leek JT. regionReport: Interactive reports for region-level and feature-level genomic analyses [version2; referees: 2 approved, 1 approved with reservations]. F1000Research. 2016 Jun;4:1–10. </p>
</div>
<div id="ref-love2014moderated">
<p>18. Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for rna-seq data with deseq2. Genome biology. 2014;15(12):1–21. </p>
</div>
<div id="ref-robinson2010">
<p>19. Robinson MD, McCarthy DJ, Smyth GK. edgeR: a bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics (Oxford, England). 2010 Jan;26(1):139–40. </p>
</div>
<div id="ref-ideal">
<p>20. Marini F. Ideal: Interactive differential expression analysis. 2017. </p>
</div>
<div id="ref-clusterProfiler">
<p>21. Yu G, Wang L-G, Han Y, He Q-Y. ClusterProfiler: An r package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012;16(5):284–7. </p>
</div>
<div id="ref-Soneson2015">
<p>22. Soneson C, Love MI, Robinson MD. Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. [version 2; referees: 2 approved]. F1000Research. 2015;4(0):1521. </p>
</div>
<div id="ref-jaffe2017">
<p>23. Jaffe AE, Straub R, Shin JH, Tao R, Gao Y, Collado-Torres L, et al. Developmental and genetic regulation of the human cortex transcriptome in schizophrenia. bioRxiv. 2017; </p>
</div>
<div id="ref-rtracklayer">
<p>24. Lawrence M, Gentleman R, Carey V. Rtracklayer: An r package for interfacing with genome browsers. Bioinformatics. 2009;25:1841–2. </p>
</div>
<div id="ref-epivizr">
<p>25. Bravo HC, Chelaru F, Smith L, Goldstein N, Kancherla J, Walter M, et al. Epivizr: R interface to epiviz web app. 2017. </p>
</div>
<div id="ref-wiggleplotr">
<p>26. Alasoo K. Wiggleplotr: Make read coverage plots from bigwig files. 2017. </p>
</div>
<div id="ref-GenomicState">
<p>27. Collado-Torres L. GenomicState: Build and access genomicstate objects for use with derfinder tools from sources like gencode [Internet]. 2019. Available from: <a href="https://github.com/LieberInstitute/GenomicState">https://github.com/LieberInstitute/GenomicState</a></p>
</div>
<div id="ref-bumphunter">
<p>28. Jaffe AE, Murakami P, Lee H, Leek JT, Fallin DM, Feinberg AP, et al. Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies. International journal of epidemiology. 2012;41(1):200–9. </p>
</div>
<div id="ref-biocWorkflow">
<p>29. Smith M, Oleś A. BiocWorkflowTools: Tools to aid the development of bioconductor workflow packages. 2017. </p>
</div>
<div id="ref-recountWorkflow">
<p>30. Collado-Torres L, Nellore A, Jaffe AE. Recount workflow: Accessing over 70,000 human rna-seq samples with bioconductor [version 1; referees: 1 approved, 2 approved with reservations]. F1000Research. 2017 Aug;6(1558). </p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
